---
title: "Untitled"
author: "Evelia Coss"
date: "7/8/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{bash Pruebas, eval=F}
# Modificaciones que hice (NO HACERLO)
sed 's/MSTRG./AtRegRNA_/g' abundance.tsv > abundance_ed.tsv
sed 's/MSTRG./BrRegRNA_/g' abundance.tsv |  sed 's/transcript://g' > abundance_ed.tsv

# Copiar = New_results_12072021/Br_all_genes_tx2gene.csv y ./Equipo_anotacion/tx2gene.csv, en la carpeta
perl -pi -e 's/[""]//g' Br_all_genes_tx2gene.csv
sed 1d Br_all_genes_tx2gene.csv | sed 's/MSTRG./BrRegRNA_/g' > Br_tx2gene_ed.csv
sed 1d At_tx2gene.csv | sed 's/MSTRG./AtRegRNA_/g' > At_tx2gene_ed.csv
cat At_tx2gene_ed.csv Br_tx2gene_ed.csv > AtBr_tx2gene_ed.csv

```

## Expresion de solo lncRNAs

```{r Cargar librerias}
library(tximport)
#install.packages("rlang")
library(tidyverse)
library(readxl)
#install.packages("magrittr")
#install.packages("dplyr")
#install.packages("vctrs")
library(magrittr) # needs to be run every time you start R and want to use %>%
library(dplyr)    # alternatively, this also loads %>%
library(statmod)
#BiocManager::install("statmod")
#BiocManager::install("DESeq2")
library("DESeq2")
library("pheatmap")
library("RColorBrewer")
library("gplots")
#BiocManager::install("apeglm")
#BiocManager::install("DESeq2")
#BiocManager::install("pheatmap")
#BiocManager::install("rhdf5")
```


```{r Solo lncRNAs-A. thaliana}
# > A. thaliana
# generar tabla de metadatos
At_metadata.tsv <- data.frame("sample" =c("KO84_At_shootR1","KO85_At_shootR2", "KO86_At_rootR1", "KO87_At_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Arabidopsis_thaliana")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
At_samples <- At_metadata.tsv
At_files   <- file.path("./allspecies_data/", At_samples$sample,"abundance.tsv")
names(At_files) <- At_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
At_tx2gene_lncRNAs   <- read.csv("./allspecies_data/At_tx2gene_lncRNAs_team.txt", sep="\t",header=TRUE) #solo lncRNAs
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
At_txi_kallisto_lncRNAs_genes <- tximport(At_files, type = "kallisto", tx2gene=At_tx2gene_lncRNAs)

#prueba
#At_txi_kallisto_lncRNAs_genes <- tximport(At_files, type = "kallisto", tx2gene=At_tx2gene_lncRNAs, countsFromAbundance ="lengthScaledTPM")

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
At_sampleTable <- data.frame("sample" = c("At_shoot_R1", "At_shoot_R2", "At_root_R1", "At_root_R2"), "dex" = At_samples$dex)
rownames(At_sampleTable) <- c("At_shoot_R1", "At_shoot_R2", "At_root_R1", "At_root_R2")
colnames(At_txi_kallisto_lncRNAs_genes$counts) <-rownames(At_sampleTable) 

# Obtener tabla
At_Abundance <- At_txi_kallisto_lncRNAs_genes$abundance
At_tpm<- At_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(At_tpm, file = "./allspecies_data/At_KallistoExpr_tpm.csv", row.names = TRUE)

head(At_tpm, 20)
```






```{r Solo lncRNAs-B.rapa}
# > B.rapa
# generar tabla de metadatos
Br_metadata.tsv <- data.frame("sample" =c("KO88_Br_shootR1","KO89_Br_shootR2", "KO90_Br_rootR1", "KO91_Br_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Brassica_rapa")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
Br_samples <- Br_metadata.tsv
Br_files   <- file.path("./allspecies_data/", Br_samples$sample,"abundance.tsv")
names(Br_files) <- Br_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
Br_tx2gene_lncRNAs   <- read.csv("./allspecies_data/Br_lncRNA_tx2gene.txt", sep="\t",header=TRUE) #solo lncRNAs
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
Br_txi_kallisto_lncRNAs_genes <- tximport(Br_files, type = "kallisto", tx2gene=Br_tx2gene_lncRNAs)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
Br_sampleTable <- data.frame("sample" = c("Br_shoot_R1", "Br_shoot_R2", "Br_root_R1", "Br_root_R2"), "dex" = Br_samples$dex)
rownames(Br_sampleTable) <- c("Br_shoot_R1", "Br_shoot_R2", "Br_root_R1", "Br_root_R2")
colnames(Br_txi_kallisto_lncRNAs_genes$counts) <-rownames(Br_sampleTable) 

# Obtener tabla
Br_Abundance <- Br_txi_kallisto_lncRNAs_genes$abundance
Br_tpm<- Br_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Br_tpm, file = "./allspecies_data/Br_KallistoExpr_tpm.csv", row.names = TRUE)

head(Br_tpm)
```

```{r Solo lncRNAs-B.oleracea}
# > B.oleracea
# generar tabla de metadatos
Bo_metadata.tsv <- data.frame("sample" =c("KO150_Bo_shootR1","KO151_Bo_shootR2", "KO152_Bo_rootR1", "KO153_Bo_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Brassica_oleracea")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
Bo_samples <- Bo_metadata.tsv
Bo_files   <- file.path("./allspecies_data/", Bo_samples$sample,"abundance.tsv")
names(Bo_files) <- Bo_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
Bo_tx2gene_lncRNAs   <- read.csv("./allspecies_data/Bo_lncRNA_tx2gene.txt", sep="\t",header=TRUE) #solo lncRNAs
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
Bo_txi_kallisto_lncRNAs_genes <- tximport(Bo_files, type = "kallisto", tx2gene=Bo_tx2gene_lncRNAs, txIn = TRUE, txOut = FALSE)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
Bo_sampleTable <- data.frame("sample" = c("Bo_shoot_R1", "Bo_shoot_R2", "Bo_root_R1", "Bo_root_R2"), "dex" = Bo_samples$dex)
rownames(Bo_sampleTable) <- c("Bo_shoot_R1", "Bo_shoot_R2", "Bo_root_R1", "Bo_root_R2")
colnames(Bo_txi_kallisto_lncRNAs_genes$counts) <-rownames(Bo_sampleTable) 

# Obtener tabla
Bo_Abundance <- Bo_txi_kallisto_lncRNAs_genes$abundance
Bo_tpm<- Bo_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Bo_tpm, file = "./allspecies_data/Bo_KallistoExpr_tpm.csv", row.names = TRUE)

head(Bo_tpm)

```

```{r Solo lncRNAs-C.rubella}
# > C. rubella
# generar tabla de metadatos
Cr_metadata.tsv <- data.frame("sample" =c("KO146_Cr_shootR1","KO147_Cr_shootR2", "KO148_Cr_rootR1", "KO149_Cr_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Capsella_rubella")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
Cr_samples <- Cr_metadata.tsv
Cr_files   <- file.path("./allspecies_data/", Cr_samples$sample,"abundance.tsv")
names(Cr_files) <- Cr_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
Cr_tx2gene_lncRNAs   <- read.csv("./allspecies_data/Cr_lncRNA_tx2gene.txt", sep="\t",header=TRUE) #solo lncRNAs
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
Cr_txi_kallisto_lncRNAs_genes <- tximport(Cr_files, type = "kallisto", tx2gene=Cr_tx2gene_lncRNAs)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
Cr_sampleTable <- data.frame("sample" = c("Cr_shoot_R1", "Cr_shoot_R2", "Cr_root_R1", "Cr_root_R2"), "dex" = Cr_samples$dex)
rownames(Cr_sampleTable) <- c("Cr_shoot_R1", "Cr_shoot_R2", "Cr_root_R1", "Cr_root_R2")
colnames(Cr_txi_kallisto_lncRNAs_genes$counts) <-rownames(Cr_sampleTable) 

# Obtener tabla
Cr_Abundance <- Cr_txi_kallisto_lncRNAs_genes$abundance
Cr_tpm<- Cr_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Cr_tpm, file = "./allspecies_data/Cr_KallistoExpr_tpm.csv", row.names = TRUE)

head(Cr_tpm)
```


```{r Solo lncRNAs-Thellungiella_parvula}
# > Thellungiella_parvula
# generar tabla de metadatos
Tp_metadata.tsv <- data.frame("sample" =c("KO92_Tp_shootR1","KO93_Tp_shootR2", "KO94_Tp_rootR1", "KO95_Tp_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Thellungiella_parvula")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
Tp_samples <- Tp_metadata.tsv
Tp_files   <- file.path("./allspecies_data/", Tp_samples$sample,"abundance.tsv")
names(Tp_files) <- Tp_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
Tp_tx2gene_lncRNAs   <- read.csv("./allspecies_data/Tp_lncRNA_tx2gene.txt", sep="\t",header=TRUE) #solo lncRNAs
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
Tp_txi_kallisto_lncRNAs_genes <- tximport(Tp_files, type = "kallisto", tx2gene=Tp_tx2gene_lncRNAs)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
Tp_sampleTable <- data.frame("sample" = c("Tp_shoot_R1", "Tp_shoot_R2", "Tp_root_R1", "Tp_root_R2"), "dex" = Br_samples$dex)
rownames(Tp_sampleTable) <- c("Tp_shoot_R1", "Tp_shoot_R2", "Tp_root_R1", "Tp_root_R2")
colnames(Tp_txi_kallisto_lncRNAs_genes$counts) <-rownames(Tp_sampleTable) 

# Obtener tabla
Tp_Abundance <- Tp_txi_kallisto_lncRNAs_genes$abundance
Tp_tpm<- Tp_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Tp_tpm, file = "./allspecies_data/Tp_KallistoExpr_tpm.csv", row.names = TRUE)

head(Tp_tpm)
```

## Seleccionar por 1 TPM (lncRNAs)

```{r lncRNAs-A_thaliana}
library(dplyr)

At_tpm_db <- as.data.frame(At_tpm)
At_tpm_db <- At_tpm_db %>% filter(KO84_At_shootR1 >=1 & KO85_At_shootR2 >=1 & KO86_At_rootR1 >=1 & KO87_At_rootR2 >=1)
At_tpm_db$lncRNAs <-row.names(At_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
At_tpm_db$lncRNAs <- gsub("MSTRG.","AtRegRNA_",At_tpm_db$lncRNAs)
row.names(At_tpm_db) <- At_tpm_db$lncRNAs
tail(At_tpm_db)

# extraer IDs
At_1TPMormore_IDs <-At_tpm_db$lncRNAs
head(At_1TPMormore_IDs)
length(At_1TPMormore_IDs)

# Guardar
write.csv(At_1TPMormore_IDs, file = "./allspecies_data/Arabidopsis_thaliana_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)
write.table(At_tpm_db, file = "./allspecies_data/Arabidopsis_thaliana_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

#cp *KallistoExpr_1tpm_RegRNAIDs.txt /mnt/s/Repositorio_scripts/Intersect_lncRNAs/Expressed_lncRNAs_RegRNAIDs
```

```{r lncRNAs-B_rapa}
library(dplyr)

Br_tpm_db <- as.data.frame(Br_tpm)
Br_tpm_db <- Br_tpm_db %>% filter(KO88_Br_shootR1 >=1 & KO89_Br_shootR2 >=1 & KO90_Br_rootR1 >=1 & KO91_Br_rootR2 >=1)
Br_tpm_db$lncRNAs <-row.names(Br_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Br_tpm_db$lncRNAs <- gsub("MSTRG.","BrRegRNA_", Br_tpm_db$lncRNAs)
row.names(Br_tpm_db) <- Br_tpm_db$lncRNAs
tail(Br_tpm_db)

# extraer IDs
Br_1TPMormore_IDs <-Br_tpm_db$lncRNAs
head(Br_1TPMormore_IDs)

# Guardar
write.csv(Br_1TPMormore_IDs, file = "./allspecies_data/Brassica_rapa_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)
write.table(Br_tpm_db, file = "./allspecies_data/Brassica_rapa_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

```


```{r lncRNAs-B_oleracea}
library(dplyr)

Bo_tpm_db <- as.data.frame(Bo_tpm)
Bo_tpm_db <- Bo_tpm_db %>% filter(KO150_Bo_shootR1 >=1 & KO151_Bo_shootR2 >=1 & KO152_Bo_rootR1 >=1 & KO153_Bo_rootR2 >=1)
Bo_tpm_db$lncRNAs <-row.names(Bo_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Bo_tpm_db$lncRNAs <- gsub("MSTRG.","BoRegRNA_",Bo_tpm_db$lncRNAs)
row.names(Bo_tpm_db) <- Bo_tpm_db$lncRNAs
tail(Bo_tpm_db)

# extraer IDs
Bo_1TPMormore_IDs <-Bo_tpm_db$lncRNAs
head(Bo_1TPMormore_IDs)

# Guardar
write.csv(Bo_1TPMormore_IDs, file = "./allspecies_data/Bo_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)
write.table(Bo_tpm_db, file = "./allspecies_data/Bo_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

```


```{r lncRNAs-C_rubella}
library(dplyr)

Cr_tpm_db <- as.data.frame(Cr_tpm)
Cr_tpm_db <- Cr_tpm_db %>% filter(KO146_Cr_shootR1 >=1 & KO147_Cr_shootR2  >=1 & KO148_Cr_rootR1 >=1 & KO149_Cr_rootR2 >=1)
Cr_tpm_db$lncRNAs <-row.names(Cr_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Cr_tpm_db$lncRNAs <- gsub("MSTRG.","CrRegRNA_",Cr_tpm_db$lncRNAs)
row.names(Cr_tpm_db) <- Cr_tpm_db$lncRNAs
tail(Cr_tpm_db)

# extraer IDs
Cr_1TPMormore_IDs <-Cr_tpm_db$lncRNAs
head(Cr_1TPMormore_IDs)

# Guardar
write.csv(Cr_1TPMormore_IDs, file = "./allspecies_data/Cr_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)
write.table(Cr_tpm_db, file = "./allspecies_data/Cr_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

```


```{r lncRNAs-T_parvula}
library(dplyr)

Tp_tpm_db <- as.data.frame(Tp_tpm)
Tp_tpm_db <- Tp_tpm_db %>% filter(KO92_Tp_shootR1 >=1 & KO93_Tp_shootR2 >=1 & KO94_Tp_rootR1 >=1 & KO95_Tp_rootR2 >=1)
Tp_tpm_db$lncRNAs <-row.names(Tp_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Tp_tpm_db$lncRNAs <- gsub("MSTRG.","TpRegRNA_",Tp_tpm_db$lncRNAs)
row.names(Tp_tpm_db) <- Tp_tpm_db$lncRNAs
tail(Tp_tpm_db)

# extraer IDs
Tp_1TPMormore_IDs <-Tp_tpm_db$lncRNAs
head(Tp_1TPMormore_IDs)

# Guardar
write.csv(Tp_1TPMormore_IDs, file = "./allspecies_data/Tp_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)
write.table(Tp_tpm_db, file = "./allspecies_data/Tp_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)


#cp *_KallistoExpr_1tpm_RegRNAIDs.txt /mnt/s/Repositorio_scripts/Expressed_lncRNAs
# cp ./allspecies_data/*_KallistoExpr_1tpm_RegRNAIDs_expression.tsv /mnt/s/Repositorio_scripts/Intersect_lncRNAs/Expressed_lncRNAs_RegRNAIDs/
```


## Expresion diferencial (lncRNAs)

```{r lncRNAs-A. thaliana}
At_ddsTxi_lncRNAs                           <- DESeqDataSetFromTximport(At_txi_kallisto_lncRNAs_genes,At_sampleTable,design=~dex) # Create a DESeq object from the tximport data
At_dds_lncRNAs                              <- DESeq(At_ddsTxi_lncRNAs) # run Differential expression analysis 
At_res_lncRNAs                              <- results(At_dds_lncRNAs)  # Save the results
mcols(At_res_lncRNAs)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
At_lncRNAs_de_gene_matrix_UP  <- subset(At_res_lncRNAs, padj < 0.05 & log2FoldChange >= 0.5)
write.table(At_lncRNAs_de_gene_matrix_UP,file ="./allspecies_data/At_lncRNAs_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

At_lncRNAs_de_gene_names_UP <- rownames(At_lncRNAs_de_gene_matrix_UP)

# extraer down genes
At_lncRNAs_de_gene_matrix_DOWN  <- subset(At_res_lncRNAs, padj < 0.05 & log2FoldChange < -0.5)
write.table(At_lncRNAs_de_gene_matrix_DOWN,file ="./allspecies_data/At_lncRNAs_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

At_lncRNAs_de_gene_names_DOWN <- rownames(At_lncRNAs_de_gene_matrix_DOWN)

# Numero de genes expresados
length(At_lncRNAs_de_gene_names_UP)
length(At_lncRNAs_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
At_lncRNAs_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","AtRegRNA_",At_lncRNAs_de_gene_names_UP)
At_lncRNAs_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","AtRegRNA_",At_lncRNAs_de_gene_names_DOWN)

tail(At_lncRNAs_de_gene_names_UP_RegRNAIDs)
tail(At_lncRNAs_de_gene_names_DOWN_RegRNAIDs)

# seleccionar por lincRNAs en shoot and root
At_team_lncRNAs_intergenicos_GENES_RegRNAIDs <-gsub("MSTRG.","AtRegRNA_",At_team_lncRNAs_intergenicos_GENES_IDs$V1)

# Repeti esto mejor en bash
#AtlincRNAs_names_shoot_RegRNAIDs <- At_lncRNAs_de_gene_names_UP_RegRNAIDs[At_lncRNAs_de_gene_names_UP_RegRNAIDs %in% At_team_lncRNAs_intergenicos_GENES_RegRNAIDs]
#AtlincRNAs_names_root_RegRNAIDs <- At_lncRNAs_de_gene_names_DOWN_RegRNAIDs[At_lncRNAs_de_gene_names_DOWN_RegRNAIDs %in% At_team_lncRNAs_intergenicos_GENES_RegRNAIDs]

# Guardar
#write.table(AtlincRNAs_names_shoot_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/AtlincRNAs_DE_GENES_root_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
#write.table(AtlincRNAs_names_root_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/AtlincRNAs_DE_GENES_shoot_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)

```

```{r lncRNAs-B.rapa}
Br_ddsTxi_lncRNAs                           <- DESeqDataSetFromTximport(Br_txi_kallisto_lncRNAs_genes,Br_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Br_dds_lncRNAs                              <- DESeq(Br_ddsTxi_lncRNAs) # run Differential expression analysis 
Br_res_lncRNAs                              <- results(Br_dds_lncRNAs)  # Save the results
mcols(Br_res_lncRNAs)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Br_lncRNAs_de_gene_matrix_UP  <- subset(Br_res_lncRNAs, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Br_lncRNAs_de_gene_matrix_UP,file ="./allspecies_data/Br_lncRNAs_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Br_lncRNAs_de_gene_names_UP <- rownames(Br_lncRNAs_de_gene_matrix_UP)

# extraer down genes
Br_lncRNAs_de_gene_matrix_DOWN  <- subset(Br_res_lncRNAs, padj < 0.05 & log2FoldChange < -0.5)
write.table(Br_lncRNAs_de_gene_matrix_DOWN,file ="./allspecies_data/Br_lncRNAs_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Br_lncRNAs_de_gene_names_DOWN <- rownames(Br_lncRNAs_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Br_lncRNAs_de_gene_names_UP)
length(Br_lncRNAs_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Br_lncRNAs_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","BrRegRNA_",Br_lncRNAs_de_gene_names_UP)
Br_lncRNAs_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","BrRegRNA_",Br_lncRNAs_de_gene_names_DOWN)

tail(Br_lncRNAs_de_gene_names_UP_RegRNAIDs)
tail(Br_lncRNAs_de_gene_names_DOWN_RegRNAIDs)

# seleccionar por lincRNAs en shoot and root
Br_lncRNAs_intergenicos_GENES_IDs<- read.table("../Correciones_geneid//Nueva_anotacion_Agosto2021/cluster_BED_gene_IDs/Br_lncRNAs_intergenicos_GENES_IDs", header =F)

Br_lncRNAs_intergenicos_GENES_RegRNAIDs <-gsub("MSTRG.","BrRegRNA_",Br_lncRNAs_intergenicos_GENES_IDs$V1)

#BrlincRNAs_names_shoot_RegRNAIDs <- Br_lncRNAs_de_gene_names_UP_RegRNAIDs[Br_lncRNAs_de_gene_names_UP_RegRNAIDs %in% Br_lncRNAs_intergenicos_GENES_RegRNAIDs]

#BrlincRNAs_names_root_RegRNAIDs <- Br_lncRNAs_de_gene_names_DOWN_RegRNAIDs[Br_lncRNAs_de_gene_names_DOWN_RegRNAIDs %in% Br_lncRNAs_intergenicos_GENES_RegRNAIDs]

# rm(BrlincRNAs_names_shoot_RegRNAIDs, BrlincRNAs_names_root_RegRNAIDs)

# Guardar
#write.table(BrlincRNAs_names_shoot_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/BrlincRNAs_DE_GENES_root_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
#write.table(BrlincRNAs_names_root_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/BrlincRNAs_DE_GENES_shoot_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
```

```{r lncRNAs-B.oleracea}
Bo_ddsTxi_lncRNAs                           <- DESeqDataSetFromTximport(Bo_txi_kallisto_lncRNAs_genes,Bo_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Bo_dds_lncRNAs                              <- DESeq(Bo_ddsTxi_lncRNAs) # run Differential expression analysis 
Bo_res_lncRNAs                              <- results(Bo_dds_lncRNAs)  # Save the results
mcols(Bo_res_lncRNAs)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Bo_lncRNAs_de_gene_matrix_UP  <- subset(Bo_res_lncRNAs, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Bo_lncRNAs_de_gene_matrix_UP,file ="./allspecies_data/Bo_lncRNAs_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Bo_lncRNAs_de_gene_names_UP <- rownames(Bo_lncRNAs_de_gene_matrix_UP)

# extraer down genes
Bo_lncRNAs_de_gene_matrix_DOWN  <- subset(Bo_res_lncRNAs, padj < 0.05 & log2FoldChange < -0.5)
write.table(Bo_lncRNAs_de_gene_matrix_DOWN,file ="./allspecies_data/Bo_lncRNAs_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Bo_lncRNAs_de_gene_names_DOWN <- rownames(Bo_lncRNAs_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Bo_lncRNAs_de_gene_names_UP)
length(Bo_lncRNAs_de_gene_names_DOWN)

head(Bo_lncRNAs_de_gene_names_UP)
head(Bo_lncRNAs_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Bo_lncRNAs_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","BoRegRNA_",Bo_lncRNAs_de_gene_names_UP)
Bo_lncRNAs_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","BoRegRNA_",Bo_lncRNAs_de_gene_names_DOWN)

tail(Bo_lncRNAs_de_gene_names_UP_RegRNAIDs)
tail(Bo_lncRNAs_de_gene_names_DOWN_RegRNAIDs)

# seleccionar por lincRNAs en shoot and root
Bo_lncRNAs_intergenicos_GENES_IDs <-read.table("../Correciones_geneid/Annotation_CrBo/cluster_bed_genes/Bo_lncRNAs_intergenicos_GENES_IDs")
Bo_lncRNAs_intergenicos_GENES_RegRNAIDs <-gsub("MSTRG.","BoRegRNA_", Bo_lncRNAs_intergenicos_GENES_IDs$V1)

#BolincRNAs_names_shoot_RegRNAIDs <- Bo_lncRNAs_de_gene_names_UP_RegRNAIDs[Bo_lncRNAs_de_gene_names_UP_RegRNAIDs %in% Bo_lncRNAs_intergenicos_GENES_RegRNAIDs]

#BolincRNAs_names_root_RegRNAIDs <- Bo_lncRNAs_de_gene_names_DOWN_RegRNAIDs[Bo_lncRNAs_de_gene_names_DOWN_RegRNAIDs %in% Bo_lncRNAs_intergenicos_GENES_RegRNAIDs]

# rm(BolincRNAs_names_shoot_RegRNAIDs,BolincRNAs_names_root_RegRNAIDs)
# Guardar
#write.table(BolincRNAs_names_shoot_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/BolincRNAs_DE_GENES_root_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
#write.table(BolincRNAs_names_root_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/BolincRNAs_DE_GENES_shoot_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)

```


```{r lncRNAs-C.rubella}
Cr_ddsTxi_lncRNAs                           <- DESeqDataSetFromTximport(Cr_txi_kallisto_lncRNAs_genes,Cr_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Cr_dds_lncRNAs                              <- DESeq(Cr_ddsTxi_lncRNAs) # run Differential expression analysis 
Cr_res_lncRNAs                              <- results(Cr_dds_lncRNAs)  # Save the results
mcols(Cr_res_lncRNAs)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Cr_lncRNAs_de_gene_matrix_UP  <- subset(Cr_res_lncRNAs, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Cr_lncRNAs_de_gene_matrix_UP,file ="./allspecies_data/Cr_lncRNAs_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Cr_lncRNAs_de_gene_names_UP <- rownames(Cr_lncRNAs_de_gene_matrix_UP)

# extraer down genes
Cr_lncRNAs_de_gene_matrix_DOWN  <- subset(Cr_res_lncRNAs, padj < 0.05 & log2FoldChange < -0.5)
write.table(Cr_lncRNAs_de_gene_matrix_DOWN,file ="./allspecies_data/Cr_lncRNAs_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Cr_lncRNAs_de_gene_names_DOWN <- rownames(Cr_lncRNAs_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Cr_lncRNAs_de_gene_names_UP)
length(Cr_lncRNAs_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Cr_lncRNAs_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","CrRegRNA_",Cr_lncRNAs_de_gene_names_UP)
Cr_lncRNAs_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","CrRegRNA_",Cr_lncRNAs_de_gene_names_DOWN)

tail(Cr_lncRNAs_de_gene_names_UP_RegRNAIDs)
tail(Cr_lncRNAs_de_gene_names_DOWN_RegRNAIDs)

# seleccionar por lincRNAs en shoot and root
Cr_lncRNAs_intergenicos_GENES_IDs <-read.table("../Correciones_geneid/Annotation_CrBo/cluster_bed_genes/Cr_lncRNAs_intergenicos_GENES_IDs")
Cr_lncRNAs_intergenicos_GENES_RegRNAIDs <-gsub("MSTRG.","CrRegRNA_", Cr_lncRNAs_intergenicos_GENES_IDs$V1)

#CrlincRNAs_names_shoot_RegRNAIDs <- Cr_lncRNAs_de_gene_names_UP_RegRNAIDs[Cr_lncRNAs_de_gene_names_UP_RegRNAIDs %in% Cr_lncRNAs_intergenicos_GENES_RegRNAIDs]

#CrlincRNAs_names_root_RegRNAIDs <- Cr_lncRNAs_de_gene_names_DOWN_RegRNAIDs[Cr_lncRNAs_de_gene_names_DOWN_RegRNAIDs %in% Cr_lncRNAs_intergenicos_GENES_RegRNAIDs]

# rm(CrlincRNAs_names_shoot_RegRNAIDs, CrlincRNAs_names_root_RegRNAIDs)

# Guardar
#write.table(CrlincRNAs_names_shoot_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/CrlincRNAs_DE_GENES_root_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
#write.table(CrlincRNAs_names_root_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/CrlincRNAs_DE_GENES_shoot_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)

```


```{r lncRNAs-T.parvula}
Tp_ddsTxi_lncRNAs                           <- DESeqDataSetFromTximport(Tp_txi_kallisto_lncRNAs_genes,Tp_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Tp_dds_lncRNAs                              <- DESeq(Tp_ddsTxi_lncRNAs) # run Differential expression analysis 
Tp_res_lncRNAs                              <- results(Tp_dds_lncRNAs)  # Save the results
mcols(Tp_res_lncRNAs)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Tp_lncRNAs_de_gene_matrix_UP  <- subset(Tp_res_lncRNAs, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Tp_lncRNAs_de_gene_matrix_UP,file ="./allspecies_data/Tp_lncRNAs_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Tp_lncRNAs_de_gene_names_UP <- rownames(Tp_lncRNAs_de_gene_matrix_UP)

# extraer down genes
Tp_lncRNAs_de_gene_matrix_DOWN  <- subset(Tp_res_lncRNAs, padj < 0.05 & log2FoldChange < -0.5)
write.table(Tp_lncRNAs_de_gene_matrix_DOWN,file ="./allspecies_data/Tp_lncRNAs_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Tp_lncRNAs_de_gene_names_DOWN <- rownames(Tp_lncRNAs_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Tp_lncRNAs_de_gene_names_UP)
length(Tp_lncRNAs_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Tp_lncRNAs_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","TpRegRNA_",Tp_lncRNAs_de_gene_names_UP)
Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","TpRegRNA_",Tp_lncRNAs_de_gene_names_DOWN)

tail(Tp_lncRNAs_de_gene_names_UP_RegRNAIDs)
tail(Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs)

# seleccionar por lincRNAs en shoot and root
Tp_lncRNAs_intergenicos_GENES_IDs<- read.table("../Correciones_geneid/Nueva_anotacion_Agosto2021/cluster_BED_gene_IDs/Tp_lncRNAs_intergenicos_GENES_IDs", header =F)

Tp_lncRNAs_intergenicos_GENES_RegRNAIDs <-gsub("MSTRG.","TpRegRNA_", Tp_lncRNAs_intergenicos_GENES_IDs$V1)

#TplincRNAs_names_shoot_RegRNAIDs <- Tp_lncRNAs_de_gene_names_UP_RegRNAIDs[Tp_lncRNAs_de_gene_names_UP_RegRNAIDs %in% Tp_lncRNAs_intergenicos_GENES_RegRNAIDs]

#TplincRNAs_names_root_RegRNAIDs <- Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs[Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs %in% Tp_lncRNAs_intergenicos_GENES_RegRNAIDs]

# rm(TplincRNAs_names_root_RegRNAIDs, TplincRNAs_names_shoot_RegRNAIDs)

# Guardar
#write.table(TplincRNAs_names_shoot_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/TplincRNAs_DE_GENES_root_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
#write.table(TplincRNAs_names_root_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/TplincRNAs_DE_GENES_shoot_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)

```



## Cuentas normalizadas para graficas (rlog) (lncRNAs)

```{r lncRNAs-A.thaliana}
At_lncRNAs_normalized <- rlog(At_dds_lncRNAs, blind=FALSE) # result rld, vst
At_lncRNAs_normalized_db <- as.data.frame(assay(At_lncRNAs_normalized))
head(At_lncRNAs_normalized_db)


At_lncRNAs_normalized_db_copy <- At_lncRNAs_normalized_db
At_lncRNAs_normalized_db_copy$GeneID <- row.names(At_lncRNAs_normalized_db_copy)
# guardar
write.table(At_lncRNAs_normalized_db_copy,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_lncRNAs_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = F)
write.table(At_lncRNAs_normalized_db_copy,file ="allspecies_data/At_lncRNAs_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)

```

```{r lncRNAs-B.rapa}
Br_lncRNAs_normalized <- rlog(Br_dds_lncRNAs, blind=FALSE) # result rld, vst
Br_lncRNAs_normalized_db <- as.data.frame(assay(Br_lncRNAs_normalized))
head(Br_lncRNAs_normalized_db)

Br_lncRNAs_normalized_db_copy <- Br_lncRNAs_normalized_db
Br_lncRNAs_normalized_db_copy$GeneID <- row.names(Br_lncRNAs_normalized_db_copy)

# guardar
write.table(Br_lncRNAs_normalized_db_copy,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_lncRNAs_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```

```{r lncRNAs-B.oleracea}
Bo_lncRNAs_normalized <- rlog(Bo_dds_lncRNAs, blind=FALSE) # result rld, vst
Bo_lncRNAs_normalized_db <- as.data.frame(assay(Bo_lncRNAs_normalized))
head(Bo_lncRNAs_normalized_db)

Bo_lncRNAs_normalized_db_copy <- Bo_lncRNAs_normalized_db
Bo_lncRNAs_normalized_db_copy$GeneID <- row.names(Bo_lncRNAs_normalized_db_copy)

# guardar
write.table(Bo_lncRNAs_normalized_db_copy,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_lncRNAs_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


```{r lncRNAs-C.rubella}
Cr_lncRNAs_normalized <- rlog(Cr_dds_lncRNAs, blind=FALSE) # result rld, vst
Cr_lncRNAs_normalized_db <- as.data.frame(assay(Cr_lncRNAs_normalized))
head(Cr_lncRNAs_normalized_db)


Cr_lncRNAs_normalized_db_copy <- Cr_lncRNAs_normalized_db
Cr_lncRNAs_normalized_db_copy$GeneID <- row.names(Cr_lncRNAs_normalized_db_copy)

# guardar
write.table(Cr_lncRNAs_normalized_db_copy,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_lncRNAs_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


```{r lncRNAs-T.parvula}
Tp_lncRNAs_normalized <- rlog(Tp_dds_lncRNAs, blind=FALSE) # result rld, vst
Tp_lncRNAs_normalized_db <- as.data.frame(assay(Tp_lncRNAs_normalized))
head(Tp_lncRNAs_normalized_db)

Tp_lncRNAs_normalized_db_copy <- Tp_lncRNAs_normalized_db
Tp_lncRNAs_normalized_db_copy$GeneID <- row.names(Tp_lncRNAs_normalized_db_copy)

# guardar
write.table(Tp_lncRNAs_normalized_db_copy,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_lncRNAs_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


## Expresion en ambos tejidos (>=1TPM) (NO CORRER, REPETIR EN BASH)

```{r lncRNAs-A.thaliana}
# Apartir del filtro de 1TPM seleccionar que se encuentre expresado en ambos tejidos
At_bothtissues_tpm_db<- At_tpm_db %>% filter(!(lncRNAs %in% c(At_lncRNAs_de_gene_names_DOWN_RegRNAIDs, At_lncRNAs_de_gene_names_UP_RegRNAIDs)))
head(At_bothtissues_tpm_db)
tail(At_bothtissues_tpm_db)

At_bothtissues_tpm_AtRegRNAIDs <- At_bothtissues_tpm_db$lncRNAs

length(At_bothtissues_tpm_AtRegRNAIDs)
# guardar
write.table(At_bothtissues_tpm_AtRegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/AtlncRNAs_DE_GENES_bothtissues_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
```


```{r lncRNAs-B.rapa}
# Apartir del filtro de 1TPM seleccionar que se encuentre expresado en ambos tejidos
Br_bothtissues_tpm_db<- Br_tpm_db %>% filter(!(lncRNAs %in% c(Br_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Br_lncRNAs_de_gene_names_UP_RegRNAIDs)))
head(Br_bothtissues_tpm_db)
tail(Br_bothtissues_tpm_db)

Br_bothtissues_tpm_RegRNAIDs <-Br_bothtissues_tpm_db$lncRNAs

length(Br_bothtissues_tpm_RegRNAIDs)

# guardar
write.table(Br_bothtissues_tpm_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/BrlncRNAs_DE_GENES_bothtissues_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
```

```{r lncRNAs-B.oleracea}
# Apartir del filtro de 1TPM seleccionar que se encuentre expresado en ambos tejidos
Bo_bothtissues_tpm_db<- Bo_tpm_db %>% filter(!(lncRNAs %in% c(Bo_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Bo_lncRNAs_de_gene_names_UP_RegRNAIDs)))
head(Bo_bothtissues_tpm_db)
tail(Bo_bothtissues_tpm_db)

Bo_bothtissues_tpm_RegRNAIDs <-Bo_bothtissues_tpm_db$lncRNAs

length(Bo_bothtissues_tpm_RegRNAIDs)

# guardar
write.table(Bo_bothtissues_tpm_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/BolncRNAs_DE_GENES_bothtissues_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
```


```{r lncRNAs-C.rubella}
# Apartir del filtro de 1TPM seleccionar que se encuentre expresado en ambos tejidos
Cr_bothtissues_tpm_db<- Cr_tpm_db %>% filter(!(lncRNAs %in% c(Cr_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Cr_lncRNAs_de_gene_names_UP_RegRNAIDs)))
head(Cr_bothtissues_tpm_db)
tail(Cr_bothtissues_tpm_db)

Cr_bothtissues_tpm_RegRNAIDs <-Cr_bothtissues_tpm_db$lncRNAs

length(Cr_bothtissues_tpm_RegRNAIDs)
# guardar
write.table(Cr_bothtissues_tpm_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/CrlncRNAs_DE_GENES_bothtissues_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
```


```{r lncRNAs-T.parvula}
# Apartir del filtro de 1TPM seleccionar que se encuentre expresado en ambos tejidos
Tp_bothtissues_tpm_db<- Tp_tpm_db %>% filter(!(lncRNAs %in% c(Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Tp_lncRNAs_de_gene_names_UP_RegRNAIDs)))
head(Tp_bothtissues_tpm_db)
tail(Tp_bothtissues_tpm_db)

Tp_bothtissues_tpm_RegRNAIDs <-Tp_bothtissues_tpm_db$lncRNAs

length(Tp_bothtissues_tpm_RegRNAIDs)

# guardar
write.table(Tp_bothtissues_tpm_RegRNAIDs,file ="S:/Repositorio_scripts/Expressed_lncRNAs/TplncRNAs_DE_GENES_bothtissues_RegRNAIDs", quote=FALSE, sep="\t", row.names = F)
```


## Expresion de todos LOS GENES ANOTADOS

```{r Cargar librerias}
library(tximport)
library(tidyverse)
library(readxl)
#install.packages("magrittr")
#install.packages("dplyr")
library(magrittr) # needs to be run every time you start R and want to use %>%
library(dplyr)    # alternatively, this also loads %>%
library(statmod)
#BiocManager::install("statmod")
library("DESeq2")
library("pheatmap")
library("RColorBrewer")
library("gplots")
#BiocManager::install("apeglm")
#BiocManager::install("DESeq2")
#BiocManager::install("pheatmap")
#BiocManager::install("rhdf5")
```


```{r ALL-A. thaliana}
# > A. thaliana
# generar tabla de metadatos
#At_metadata.tsv <- data.frame("sample" =c("KO84_At_shootR1","KO85_At_shootR2", "KO86_At_rootR1", "KO87_At_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Arabidopsis_thaliana")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
#At_samples <- At_metadata.tsv
#At_files   <- file.path("./allspecies_data/", At_samples$sample,"abundance.tsv")
#names(At_files) <- At_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
At_tx2gene_all   <- read.csv("./allspecies_data/At_all_genes_tx2gene.csv", sep=",",header=TRUE) #solo lncRNAs
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
At_txi_kallisto_all_genes <- tximport(At_files, type = "kallisto", tx2gene=At_tx2gene_all)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
#At_sampleTable <- data.frame("sample" = c("At_shoot_R1", "At_shoot_R2", "At_root_R1", "At_root_R2"), "dex" = At_samples$dex)
#rownames(At_sampleTable) <- c("At_shoot_R1", "At_shoot_R2", "At_root_R1", "At_root_R2")
colnames(At_txi_kallisto_all_genes$counts) <-rownames(At_sampleTable) 

# Obtener tabla
At_all_Abundance <- At_txi_kallisto_all_genes$abundance
At_all_tpm<- At_all_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(At_all_tpm, file = "./allspecies_data/Arabidopsis_thaliana_all_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(At_all_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Arabidopsis_thaliana_all_KallistoExpr_tpm.csv", row.names = TRUE)

head(At_all_tpm)
```

```{r ALL-B.rapa}
# > B.rapa
# generar tabla de metadatos
#Br_metadata.tsv <- data.frame("sample" =c("KO88_Br_shootR1","KO89_Br_shootR2", "KO90_Br_rootR1", "KO91_Br_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Brassica_rapa")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
#Br_samples <- Br_metadata.tsv
#Br_files   <- file.path("./allspecies_data/", Br_samples$sample,"abundance.tsv")
#names(Br_files) <- Br_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
Br_tx2gene_all   <- read.csv("./allspecies_data/Br_all_genes_tx2gene.csv", sep=",",header=TRUE) #solo lncRNAs
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
Br_txi_kallisto_all_genes <- tximport(Br_files, type = "kallisto", tx2gene=Br_tx2gene_all)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
#Br_sampleTable <- data.frame("sample" = c("Br_shoot_R1", "Br_shoot_R2", "Br_root_R1", "Br_root_R2"), "dex" = Br_samples$dex)
#rownames(Br_sampleTable) <- c("Br_shoot_R1", "Br_shoot_R2", "Br_root_R1", "Br_root_R2")
colnames(Br_txi_kallisto_all_genes$counts) <-rownames(Br_sampleTable) 

# Obtener tabla
Br_all_Abundance <- Br_txi_kallisto_all_genes$abundance
Br_all_tpm<- Br_all_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Br_all_tpm, file = "./allspecies_data/Brassica_rapa_all_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Br_all_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_rapa_all_KallistoExpr_tpm.csv", row.names = TRUE)

head(Br_all_tpm)
head(Br_tx2gene_all)
```


```{r ALL-B.oleracea}
# > B.oleracea
# generar tabla de metadatos
#Bo_metadata.tsv <- data.frame("sample" =c("KO150_Bo_shootR1","KO151_Bo_shootR2", "KO152_Bo_rootR1", "KO153_Bo_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Brassica_oleracea")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
#Bo_samples <- Bo_metadata.tsv
#Bo_files   <- file.path("./allspecies_data/", Bo_samples$sample,"abundance.tsv")
#names(Bo_files) <- Bo_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
Bo_tx2gene_all   <- read.csv("./allspecies_data/Bo_all_genes_tx2gene.csv", sep="\t",header=TRUE)
Bo_tx2gene_all <- Bo_tx2gene_all[,c(2,1)] # cambiar lugares
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
Bo_txi_kallisto_all_genes <- tximport(Bo_files, type = "kallisto", tx2gene=Bo_tx2gene_all, txIn = TRUE, txOut = FALSE)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
#Bo_sampleTable <- data.frame("sample" = c("Bo_shoot_R1", "Bo_shoot_R2", "Bo_root_R1", "Bo_root_R2"), "dex" = Bo_samples$dex)
#rownames(Bo_sampleTable) <- c("Bo_shoot_R1", "Bo_shoot_R2", "Bo_root_R1", "Bo_root_R2")
colnames(Bo_txi_kallisto_all_genes$counts) <-rownames(Bo_sampleTable) 

# Obtener tabla
Bo_all_Abundance <- Bo_txi_kallisto_all_genes$abundance
Bo_all_tpm<- Bo_all_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Bo_all_tpm, file = "./allspecies_data/Brassica_oleracea_all_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Bo_all_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_oleracea_all_KallistoExpr_tpm.csv", row.names = TRUE)

head(Bo_all_tpm)

```

```{r}
#BoRegRNA_31881

Bo_txi_kallisto_all_gened_db <- as.data.frame(Bo_txi_kallisto_all_genes)
Bo_txi_kallisto_all_gened_db["MSTRG.31881",]
```



```{r ALL-C.rubella}
# > C. rubella
# generar tabla de metadatos
#Cr_metadata.tsv <- data.frame("sample" =c("KO146_Cr_shootR1","KO147_Cr_shootR2", "KO148_Cr_rootR1", "KO149_Cr_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Capsella_rubella")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
#Cr_samples <- Cr_metadata.tsv
#Cr_files   <- file.path("./allspecies_data/", Cr_samples$sample,"abundance.tsv")
#names(Cr_files) <- Cr_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
Cr_tx2gene_all   <- read.csv("./allspecies_data/Cr_all_genes_tx2gene.csv", sep="\t",header=TRUE) #solo lncRNAs
Cr_tx2gene_all   <- Cr_tx2gene_all[,c(2,1)]
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
Cr_txi_kallisto_all_genes <- tximport(Cr_files, type = "kallisto", tx2gene=Cr_tx2gene_all)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
#Cr_sampleTable <- data.frame("sample" = c("Cr_shoot_R1", "Cr_shoot_R2", "Cr_root_R1", "Cr_root_R2"), "dex" = Cr_samples$dex)
#rownames(Cr_sampleTable) <- c("Cr_shoot_R1", "Cr_shoot_R2", "Cr_root_R1", "Cr_root_R2")
colnames(Cr_txi_kallisto_all_genes$counts) <-rownames(Cr_sampleTable) 

# Obtener tabla
Cr_all_Abundance <- Cr_txi_kallisto_all_genes$abundance
Cr_all_tpm<- Cr_all_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Cr_all_tpm, file = "./allspecies_data/Capsella_rubella_all_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Cr_all_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Capsella_rubella_all_KallistoExpr_tpm.csv", row.names = TRUE)

head(Cr_all_tpm)
```


```{r ALL-Thellungiella_parvula}
# > Thellungiella_parvula
# generar tabla de metadatos
#Tp_metadata.tsv <- data.frame("sample" =c("KO92_Tp_shootR1","KO93_Tp_shootR2", "KO94_Tp_rootR1", "KO95_Tp_rootR2"), "dex" = c(rep("shoot",2), rep("root",2)), "species" = "Thellungiella_parvula")

# Anotacion articulo
# Cargar los archivos por nombre del archivo (ubicacion empleando la anotacion de los 224 transcriptomas) 
#Tp_samples <- Tp_metadata.tsv
#Tp_files   <- file.path("./allspecies_data/", Tp_samples$sample,"abundance.tsv")
#names(Tp_files) <- Tp_samples$sample

# Empleando la misma anotacion del articulo
# previamente generado con clusterBED y R
Tp_tx2gene_all   <- read.csv("./allspecies_data/Tp_all_genes_tx2gene.csv", sep=",",header=TRUE) #solo lncRNAs
#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
Tp_txi_kallisto_all_genes <- tximport(Tp_files, type = "kallisto", tx2gene=Tp_tx2gene_all)

# Default = txIn = TRUE, txOut = FALSE, countsFromAbundance = "no"

# nombre de los transcriptomas
#Tp_sampleTable <- data.frame("sample" = c("Tp_shoot_R1", "Tp_shoot_R2", "Tp_root_R1", "Tp_root_R2"), "dex" = Br_samples$dex)
#rownames(Tp_sampleTable) <- c("Tp_shoot_R1", "Tp_shoot_R2", "Tp_root_R1", "Tp_root_R2")
colnames(Tp_txi_kallisto_all_genes$counts) <-rownames(Tp_sampleTable) 

# Obtener tabla
Tp_all_Abundance <- Tp_txi_kallisto_all_genes$abundance
Tp_all_tpm<- Tp_all_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Tp_all_tpm, file = "./allspecies_data/Thellungiella_parvula_all_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Tp_all_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Thellungiella_parvula_all_KallistoExpr_tpm.csv", row.names = TRUE)

head(Tp_all_tpm)
```

## Seleccionar por 1 TPM (TODOS LOS TRANSCRITOS)

```{r ALL-A_thaliana}
library(dplyr)

At_all_tpm_db <- as.data.frame(At_all_tpm)
At_all_tpm_db <- At_all_tpm_db %>% filter(KO84_At_shootR1 >=1 & KO85_At_shootR2 >=1 & KO86_At_rootR1 >=1 & KO87_At_rootR2 >=1)
At_all_tpm_db$lncRNAs <-row.names(At_all_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
At_all_tpm_db$lncRNAs <- gsub("MSTRG.","AtRegRNA_",At_all_tpm_db$lncRNAs)
row.names(At_all_tpm_db) <- At_all_tpm_db$lncRNAs
tail(At_all_tpm_db)

# extraer IDs
At_all_1TPMormore_IDs <-At_all_tpm_db$lncRNAs
head(At_all_1TPMormore_IDs)
length(At_all_1TPMormore_IDs)

# Guardar
write.csv(At_all_1TPMormore_IDs, file = "./allspecies_data/Arabidopsis_thaliana_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(At_all_tpm_db, file = "./allspecies_data/Arabidopsis_thaliana_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(At_all_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Arabidopsis_thaliana_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(At_all_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Arabidopsis_thaliana_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

#cp *KallistoExpr_1tpm_RegRNAIDs.txt /mnt/s/Repositorio_scripts/Intersect_lncRNAs/Expressed_lncRNAs_RegRNAIDs
```

```{r ALL-B_rapa}
library(dplyr)

Br_all_tpm_db <- as.data.frame(Br_all_tpm)
Br_all_tpm_db <- Br_all_tpm_db %>% filter(KO88_Br_shootR1 >=1 & KO89_Br_shootR2 >=1 & KO90_Br_rootR1 >=1 & KO91_Br_rootR2 >=1)
Br_all_tpm_db$lncRNAs <-row.names(Br_all_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Br_all_tpm_db$lncRNAs <- gsub("MSTRG.","BrRegRNA_", Br_all_tpm_db$lncRNAs)
row.names(Br_all_tpm_db) <- Br_all_tpm_db$lncRNAs
tail(Br_all_tpm_db)

# extraer IDs
Br_all_1TPMormore_IDs <-Br_all_tpm_db$lncRNAs
head(Br_all_1TPMormore_IDs)

# Guardar
write.csv(Br_all_1TPMormore_IDs, file = "./allspecies_data/Brassica_rapa_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Br_all_tpm_db, file = "./allspecies_data/Brassica_rapa_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(Br_all_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_rapa_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Br_all_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_rapa_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)
```


```{r ALL-B_oleracea}
library(dplyr)

Bo_all_tpm_db <- as.data.frame(Bo_all_tpm)
Bo_all_tpm_db <- Bo_all_tpm_db %>% filter(KO150_Bo_shootR1 >=1 & KO151_Bo_shootR2 >=1 & KO152_Bo_rootR1 >=1 & KO153_Bo_rootR2 >=1)
Bo_all_tpm_db$lncRNAs <-row.names(Bo_all_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Bo_all_tpm_db$lncRNAs <- gsub("MSTRG.","BoRegRNA_",Bo_all_tpm_db$lncRNAs)
row.names(Bo_all_tpm_db) <- Bo_all_tpm_db$lncRNAs
tail(Bo_all_tpm_db)

# extraer IDs
Bo_all_1TPMormore_IDs <- Bo_all_tpm_db$lncRNAs
head(Bo_all_1TPMormore_IDs)

# Guardar
write.csv(Bo_all_1TPMormore_IDs, file = "./allspecies_data/Brassica_oleracea_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Bo_all_tpm_db, file = "./allspecies_data/Brassica_oleracea_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(Bo_all_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_oleracea_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Bo_all_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_oleracea_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)
```


```{r ALL-C_rubella}
library(dplyr)

Cr_all_tpm_db <- as.data.frame(Cr_all_tpm)
Cr_all_tpm_db <- Cr_all_tpm_db %>% filter(KO146_Cr_shootR1 >=1 & KO147_Cr_shootR2  >=1 & KO148_Cr_rootR1 >=1 & KO149_Cr_rootR2 >=1)
Cr_all_tpm_db$lncRNAs <-row.names(Cr_all_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Cr_all_tpm_db$lncRNAs <- gsub("MSTRG.","CrRegRNA_",Cr_all_tpm_db$lncRNAs)
row.names(Cr_all_tpm_db) <- Cr_all_tpm_db$lncRNAs
tail(Cr_all_tpm_db)

# extraer IDs
Cr_all_1TPMormore_IDs <-Cr_all_tpm_db$lncRNAs
head(Cr_all_1TPMormore_IDs)

# Guardar
write.csv(Cr_all_1TPMormore_IDs, file = "./allspecies_data/Capsella_rubella_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Cr_all_tpm_db, file = "./allspecies_data/Capsella_rubella_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(Cr_all_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Capsella_rubella_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Cr_all_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Capsella_rubella_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)
```


```{r lncRNAs-T_parvula}
library(dplyr)

Tp_all_tpm_db <- as.data.frame(Tp_all_tpm)
Tp_all_tpm_db <- Tp_all_tpm_db %>% filter(KO92_Tp_shootR1 >=1 & KO93_Tp_shootR2 >=1 & KO94_Tp_rootR1 >=1 & KO95_Tp_rootR2 >=1)
Tp_all_tpm_db$lncRNAs <-row.names(Tp_all_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Tp_all_tpm_db$lncRNAs <- gsub("MSTRG.","TpRegRNA_",Tp_all_tpm_db$lncRNAs)
row.names(Tp_all_tpm_db) <- Tp_all_tpm_db$lncRNAs
tail(Tp_all_tpm_db)

# extraer IDs
Tp_all_1TPMormore_IDs <-Tp_all_tpm_db$lncRNAs
head(Tp_all_1TPMormore_IDs)

# Guardar
write.csv(Tp_all_1TPMormore_IDs, file = "./allspecies_data/Thellungiella_parvula_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Tp_all_tpm_db, file = "./allspecies_data/Thellungiella_parvula_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(Tp_all_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Thellungiella_parvula_all_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Tp_all_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Thellungiella_parvula_all_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

#cp *_KallistoExpr_1tpm_RegRNAIDs.txt /mnt/s/Repositorio_scripts/Expressed_lncRNAs
# cp ./allspecies_data/*_KallistoExpr_1tpm_RegRNAIDs_expression.tsv /mnt/s/Repositorio_scripts/Intersect_lncRNAs/Expressed_lncRNAs_RegRNAIDs/
```


## Expresion diferencial (TODOS LOS TRANSCRITOS)

```{r ALL-A. thaliana}
At_ddsTxi_all                          <- DESeqDataSetFromTximport(At_txi_kallisto_all_genes,At_sampleTable,design=~dex) # Create a DESeq object from the tximport data
At_dds_all                              <- DESeq(At_ddsTxi_all) # run Differential expression analysis 
At_res_all                              <- results(At_dds_all)  # Save the results
mcols(At_res_all)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
At_all_de_gene_matrix_UP  <- subset(At_res_all, padj < 0.05 & log2FoldChange >= 0.5)
write.table(At_all_de_gene_matrix_UP,file ="./allspecies_data/At_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(At_all_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

At_all_de_gene_names_UP <- rownames(At_all_de_gene_matrix_UP)

# extraer down genes
At_all_de_gene_matrix_DOWN  <- subset(At_res_all, padj < 0.05 & log2FoldChange < -0.5)
write.table(At_all_de_gene_matrix_DOWN,file ="./allspecies_data/At_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(At_all_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

At_all_de_gene_names_DOWN <- rownames(At_all_de_gene_matrix_DOWN)

# Numero de genes expresados
length(At_all_de_gene_names_UP)
length(At_all_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
At_all_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","AtRegRNA_",At_all_de_gene_names_UP)
At_all_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","AtRegRNA_",At_all_de_gene_names_DOWN)

tail(At_all_de_gene_names_UP_RegRNAIDs)
tail(At_all_de_gene_names_DOWN_RegRNAIDs)
```


```{r ALL-B.rapa}
Br_ddsTxi_all                           <- DESeqDataSetFromTximport(Br_txi_kallisto_all_genes,Br_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Br_dds_all                              <- DESeq(Br_ddsTxi_all) # run Differential expression analysis 
Br_res_all                              <- results(Br_dds_all)  # Save the results
mcols(Br_res_all)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Br_all_de_gene_matrix_UP  <- subset(Br_res_all, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Br_all_de_gene_matrix_UP,file ="./allspecies_data/Br_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Br_all_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Br_all_de_gene_names_UP <- rownames(Br_all_de_gene_matrix_UP)

# extraer down genes
Br_all_de_gene_matrix_DOWN  <- subset(Br_res_all, padj < 0.05 & log2FoldChange < -0.5)
write.table(Br_all_de_gene_matrix_DOWN,file ="./allspecies_data/Br_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Br_all_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Br_all_de_gene_names_DOWN <- rownames(Br_all_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Br_all_de_gene_names_UP)
length(Br_all_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Br_all_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","BrRegRNA_",Br_all_de_gene_names_UP)
Br_all_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","BrRegRNA_",Br_all_de_gene_names_DOWN)

tail(Br_all_de_gene_names_UP_RegRNAIDs)
tail(Br_all_de_gene_names_DOWN_RegRNAIDs)

```


```{r ALL-B.oleracea}
Bo_ddsTxi_all                          <- DESeqDataSetFromTximport(Bo_txi_kallisto_all_genes,Bo_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Bo_dds_all                              <- DESeq(Bo_ddsTxi_all) # run Differential expression analysis 
Bo_res_all                              <- results(Bo_dds_all)  # Save the results
mcols(Bo_res_all)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Bo_all_de_gene_matrix_UP  <- subset(Bo_res_all, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Bo_all_de_gene_matrix_UP,file ="./allspecies_data/Bo_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Bo_all_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Bo_all_de_gene_names_UP <- rownames(Bo_all_de_gene_matrix_UP)

# extraer down genes
Bo_all_de_gene_matrix_DOWN  <- subset(Bo_res_all, padj < 0.05 & log2FoldChange < -0.5)
write.table(Bo_all_de_gene_matrix_DOWN,file ="./allspecies_data/Bo_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Bo_all_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Bo_all_de_gene_names_DOWN <- rownames(Bo_all_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Bo_all_de_gene_names_UP)
length(Bo_all_de_gene_names_DOWN)

head(Bo_all_de_gene_names_UP)
head(Bo_all_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Bo_all_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","BoRegRNA_",Bo_all_de_gene_names_UP)
Bo_all_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","BoRegRNA_",Bo_all_de_gene_names_DOWN)

tail(Bo_all_de_gene_names_UP_RegRNAIDs)
tail(Bo_all_de_gene_names_DOWN_RegRNAIDs)
```



```{r ALL-C.rubella}
Cr_ddsTxi_all                           <- DESeqDataSetFromTximport(Cr_txi_kallisto_all_genes,Cr_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Cr_dds_all                              <- DESeq(Cr_ddsTxi_all) # run Differential expression analysis 
Cr_res_all                              <- results(Cr_dds_all)  # Save the results
mcols(Cr_res_all)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Cr_all_de_gene_matrix_UP  <- subset(Cr_res_all, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Cr_all_de_gene_matrix_UP,file ="./allspecies_data/Cr_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Cr_all_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Cr_all_de_gene_names_UP <- rownames(Cr_all_de_gene_matrix_UP)

# extraer down genes
Cr_all_de_gene_matrix_DOWN  <- subset(Cr_res_all, padj < 0.05 & log2FoldChange < -0.5)
write.table(Cr_all_de_gene_matrix_DOWN,file ="./allspecies_data/Cr_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Cr_all_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Cr_all_de_gene_names_DOWN <- rownames(Cr_all_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Cr_all_de_gene_names_UP)
length(Cr_all_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Cr_all_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","CrRegRNA_",Cr_all_de_gene_names_UP)
Cr_all_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","CrRegRNA_",Cr_all_de_gene_names_DOWN)

tail(Cr_all_de_gene_names_UP_RegRNAIDs)
tail(Cr_all_de_gene_names_DOWN_RegRNAIDs)

```


```{r ALL-T.parvula}
Tp_ddsTxi_all                           <- DESeqDataSetFromTximport(Tp_txi_kallisto_all_genes,Tp_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Tp_dds_all                              <- DESeq(Tp_ddsTxi_all) # run Differential expression analysis 
Tp_res_all                              <- results(Tp_dds_all)  # Save the results
mcols(Tp_res_all)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Tp_all_de_gene_matrix_UP  <- subset(Tp_res_all, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Tp_all_de_gene_matrix_UP,file ="./allspecies_data/Tp_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Tp_all_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_all_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Tp_all_de_gene_names_UP <- rownames(Tp_all_de_gene_matrix_UP)

# extraer down genes
Tp_all_de_gene_matrix_DOWN  <- subset(Tp_res_all, padj < 0.05 & log2FoldChange < -0.5)
write.table(Tp_all_de_gene_matrix_DOWN,file ="./allspecies_data/Tp_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Tp_all_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_all_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Tp_all_de_gene_names_DOWN <- rownames(Tp_all_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Tp_all_de_gene_names_UP)
length(Tp_all_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Tp_all_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","TpRegRNA_",Tp_all_de_gene_names_UP)
Tp_all_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","TpRegRNA_",Tp_all_de_gene_names_DOWN)

tail(Tp_all_de_gene_names_UP_RegRNAIDs)
tail(Tp_all_de_gene_names_DOWN_RegRNAIDs)
```



## Cuentas normalizadas para graficas (rlog) (TODOS LOS TRANSCRITOS)

```{r ALL-A.thaliana}
At_all_normalized <- rlog(At_dds_all, blind=FALSE) # result rld, vst
At_all_normalized_db <- as.data.frame(assay(At_all_normalized))
head(At_all_normalized_db)

# guardar
write.table(At_all_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_all_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)

write.table(At_all_normalized_db,file ="allspecies_data/At_all_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```

```{r lncRNAs-B.rapa}
Br_all_normalized <- rlog(Br_dds_all, blind=FALSE) # result rld, vst
Br_all_normalized_db <- as.data.frame(assay(Br_all_normalized))
head(Br_all_normalized_db)

# guardar
write.table(Br_all_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_all_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```

```{r lncRNAs-B.oleracea}
Bo_all_normalized <- rlog(Bo_dds_all, blind=FALSE) # result rld, vst
Bo_all_normalized_db <- as.data.frame(assay(Bo_all_normalized))
head(Bo_all_normalized_db)

# guardar
write.table(Bo_all_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_all_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


```{r lncRNAs-C.rubella}
Cr_all_normalized <- rlog(Cr_dds_all, blind=FALSE) # result rld, vst
Cr_all_normalized_db <- as.data.frame(assay(Cr_all_normalized))
head(Cr_all_normalized_db)

# guardar
write.table(Cr_all_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_all_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


```{r lncRNAs-T.parvula}
Tp_all_normalized <- rlog(Tp_dds_all, blind=FALSE) # result rld, vst
Tp_all_normalized_db <- as.data.frame(assay(Tp_all_normalized))
head(Tp_all_normalized_db)

# guardar
write.table(Tp_all_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_all_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```

```{r}
At_pcaData <-plotPCA(At_all_normalized, intgroup = "dex", returnData=TRUE)
plotPCA(Br_all_normalized, intgroup = "dex")
plotPCA(Bo_all_normalized, intgroup = "dex")
plotPCA(Tp_all_normalized, intgroup = "dex")
plotPCA(Cr_all_normalized, intgroup = "dex")


library(ggplot2)
At_percentVar <- round(100 * attr(At_pcaData, "percentVar"))
ggplot(At_pcaData, aes(PC1, PC2, color=dex)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",At_percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",At_percentVar[2],"% variance")) + 
  coord_fixed()
```


## Expresion de SOLO PROTEINAS

```{r Cargar librerias}
library(tximport)
library(tidyverse)
library(readxl)
#install.packages("magrittr")
#install.packages("dplyr")
library(magrittr) # needs to be run every time you start R and want to use %>%
library(dplyr)    # alternatively, this also loads %>%
library(statmod)
#BiocManager::install("statmod")
library("DESeq2")
library("pheatmap")
library("RColorBrewer")
library("gplots")
#BiocManager::install("apeglm")
#BiocManager::install("DESeq2")
#BiocManager::install("pheatmap")
#BiocManager::install("rhdf5")
```


```{r SOLO PROTEINAS-A. thaliana}
# > A. thaliana
library(dplyr)

Araport11_pcoding_genes_IDs <- read.table("../STAR-StringTie/Araport11_pcoding_genes_without_lncRNAs_IDs", header = F)
# convertir a vector
Araport11_pcoding_genes_IDs_list <-as.character(rbind(Araport11_pcoding_genes_IDs$V1))
# extraer Genes codificantes
At_tx2gene_protein <- filter(At_tx2gene_all, GENEID %in% Araport11_pcoding_genes_IDs_list)

#tr2gn <- data.frame("gene_id"= tx2gene$geneID, "transcript_id" = tx2gene$transcriptID)
At_txi_kallisto_protein_genes <- tximport(At_files, type = "kallisto", tx2gene=At_tx2gene_protein)

colnames(At_txi_kallisto_protein_genes$counts) <-rownames(At_sampleTable) 

# Obtener tabla
At_protein_Abundance <- At_txi_kallisto_protein_genes$abundance
At_protein_tpm<- At_protein_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(At_protein_tpm, file = "./allspecies_data/Arabidopsis_thaliana_protein_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(At_protein_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Arabidopsis_thaliana_protein_KallistoExpr_tpm.csv", row.names = TRUE)

head(At_protein_tpm)
head(At_tx2gene_protein)
```

```{r SOLO PROTEINAS-B.rapa}
# > B.rapa
#Br_protein_tx2gene <- read.table("S:/Repositorio_scripts/Exons_transcript_info/Brassica_rapa_tr2gn_proteincoding", header=F) 
Br_protein_tx2gene <- read.table("./allspecies_data/Br_protein_tx2gene.txt", sep="\t", header=F)
colnames(Br_protein_tx2gene) <- c("transcriptID", "geneID")

#Br_tx2gene_protein <- filter(Br_tx2gene_all, geneID %in% Br_protein_tx2gene$geneID)

Br_txi_kallisto_protein_genes <- tximport(Br_files, type = "kallisto", tx2gene=Br_protein_tx2gene)
colnames(Br_txi_kallisto_protein_genes$counts) <-rownames(Br_sampleTable) 

# Obtener tabla
Br_protein_Abundance <- Br_txi_kallisto_protein_genes$abundance
Br_protein_tpm<- Br_protein_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Br_protein_tpm, file = "./allspecies_data/Brassica_rapa_protein_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Br_protein_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_rapa_protein_KallistoExpr_tpm.csv", row.names = TRUE)

head(Br_protein_tpm)
```


```{r SOLO PROTEINAS-B.oleracea}
# > B.oleracea
Bo_tx2gene_protein <- read.table("./allspecies_data/Bo_protein_tx2gene.txt", header=F) 
# tabla de todos los genes menos los lncRNAs, para incorporarlo acomodados por genes
Bo_tx2gene_protein <- Bo_tx2gene_protein[,c(2,1)]
colnames(Bo_tx2gene_protein) <- c("transcriptID", "geneID")

Bo_txi_kallisto_protein_genes <- tximport(Bo_files, type = "kallisto", tx2gene=Bo_tx2gene_protein)

colnames(Bo_txi_kallisto_protein_genes$counts) <-rownames(Bo_sampleTable) 

# Obtener tabla
Bo_protein_Abundance <- Bo_txi_kallisto_protein_genes$abundance
Bo_protein_tpm<- Bo_protein_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Bo_protein_tpm, file = "./allspecies_data/Brassica_oleracea_protein_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Bo_protein_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_oleracea_protein_KallistoExpr_tpm.csv", row.names = TRUE)

head(Bo_protein_tpm)

```

```{r SOLO PROTEINAS-C.rubella}
# > C. rubella
Cr_tx2gene_protein <- read.table("./allspecies_data/Cr_protein_tx2gene.txt", header=F) 
# tabla de todos los genes menos los lncRNAs, para incorporarlo acomodados por genes
Cr_tx2gene_protein <- Cr_tx2gene_protein[,c(2,1)]
colnames(Cr_tx2gene_protein) <- c("transcriptID", "geneID")

Cr_txi_kallisto_protein_genes <- tximport(Cr_files, type = "kallisto", tx2gene=Cr_tx2gene_protein)

colnames(Cr_txi_kallisto_protein_genes$counts) <-rownames(Cr_sampleTable) 

# Obtener tabla
Cr_protein_Abundance <- Cr_txi_kallisto_protein_genes$abundance
Cr_protein_tpm<- Cr_protein_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Cr_protein_tpm, file = "./allspecies_data/Capsella_rubella_protein_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Cr_protein_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Capsella_rubella_protein_KallistoExpr_tpm.csv", row.names = TRUE)

head(Cr_protein_tpm)
```


```{r SOLO PROTEINAS-Thellungiella_parvula}
# > Thellungiella_parvula
#Tp_protein_tx2gene <- read.table("S:/Repositorio_scripts/Exons_transcript_info/Thellungiella_parvula_tr2gn_proteincoding", header=F) 
Tp_protein_tx2gene <- read.table("./allspecies_data/Tp_protein_tx2gene.txt", sep="\t",header=F) 
colnames(Tp_protein_tx2gene) <- c("transcriptID", "geneID")

Tp_txi_kallisto_protein_genes <- tximport(Tp_files, type = "kallisto", tx2gene=Tp_protein_tx2gene)

colnames(Tp_txi_kallisto_protein_genes$counts) <-rownames(Tp_sampleTable) 

# Obtener tabla
Tp_protein_Abundance <- Tp_txi_kallisto_protein_genes$abundance
Tp_protein_tpm<- Tp_protein_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Tp_protein_tpm, file = "./allspecies_data/Thellungiella_parvula_protein_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Tp_protein_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Thellungiella_parvula_protein_KallistoExpr_tpm.csv", row.names = TRUE)

head(Tp_protein_tpm)
```

## Seleccionar por 1 TPM (SOLO PROTEINAS)

```{r SOLO PROTEINAS-A_thaliana}
library(dplyr)

At_protein_tpm_db <- as.data.frame(At_protein_tpm)
At_protein_tpm_db <- At_protein_tpm_db %>% filter(KO84_At_shootR1 >=1 & KO85_At_shootR2 >=1 & KO86_At_rootR1 >=1 & KO87_At_rootR2 >=1)
At_protein_tpm_db$lncRNAs <-row.names(At_protein_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
At_protein_tpm_db$lncRNAs <- gsub("MSTRG.","AtRegRNA_",At_protein_tpm_db$lncRNAs)
row.names(At_protein_tpm_db) <- At_protein_tpm_db$lncRNAs
tail(At_protein_tpm_db)

# extraer IDs
At_protein_1TPMormore_IDs <-At_protein_tpm_db$lncRNAs
head(At_protein_1TPMormore_IDs)
length(At_protein_1TPMormore_IDs)

# Guardar
write.csv(At_protein_1TPMormore_IDs, file = "./allspecies_data/Arabidopsis_thaliana_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(At_protein_tpm_db, file = "./allspecies_data/Arabidopsis_thaliana_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(At_protein_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Arabidopsis_thaliana_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(At_protein_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Arabidopsis_thaliana_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

#cp *KallistoExpr_1tpm_RegRNAIDs.txt /mnt/s/Repositorio_scripts/Intersect_lncRNAs/Expressed_lncRNAs_RegRNAIDs
```

```{r SOLO PROTEINAS-B_rapa}
library(dplyr)

Br_protein_tpm_db <- as.data.frame(Br_protein_tpm)
Br_protein_tpm_db <- Br_protein_tpm_db %>% filter(KO88_Br_shootR1 >=1 & KO89_Br_shootR2 >=1 & KO90_Br_rootR1 >=1 & KO91_Br_rootR2 >=1)
row.names(Br_protein_tpm_db)<- gsub("gene:","", row.names(Br_protein_tpm_db))

Br_protein_tpm_db$lncRNAs <-row.names(Br_protein_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Br_protein_tpm_db$lncRNAs <- gsub("MSTRG.","BrRegRNA_", Br_protein_tpm_db$lncRNAs)
row.names(Br_protein_tpm_db) <- Br_protein_tpm_db$lncRNAs
tail(Br_protein_tpm_db)

# extraer IDs
Br_protein_1TPMormore_IDs <-Br_protein_tpm_db$lncRNAs
head(Br_protein_1TPMormore_IDs)

# Guardar
write.csv(Br_protein_1TPMormore_IDs, file = "./allspecies_data/Brassica_rapa_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Br_protein_tpm_db, file = "./allspecies_data/Brassica_rapa_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(Br_all_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_rapa_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Br_protein_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_rapa_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)
```


```{r SOLO PROTEINAS-B_oleracea}
library(dplyr)

Bo_protein_tpm_db <- as.data.frame(Bo_protein_tpm)
Bo_protein_tpm_db <- Bo_protein_tpm_db %>% filter(KO150_Bo_shootR1 >=1 & KO151_Bo_shootR2 >=1 & KO152_Bo_rootR1 >=1 & KO153_Bo_rootR2 >=1)
Bo_protein_tpm_db$lncRNAs <-row.names(Bo_protein_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Bo_protein_tpm_db$lncRNAs <- gsub("MSTRG.","BoRegRNA_",Bo_protein_tpm_db$lncRNAs)
row.names(Bo_protein_tpm_db) <- Bo_protein_tpm_db$lncRNAs
tail(Bo_protein_tpm_db)

# extraer IDs
Bo_protein_1TPMormore_IDs <- Bo_protein_tpm_db$lncRNAs
head(Bo_protein_1TPMormore_IDs)

# Guardar
write.csv(Bo_protein_1TPMormore_IDs, file = "./allspecies_data/Brassica_oleracea_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Bo_protein_tpm_db, file = "./allspecies_data/Brassica_oleracea_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(Bo_protein_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_oleracea_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Bo_protein_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_oleracea_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)
```


```{r SOLO PROTEINAS-C_rubella}
library(dplyr)

Cr_protein_tpm_db <- as.data.frame(Cr_protein_tpm)
Cr_protein_tpm_db <- Cr_protein_tpm_db %>% filter(KO146_Cr_shootR1 >=1 & KO147_Cr_shootR2  >=1 & KO148_Cr_rootR1 >=1 & KO149_Cr_rootR2 >=1)
Cr_protein_tpm_db$lncRNAs <-row.names(Cr_protein_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Cr_protein_tpm_db$lncRNAs <- gsub("MSTRG.","CrRegRNA_",Cr_protein_tpm_db$lncRNAs)
row.names(Cr_protein_tpm_db) <- Cr_protein_tpm_db$lncRNAs
tail(Cr_protein_tpm_db)

# extraer IDs
Cr_protein_1TPMormore_IDs <-Cr_protein_tpm_db$lncRNAs
head(Cr_protein_1TPMormore_IDs)

# Guardar
write.csv(Cr_protein_1TPMormore_IDs, file = "./allspecies_data/Capsella_rubella_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Cr_protein_tpm_db, file = "./allspecies_data/Capsella_rubella_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(Cr_protein_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Capsella_rubella_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Cr_protein_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Capsella_rubella_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)
```


```{r SOLO PROTEINAS-T_parvula}
library(dplyr)

Tp_protein_tpm_db <- as.data.frame(Tp_protein_tpm)
Tp_protein_tpm_db <- Tp_protein_tpm_db %>% filter(KO92_Tp_shootR1 >=1 & KO93_Tp_shootR2 >=1 & KO94_Tp_rootR1 >=1 & KO95_Tp_rootR2 >=1)
Tp_protein_tpm_db$lncRNAs <-row.names(Tp_protein_tpm_db)

# Sustutir IDs 's/MSTRG./AtRegRNA_/g'
Tp_protein_tpm_db$lncRNAs <- gsub("MSTRG.","TpRegRNA_",Tp_protein_tpm_db$lncRNAs)
row.names(Tp_protein_tpm_db) <- Tp_protein_tpm_db$lncRNAs
tail(Tp_protein_tpm_db)

# extraer IDs
Tp_protein_1TPMormore_IDs <-Tp_protein_tpm_db$lncRNAs
head(Tp_protein_1TPMormore_IDs)

# Guardar
write.csv(Tp_protein_1TPMormore_IDs, file = "./allspecies_data/Thellungiella_parvula_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Tp_protein_tpm_db, file = "./allspecies_data/Thellungiella_parvula_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

write.csv(Tp_protein_1TPMormore_IDs, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Thellungiella_parvula_protein_KallistoExpr_1tpm_RegRNAIDs.txt", row.names = F, quote=FALSE)

write.table(Tp_protein_tpm_db, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Thellungiella_parvula_protein_KallistoExpr_1tpm_RegRNAIDs_expression.tsv", quote=FALSE, sep="\t", row.names = T,)

#cp *_KallistoExpr_1tpm_RegRNAIDs.txt /mnt/s/Repositorio_scripts/Expressed_lncRNAs
# cp ./allspecies_data/*_KallistoExpr_1tpm_RegRNAIDs_expression.tsv /mnt/s/Repositorio_scripts/Intersect_lncRNAs/Expressed_lncRNAs_RegRNAIDs/
```


## Expresion diferencial (SOLO PROTEINAS)

```{r SOLO PROTEINAS-A. thaliana}
At_ddsTxi_protein                          <- DESeqDataSetFromTximport(At_txi_kallisto_protein_genes,At_sampleTable,design=~dex) # Create a DESeq object from the tximport data
At_dds_protein                              <- DESeq(At_ddsTxi_protein) # run Differential expression analysis 
At_res_protein                              <- results(At_dds_protein)  # Save the results
mcols(At_res_protein)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
At_protein_de_gene_matrix_UP  <- subset(At_res_protein, padj < 0.05 & log2FoldChange >= 0.5)
write.table(At_protein_de_gene_matrix_UP,file ="./allspecies_data/At_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(At_protein_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

At_protein_de_gene_names_UP <- rownames(At_protein_de_gene_matrix_UP)
head(At_protein_de_gene_matrix_UP)

# extraer down genes
At_protein_de_gene_matrix_DOWN  <- subset(At_res_protein, padj < 0.05 & log2FoldChange < -0.5)
write.table(At_protein_de_gene_matrix_DOWN,file ="./allspecies_data/At_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(At_protein_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

At_protein_de_gene_names_DOWN <- rownames(At_protein_de_gene_matrix_DOWN)

# Numero de genes expresados
length(At_protein_de_gene_names_UP)
length(At_protein_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
At_protein_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","AtRegRNA_",At_protein_de_gene_names_UP)
At_protein_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","AtRegRNA_",At_protein_de_gene_names_DOWN)

tail(At_protein_de_gene_names_UP_RegRNAIDs)
tail(At_protein_de_gene_names_DOWN_RegRNAIDs)
```


```{r SOLO PROTEINAS-B.rapa}
Br_ddsTxi_protein                           <- DESeqDataSetFromTximport(Br_txi_kallisto_protein_genes,Br_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Br_dds_protein                              <- DESeq(Br_ddsTxi_protein) # run Differential expression analysis 
Br_res_protein                              <- results(Br_dds_protein)  # Save the results
mcols(Br_res_protein)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Br_protein_de_gene_matrix_UP  <- subset(Br_res_protein, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Br_protein_de_gene_matrix_UP,file ="./allspecies_data/Br_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Br_protein_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

#Br_protein_de_gene_names_UP <- rownames(Br_protein_de_gene_matrix_UP)
Br_protein_de_gene_names_UP <- gsub("gene:","", rownames(Br_protein_de_gene_matrix_UP))

# extraer down genes
Br_protein_de_gene_matrix_DOWN  <- subset(Br_res_protein, padj < 0.05 & log2FoldChange < -0.5)
write.table(Br_protein_de_gene_matrix_DOWN,file ="./allspecies_data/Br_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Br_protein_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

# Br_protein_de_gene_names_DOWN <- rownames(Br_protein_de_gene_matrix_DOWN)
Br_protein_de_gene_names_DOWN <- gsub("gene:","", rownames(Br_protein_de_gene_matrix_DOWN))

# Numero de genes expresados
length(Br_protein_de_gene_names_UP)
length(Br_protein_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Br_protein_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","BrRegRNA_",Br_protein_de_gene_names_UP)
Br_protein_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","BrRegRNA_",Br_protein_de_gene_names_DOWN)

tail(Br_protein_de_gene_names_UP_RegRNAIDs)
tail(Br_protein_de_gene_names_DOWN_RegRNAIDs)

```


```{r SOLO PROTEINAS-B.oleracea}
Bo_ddsTxi_protein                          <- DESeqDataSetFromTximport(Bo_txi_kallisto_protein_genes,Bo_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Bo_dds_protein                              <- DESeq(Bo_ddsTxi_protein) # run Differential expression analysis 
Bo_res_protein                              <- results(Bo_dds_protein)  # Save the results
mcols(Bo_res_protein)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Bo_protein_de_gene_matrix_UP  <- subset(Bo_res_protein, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Bo_protein_de_gene_matrix_UP,file ="./allspecies_data/Bo_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Bo_protein_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Bo_protein_de_gene_names_UP <- rownames(Bo_protein_de_gene_matrix_UP)

# extraer down genes
Bo_protein_de_gene_matrix_DOWN  <- subset(Bo_res_all, padj < 0.05 & log2FoldChange < -0.5)
write.table(Bo_protein_de_gene_matrix_DOWN,file ="./allspecies_data/Bo_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Bo_protein_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Bo_protein_de_gene_names_DOWN <- rownames(Bo_protein_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Bo_protein_de_gene_names_UP)
length(Bo_protein_de_gene_names_DOWN)

head(Bo_protein_de_gene_names_UP)
head(Bo_protein_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Bo_protein_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","BoRegRNA_",Bo_protein_de_gene_names_UP)
Bo_protein_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","BoRegRNA_",Bo_protein_de_gene_names_DOWN)

tail(Bo_protein_de_gene_names_UP_RegRNAIDs)
tail(Bo_protein_de_gene_names_DOWN_RegRNAIDs)

```


```{r SOLO PROTEINAS-C.rubella}
Cr_ddsTxi_protein                          <- DESeqDataSetFromTximport(Cr_txi_kallisto_protein_genes,Cr_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Cr_dds_protein                              <- DESeq(Cr_ddsTxi_protein) # run Differential expression analysis 
Cr_res_protein                              <- results(Cr_dds_protein)  # Save the results
mcols(Cr_res_protein)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Cr_protein_de_gene_matrix_UP  <- subset(Cr_res_protein, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Cr_protein_de_gene_matrix_UP,file ="./allspecies_data/Cr_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Cr_protein_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Cr_protein_de_gene_names_UP <- rownames(Cr_protein_de_gene_matrix_UP)

# extraer down genes
Cr_protein_de_gene_matrix_DOWN  <- subset(Cr_res_protein, padj < 0.05 & log2FoldChange < -0.5)
write.table(Cr_protein_de_gene_matrix_DOWN,file ="./allspecies_data/Cr_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Cr_protein_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Cr_protein_de_gene_names_DOWN <- rownames(Cr_protein_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Cr_protein_de_gene_names_UP)
length(Cr_protein_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Cr_protein_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","CrRegRNA_",Cr_protein_de_gene_names_UP)
Cr_protein_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","CrRegRNA_",Cr_protein_de_gene_names_DOWN)

tail(Cr_protein_de_gene_names_UP_RegRNAIDs)
tail(Cr_protein_de_gene_names_DOWN_RegRNAIDs)

```


```{r SOLO PROTEINAS-T.parvula}
Tp_ddsTxi_protein                           <- DESeqDataSetFromTximport(Tp_txi_kallisto_protein_genes,Tp_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Tp_dds_protein                              <- DESeq(Tp_ddsTxi_protein) # run Differential expression analysis 
Tp_res_protein                              <- results(Tp_dds_protein)  # Save the results
mcols(Tp_res_protein)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Tp_protein_de_gene_matrix_UP  <- subset(Tp_res_protein, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Tp_protein_de_gene_matrix_UP,file ="./allspecies_data/Tp_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Tp_protein_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_protein_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Tp_protein_de_gene_names_UP <- rownames(Tp_protein_de_gene_matrix_UP)

# extraer down genes
Tp_protein_de_gene_matrix_DOWN  <- subset(Tp_res_protein, padj < 0.05 & log2FoldChange < -0.5)
write.table(Tp_protein_de_gene_matrix_DOWN,file ="./allspecies_data/Tp_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Tp_protein_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_protein_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Tp_protein_de_gene_names_DOWN <- rownames(Tp_protein_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Tp_protein_de_gene_names_UP)
length(Tp_protein_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Tp_protein_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","TpRegRNA_",Tp_protein_de_gene_names_UP)
Tp_protein_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","TpRegRNA_",Tp_protein_de_gene_names_DOWN)

tail(Tp_protein_de_gene_names_UP_RegRNAIDs)
tail(Tp_protein_de_gene_names_DOWN_RegRNAIDs)
```



## Cuentas normalizadas para graficas (rlog) (SOLO PROTEINAS)

```{r SOLO PROTEINAS-A.thaliana}
At_protein_normalized <- rlog(At_dds_protein, blind=FALSE) # result rld, vst
At_protein_normalized_db <- as.data.frame(assay(At_protein_normalized))
head(At_protein_normalized_db)

# guardar
write.table(At_protein_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_protein_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
write.table(At_protein_normalized_db,file ="allspecies_data/At_protein_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)

```

```{r SOLO PROTEINAS-B.rapa}
Br_protein_normalized <- rlog(Br_dds_protein, blind=FALSE) # result rld, vst
Br_protein_normalized_db <- as.data.frame(assay(Br_protein_normalized))

rownames(Br_protein_normalized_db) <- gsub("gene:","", rownames(Br_protein_normalized_db))

head(Br_protein_normalized_db)

# guardar
write.table(Br_protein_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_protein_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```

```{r SOLO PROTEINAS-B.oleracea}
Bo_protein_normalized <- rlog(Bo_dds_protein, blind=FALSE) # result rld, vst
Bo_protein_normalized_db <- as.data.frame(assay(Bo_protein_normalized))
head(Bo_protein_normalized_db)

# guardar
write.table(Bo_protein_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_protein_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


```{r SOLO PROTEINAS-C.rubella}
Cr_protein_normalized <- rlog(Cr_dds_protein, blind=FALSE) # result rld, vst
Cr_protein_normalized_db <- as.data.frame(assay(Cr_protein_normalized))
head(Cr_protein_normalized_db)

# guardar
write.table(Cr_protein_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_protein_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


```{r SOLO PROTEINAS-T.parvula}
Tp_protein_normalized <- rlog(Tp_dds_protein, blind=FALSE) # result rld, vst
Tp_protein_normalized_db <- as.data.frame(assay(Tp_protein_normalized))
head(Tp_protein_normalized_db)

# guardar
write.table(Tp_protein_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_protein_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


## Graficas


```{r}
At_protein_tpm_db_all <- as.data.frame(At_protein_tpm)
At_lncRNA_tpm_db_all <- as.data.frame(At_tpm)

head(At_lncRNA_tpm_db_all)
head(At_protein_tpm_db_all)

At_vsd_lncRNAs_cv <- as.data.frame(assay(varianceStabilizingTransformation(At_ddsTxi_lncRNAs,blind=T)),check.names=F)
boxplot(At_vsd_lncRNAs_cv,ylab=expression('Log'[2]~'Read counts'),las=2,main="VST")


head(At_vsd_lncRNAs_cv)
```



```{r log2MaxTPM- A.thaliana}
At_protein_tpm_db_all <- as.data.frame(At_protein_tpm)
At_lncRNA_tpm_db_all <- as.data.frame(At_tpm)

# seleccionar maxima expresion en TPM de lncRNas y proteinas (hezroni y Palos lo hacen)
At_lncRNA_MAxtpm <- apply(At_lncRNA_tpm_db_all,1,max)
At_protein_MAxtpm <- apply(At_protein_tpm_db_all,1,max)

# Dataframe proteinas
At_protein_MAxtpm_db <- as.data.frame(At_protein_MAxtpm)
At_protein_MAxtpm_db$type <- "protein"
At_protein_MAxtpm_db$geneID <-row.names(At_protein_MAxtpm_db)
colnames(At_protein_MAxtpm_db) <- c("MaxTPM", "Type", "geneID")
At_protein_MAxtpm_db$specie <- "A. thaliana"

#head(At_protein_MAxtpm_db)

# Dataframe lncRNAs
At_lncRNA_MAxtpm_db <- as.data.frame(At_lncRNA_MAxtpm)
At_lncRNA_MAxtpm_db$type <- "lncRNA"
At_lncRNA_MAxtpm_db$geneID <-row.names(At_lncRNA_MAxtpm_db)
colnames(At_lncRNA_MAxtpm_db) <- c("MaxTPM", "Type", "geneID")
At_lncRNA_MAxtpm_db$specie <- "A. thaliana"

#head(At_lncRNA_MAxtpm_db)

# unir
At_expression_protein_lncRNA_MAxtpm_db <- rbind(At_protein_MAxtpm_db,At_lncRNA_MAxtpm_db)
At_expression_protein_lncRNA_MAxtpm_db$log2MaxTPM <- log2(At_expression_protein_lncRNA_MAxtpm_db$MaxTPM)

# Grafica
library(ggplot2)

ggplot(At_expression_protein_lncRNA_MAxtpm_db, aes(x= Type, y=log2MaxTPM)) +
  geom_boxplot() 

#  scale_y_continuous(breaks=c(0,500,1000,2000,3000,4000,5000), limits = c(0,5000)) # delimitar ylim y divider el eje
```






## Expresion de todos LOS GENES ANOTADOS (solo Comparacion AtlincRNA-SpTranscript)

```{r Cargar librerias}
library(tximport)
library(tidyverse)
library(readxl)
#install.packages("magrittr")
#install.packages("dplyr")
library(magrittr) # needs to be run every time you start R and want to use %>%
library(dplyr)    # alternatively, this also loads %>%
library(statmod)
#BiocManager::install("statmod")
library("DESeq2")
library("pheatmap")
library("RColorBrewer")
library("gplots")
#BiocManager::install("apeglm")
#BiocManager::install("DESeq2")
#BiocManager::install("pheatmap")
#BiocManager::install("rhdf5")
```


```{r ALL-B.rapa}
# > B.rapa
# TOdos los transcritos que estan conservados por sintenia y secuencia (CSS) con relacion a los AtlincRNAs
Br_tx2gene_CSS_all_ed <- Br_tx2gene_CSS_all
Br_tx2gene_CSS_all_ed$transcript_id <-gsub("BrRegRNA_","MSTRG.", Br_tx2gene_CSS_all$Brall_TransID)
Br_tx2gene_CSS_all_ed$gene_id <- gsub("BrRegRNA_","MSTRG.", Br_tx2gene_CSS_all$Brall_geneID)
Br_tx2gene_CSS_all_ed <- Br_tx2gene_CSS_all_ed[,c(3,4)]

Br_txi_kallisto_all_CSS_genes <- tximport(Br_files, type = "kallisto", tx2gene=Br_tx2gene_CSS_all_ed)
colnames(Br_txi_kallisto_all_CSS_genes$counts) <-rownames(Br_sampleTable) 

# Obtener tabla
Br_all_CSS_Abundance <- Br_txi_kallisto_all_CSS_genes$abundance
Br_all_CSS_tpm<- Br_all_CSS_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Br_all_CSS_Abundance, file = "./allspecies_data/Brassica_rapa_all_CSS_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Br_all_CSS_Abundance, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_rapa_all_CSS_KallistoExpr_tpm.csv", row.names = TRUE)

head(Br_all_CSS_tpm)
```


```{r ALL-B.oleracea}
# > B.oleracea
Bo_tx2gene_CSS_all_ed <- Bo_tx2gene_CSS_all
Bo_tx2gene_CSS_all_ed$transcript_id <-gsub("BoRegRNA_","MSTRG.", Bo_tx2gene_CSS_all$Boall_TransID)
Bo_tx2gene_CSS_all_ed$gene_id <- gsub("BoRegRNA_","MSTRG.", Bo_tx2gene_CSS_all$Boall_geneID)
Bo_tx2gene_CSS_all_ed <- Bo_tx2gene_CSS_all_ed[,c(3,4)]

Bo_txi_kallisto_all_CSS_genes <- tximport(Bo_files, type = "kallisto", tx2gene=Bo_tx2gene_CSS_all_ed, txIn = TRUE, txOut = FALSE)
colnames(Bo_txi_kallisto_all_CSS_genes$counts) <-rownames(Bo_sampleTable) 

# Obtener tabla
Bo_all_CSS_Abundance <- Bo_txi_kallisto_all_CSS_genes$abundance
Bo_all_CSS_tpm<- Bo_all_CSS_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Bo_all_CSS_tpm, file = "./allspecies_data/Brassica_oleracea_all_CSS_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Bo_all_CSS_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Brassica_oleracea_all_CSS_KallistoExpr_tpm.csv", row.names = TRUE)

head(Bo_all_CSS_tpm)

```

```{r ALL-C.rubella}
# > C. rubella
Cr_tx2gene_CSS_all_ed <- Cr_tx2gene_CSS_all
Cr_tx2gene_CSS_all_ed$transcript_id <-gsub("CrRegRNA_","MSTRG.", Cr_tx2gene_CSS_all$Crall_TransID)
Cr_tx2gene_CSS_all_ed$gene_id <- gsub("CrRegRNA_","MSTRG.", Cr_tx2gene_CSS_all$Crall_geneID)
Cr_tx2gene_CSS_all_ed <- Cr_tx2gene_CSS_all_ed[,c(3,4)]

Cr_txi_kallisto_all_CSS_genes <- tximport(Cr_files, type = "kallisto", tx2gene=Cr_tx2gene_CSS_all_ed)

colnames(Cr_txi_kallisto_all_CSS_genes$counts) <-rownames(Cr_sampleTable) 

# Obtener tabla
Cr_all_CSS_Abundance <- Cr_txi_kallisto_all_CSS_genes$abundance
Cr_all_CSS_tpm<- Cr_all_CSS_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Cr_all_CSS_tpm, file = "./allspecies_data/Capsella_rubella_all_CSS_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Cr_all_CSS_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Capsella_rubella_all_CSS_KallistoExpr_tpm.csv", row.names = TRUE)

head(Cr_all_CSS_tpm)
```


```{r ALL-Thellungiella_parvula}
# > Thellungiella_parvula
Tp_tx2gene_CSS_all_ed <- Tp_tx2gene_CSS_all
Tp_tx2gene_CSS_all_ed$transcript_id <-gsub("TpRegRNA_","MSTRG.", Tp_tx2gene_CSS_all$Tpall_TransID)
Tp_tx2gene_CSS_all_ed$gene_id <- gsub("TpRegRNA_","MSTRG.", Tp_tx2gene_CSS_all$Tpall_geneID)
Tp_tx2gene_CSS_all_ed <- Tp_tx2gene_CSS_all_ed[,c(3,4)]

Tp_txi_kallisto_all_CSS_genes <- tximport(Tp_files, type = "kallisto", tx2gene=Tp_tx2gene_CSS_all_ed)

colnames(Tp_txi_kallisto_all_CSS_genes$counts) <-rownames(Tp_sampleTable) 

# Obtener tabla
Tp_all_CSS_Abundance <- Tp_txi_kallisto_all_CSS_genes$abundance
Tp_all_CSS_tpm<- Tp_all_CSS_Abundance #usando lo que nos da automatico tximport
#Cuentas <- txi$counts

# Guardar tabla
# write.csv(Cuentas, file = "KallistoExpr.csv", row.names = TRUE)
write.csv(Tp_all_CSS_tpm, file = "./allspecies_data/Thellungiella_parvula_all_CSS_KallistoExpr_tpm.csv", row.names = TRUE)
write.csv(Tp_all_CSS_tpm, file = "S:/Repositorio_scripts/Expressed_lncRNAs/Thellungiella_parvula_all_CSS_KallistoExpr_tpm.csv", row.names = TRUE)

head(Tp_all_CSS_tpm)
```


## Expresion diferencial (TODOS LOS TRANSCRITOS_CSS)


```{r ALL-B.rapa}
Br_ddsTxi_all_CSS                           <- DESeqDataSetFromTximport(Br_txi_kallisto_all_CSS_genes,Br_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Br_dds_all_CSS                              <- DESeq(Br_ddsTxi_all_CSS) # run Differential expression analysis 
Br_res_all_CSS                              <- results(Br_dds_all_CSS)  # Save the results
mcols(Br_res_all_CSS)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Br_all_CSS_de_gene_matrix_UP  <- subset(Br_res_all_CSS, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Br_all_CSS_de_gene_matrix_UP,file ="./allspecies_data/Br_all_CSS_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Br_all_CSS_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_all_CSS_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Br_all_CSS_de_gene_names_UP <- rownames(Br_all_CSS_de_gene_matrix_UP)

# extraer down genes
Br_all_CSS_de_gene_matrix_DOWN  <- subset(Br_res_all_CSS, padj < 0.05 & log2FoldChange < -0.5)
write.table(Br_all_CSS_de_gene_matrix_DOWN,file ="./allspecies_data/Br_all_CSS_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Br_all_CSS_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_all_CSS_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Br_all_CSS_de_gene_names_DOWN <- rownames(Br_all_CSS_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Br_all_CSS_de_gene_names_UP)
length(Br_all_CSS_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Br_all_CSS_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","BrRegRNA_",Br_all_CSS_de_gene_names_UP)
Br_all_CSS_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","BrRegRNA_",Br_all_CSS_de_gene_names_DOWN)

tail(Br_all_CSS_de_gene_names_UP_RegRNAIDs)
tail(Br_all_CSS_de_gene_names_DOWN_RegRNAIDs)

```


```{r ALL-B.oleracea}
Bo_ddsTxi_all_CSS                          <- DESeqDataSetFromTximport(Bo_txi_kallisto_all_CSS_genes,Bo_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Bo_dds_all_CSS                              <- DESeq(Bo_ddsTxi_all_CSS) # run Differential expression analysis 
Bo_res_all_CSS                              <- results(Bo_dds_all_CSS)  # Save the results
mcols(Bo_res_all_CSS)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Bo_all_CSS_de_gene_matrix_UP  <- subset(Bo_res_all_CSS, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Bo_all_CSS_de_gene_matrix_UP,file ="./allspecies_data/Bo_all_CSS_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Bo_all_CSS_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_all_CSS_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Bo_all_CSS_de_gene_names_UP <- rownames(Bo_all_CSS_de_gene_matrix_UP)

# extraer down genes
Bo_all_CSS_de_gene_matrix_DOWN  <- subset(Bo_res_all, padj < 0.05 & log2FoldChange < -0.5)
write.table(Bo_all_CSS_de_gene_matrix_DOWN,file ="./allspecies_data/Bo_all_CSS_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Bo_all_CSS_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_all_CSS_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Bo_all_CSS_de_gene_names_DOWN <- rownames(Bo_all_CSS_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Bo_all_CSS_de_gene_names_UP)
length(Bo_all_CSS_de_gene_names_DOWN)

head(Bo_all_CSS_de_gene_names_UP)
head(Bo_all_CSS_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Bo_all_CSS_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","BoRegRNA_",Bo_all_CSS_de_gene_names_UP)
Bo_all_CSS_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","BoRegRNA_",Bo_all_CSS_de_gene_names_DOWN)

tail(Bo_all_CSS_de_gene_names_UP_RegRNAIDs)
tail(Bo_all_CSS_de_gene_names_DOWN_RegRNAIDs)

```


```{r ALL-C.rubella}
Cr_ddsTxi_all_CSS                           <- DESeqDataSetFromTximport(Cr_txi_kallisto_all_CSS_genes,Cr_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Cr_dds_all_CSS                              <- DESeq(Cr_ddsTxi_all_CSS) # run Differential expression analysis 
Cr_res_all_CSS                              <- results(Cr_dds_all_CSS)  # Save the results
mcols(Cr_res_all_CSS)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Cr_all_CSS_de_gene_matrix_UP  <- subset(Cr_res_all_CSS, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Cr_all_CSS_de_gene_matrix_UP,file ="./allspecies_data/Cr_all_CSS_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Cr_all_CSS_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_all_CSS_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Cr_all_CSS_de_gene_names_UP <- rownames(Cr_all_CSS_de_gene_matrix_UP)

# extraer down genes
Cr_all_CSS_de_gene_matrix_DOWN  <- subset(Cr_res_all_CSS, padj < 0.05 & log2FoldChange < -0.5)
write.table(Cr_all_CSS_de_gene_matrix_DOWN,file ="./allspecies_data/Cr_all_CSS_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Cr_all_CSS_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_all_CSS_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Cr_all_CSS_de_gene_names_DOWN <- rownames(Cr_all_CSS_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Cr_all_CSS_de_gene_names_UP)
length(Cr_all_CSS_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Cr_all_CSS_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","CrRegRNA_",Cr_all_CSS_de_gene_names_UP)
Cr_all_CSS_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","CrRegRNA_",Cr_all_CSS_de_gene_names_DOWN)

tail(Cr_all_CSS_de_gene_names_UP_RegRNAIDs)
tail(Cr_all_CSS_de_gene_names_DOWN_RegRNAIDs)

```


```{r ALL-T.parvula}
Tp_ddsTxi_all_CSS                           <- DESeqDataSetFromTximport(Tp_txi_kallisto_all_CSS_genes,Tp_sampleTable,design=~dex) # Create a DESeq object from the tximport data
Tp_dds_all_CSS                              <- DESeq(Tp_ddsTxi_all_CSS) # run Differential expression analysis 
Tp_res_all_CSS                              <- results(Tp_dds_all_CSS)  # Save the results
mcols(Tp_res_all_CSS)$description # contraste shoot vs root
#https://github.com/COMBINE-lab/salmon/issues/581

# extraer UP
Tp_all_CSS_de_gene_matrix_UP  <- subset(Tp_res_all_CSS, padj < 0.05 & log2FoldChange >= 0.5)
write.table(Tp_all_CSS_de_gene_matrix_UP,file ="./allspecies_data/Tp_all_CSS_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")
write.table(Tp_all_CSS_de_gene_matrix_UP,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_all_CSS_DEG_kallisto_shoot.tsv", quote=FALSE, sep="\t")

Tp_all_CSS_de_gene_names_UP <- rownames(Tp_all_CSS_de_gene_matrix_UP)

# extraer down genes
Tp_all_CSS_de_gene_matrix_DOWN  <- subset(Tp_res_all_CSS, padj < 0.05 & log2FoldChange < -0.5)
write.table(Tp_all_CSS_de_gene_matrix_DOWN,file ="./allspecies_data/Tp_all_CSS_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")
write.table(Tp_all_CSS_de_gene_matrix_DOWN,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_all_CSS_DEG_kallisto_root.tsv", quote=FALSE, sep="\t")

Tp_all_CSS_de_gene_names_DOWN <- rownames(Tp_all_CSS_de_gene_matrix_DOWN)

# Numero de genes expresados
length(Tp_all_CSS_de_gene_names_UP)
length(Tp_all_CSS_de_gene_names_DOWN)

# Cambiar a AtRegRNAIDs
Tp_all_CSS_de_gene_names_UP_RegRNAIDs <-gsub("MSTRG.","TpRegRNA_",Tp_all_CSS_de_gene_names_UP)
Tp_all_CSS_de_gene_names_DOWN_RegRNAIDs<-gsub("MSTRG.","TpRegRNA_",Tp_all_CSS_de_gene_names_DOWN)

tail(Tp_all_CSS_de_gene_names_UP_RegRNAIDs)
tail(Tp_all_CSS_de_gene_names_DOWN_RegRNAIDs)
```



## Cuentas normalizadas para graficas (rlog) (TODOS LOS TRANSCRITOS_CSS)

```{r lncRNAs-B.rapa}
Br_all_CSS_normalized <- rlog(Br_dds_all_CSS, blind=FALSE) # result rld, vst
Br_all_CSS_normalized_db <- as.data.frame(assay(Br_all_CSS_normalized))
head(Br_all_CSS_normalized_db)

# guardar
write.table(Br_all_CSS_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_all_CSS_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```

```{r lncRNAs-B.oleracea}
Bo_all_CSS_normalized <- rlog(Bo_dds_all_CSS, blind=FALSE) # result rld, vst
Bo_all_CSS_normalized_db <- as.data.frame(assay(Bo_all_CSS_normalized))
head(Bo_all_CSS_normalized_db)

# guardar
write.table(Bo_all_CSS_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_all_CSS_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


```{r lncRNAs-C.rubella}
Cr_all_CSS_normalized <- rlog(Cr_dds_all_CSS, blind=FALSE) # result rld, vst
Cr_all_CSS_normalized_db <- as.data.frame(assay(Cr_all_CSS_normalized))
head(Cr_all_CSS_normalized_db)

# guardar
write.table(Cr_all_CSS_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_all_CSS_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


```{r lncRNAs-T.parvula}
Tp_all_CSS_normalized <- rlog(Tp_dds_all_CSS, blind=FALSE) # result rld, vst
Tp_all_CSS_normalized_db <- as.data.frame(assay(Tp_all_CSS_normalized))
head(Tp_all_CSS_normalized_db)

# guardar
write.table(Tp_all_CSS_normalized_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_all_CSS_expressed_RLOGnormalized.tsv", quote=FALSE, sep="\t", row.names = T)
```


## Figura 1B- Grafica de la expresion de los genes (Todos los genes) (5 especies) (MaxTPM)

Las tablas At_protein_tpm y At_tpm de cada especie se encuentran en DESeq2_allspecies.Rmd

```{r log2MaxTPM- Allspecies}
library(dplyr)

# > A.thaliana
At_protein_tpm_db_all <- as.data.frame(At_protein_tpm)
At_lncRNA_tpm_db_all <- as.data.frame(At_tpm)

# seleccionar maxima expresion en TPM de lncRNas y proteinas (hezroni y Palos lo hacen)
At_lncRNA_MAxtpm <- apply(At_lncRNA_tpm_db_all,1,max)
At_protein_MAxtpm <- apply(At_protein_tpm_db_all,1,max)

# Dataframe proteinas
At_protein_MAxtpm_db <- as.data.frame(At_protein_MAxtpm)
At_protein_MAxtpm_db$category <- "protein-coding"
At_protein_MAxtpm_db$geneID <-row.names(At_protein_MAxtpm_db)
colnames(At_protein_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
At_protein_MAxtpm_db$specie <- "A.thaliana"

#head(At_protein_MAxtpm_db)

# Dataframe lncRNAs
At_lncRNA_MAxtpm_db <- as.data.frame(At_lncRNA_MAxtpm)
At_lncRNA_MAxtpm_db$category <- "lncRNA"
At_lncRNA_MAxtpm_db$geneID <-row.names(At_lncRNA_MAxtpm_db)
colnames(At_lncRNA_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
At_lncRNA_MAxtpm_db$specie <- "A.thaliana"

# seleccionar solo lincRNAs (solo expreados para estos 4 transcriptomas)
At_THESIS_lncRNAs_intergenicos_GENES_IDs <- At_THESIS_lncRNAs_annotated_bed %>% filter(BioType == "lincRNA") %>% select(GeneID)
At_lincRNA_MAxtpm_db <- At_lncRNA_MAxtpm_db
At_lincRNA_MAxtpm_db$geneID <- gsub("MSTRG.", "AtRegRNA_", At_lincRNA_MAxtpm_db$geneID)
At_lincRNA_MAxtpm_db <- At_lincRNA_MAxtpm_db %>% filter(geneID %in% At_THESIS_lncRNAs_intergenicos_GENES_IDs$GeneID)
At_lincRNA_MAxtpm_db$category <- "lincRNA"

# > B. rapa
Br_protein_tpm_db_all <- as.data.frame(Br_protein_tpm)
Br_lncRNA_tpm_db_all <- as.data.frame(Br_tpm)

# seleccionar maxima expresion en TPM de lncRNas y proteinas (hezroni y Palos lo hacen)
Br_lncRNA_MAxtpm <- apply(Br_lncRNA_tpm_db_all,1,max)
Br_protein_MAxtpm <- apply(Br_protein_tpm_db_all,1,max)

# Dataframe proteinas
Br_protein_MAxtpm_db <- as.data.frame(Br_protein_MAxtpm)
Br_protein_MAxtpm_db$category <- "protein-coding"
Br_protein_MAxtpm_db$geneID <-row.names(Br_protein_MAxtpm_db)
colnames(Br_protein_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
Br_protein_MAxtpm_db$specie <- "B.rapa"

#head(At_protein_MAxtpm_db)

# Dataframe lncRNAs
Br_lncRNA_MAxtpm_db <- as.data.frame(Br_lncRNA_MAxtpm)
Br_lncRNA_MAxtpm_db$category <- "lncRNA"
Br_lncRNA_MAxtpm_db$geneID <-row.names(Br_lncRNA_MAxtpm_db)
colnames(Br_lncRNA_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
Br_lncRNA_MAxtpm_db$specie <- "B.rapa"

# seleccionar solo lincRNAs
Br_lincRNA_MAxtpm_db <- Br_lncRNA_MAxtpm_db %>% filter(geneID %in% Br_lncRNAs_intergenicos_GENES_IDs$V1)
Br_lincRNA_MAxtpm_db$category <- "lincRNA"

# > B.oleracea
Bo_protein_tpm_db_all <- as.data.frame(Bo_protein_tpm)
Bo_lncRNA_tpm_db_all <- as.data.frame(Bo_tpm)

# seleccionar maxima expresion en TPM de lncRNas y proteinas (hezroni y Palos lo hacen)
Bo_lncRNA_MAxtpm <- apply(Bo_lncRNA_tpm_db_all,1,max)
Bo_protein_MAxtpm <- apply(Bo_protein_tpm_db_all,1,max)

# Dataframe proteinas
Bo_protein_MAxtpm_db <- as.data.frame(Bo_protein_MAxtpm)
Bo_protein_MAxtpm_db$category <- "protein-coding"
Bo_protein_MAxtpm_db$geneID <-row.names(Bo_protein_MAxtpm_db)
colnames(Bo_protein_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
Bo_protein_MAxtpm_db$specie <- "B.oleracea"

#head(At_protein_MAxtpm_db)

# Dataframe lncRNAs
Bo_lncRNA_MAxtpm_db <- as.data.frame(Bo_lncRNA_MAxtpm)
Bo_lncRNA_MAxtpm_db$category <- "lncRNA"
Bo_lncRNA_MAxtpm_db$geneID <-row.names(Bo_lncRNA_MAxtpm_db)
colnames(Bo_lncRNA_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
Bo_lncRNA_MAxtpm_db$specie <- "B.oleracea"

# seleccionar solo lincRNAs
Bo_lincRNA_MAxtpm_db <- Bo_lncRNA_MAxtpm_db %>% filter(geneID %in% Bo_lncRNAs_intergenicos_GENES_IDs$V1)
Bo_lincRNA_MAxtpm_db$category <- "lincRNA"


# > C.rubell
Cr_protein_tpm_db_all <- as.data.frame(Cr_protein_tpm)
Cr_lncRNA_tpm_db_all <- as.data.frame(Cr_tpm)

# seleccionar maxima expresion en TPM de lncRNas y proteinas (hezroni y Palos lo hacen)
Cr_lncRNA_MAxtpm <- apply(Cr_lncRNA_tpm_db_all,1,max)
Cr_protein_MAxtpm <- apply(Cr_protein_tpm_db_all,1,max)

# Dataframe proteinas
Cr_protein_MAxtpm_db <- as.data.frame(Cr_protein_MAxtpm)
Cr_protein_MAxtpm_db$category <- "protein-coding"
Cr_protein_MAxtpm_db$geneID <-row.names(Cr_protein_MAxtpm_db)
colnames(Cr_protein_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
Cr_protein_MAxtpm_db$specie <- "C.rubella"

#head(At_protein_MAxtpm_db)

# Dataframe lncRNAs
Cr_lncRNA_MAxtpm_db <- as.data.frame(Cr_lncRNA_MAxtpm)
Cr_lncRNA_MAxtpm_db$category <- "lncRNA"
Cr_lncRNA_MAxtpm_db$geneID <-row.names(Cr_lncRNA_MAxtpm_db)
colnames(Cr_lncRNA_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
Cr_lncRNA_MAxtpm_db$specie <- "C.rubella"

# seleccionar solo lincRNAs
Cr_lincRNA_MAxtpm_db <- Cr_lncRNA_MAxtpm_db %>% filter(geneID %in% Cr_lncRNAs_intergenicos_GENES_IDs$V1)
Cr_lincRNA_MAxtpm_db$category <- "lincRNA"


# > T.parvula
Tp_protein_tpm_db_all <- as.data.frame(Tp_protein_tpm)
Tp_lncRNA_tpm_db_all <- as.data.frame(Tp_tpm)

# seleccionar maxima expresion en TPM de lncRNas y proteinas (hezroni y Palos lo hacen)
Tp_lncRNA_MAxtpm <- apply(Tp_lncRNA_tpm_db_all,1,max)
Tp_protein_MAxtpm <- apply(Tp_protein_tpm_db_all,1,max)

# Dataframe proteinas
Tp_protein_MAxtpm_db <- as.data.frame(Tp_protein_MAxtpm)
Tp_protein_MAxtpm_db$category <- "protein-coding"
Tp_protein_MAxtpm_db$geneID <-row.names(Tp_protein_MAxtpm_db)
colnames(Tp_protein_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
Tp_protein_MAxtpm_db$specie <- "T.parvula"

#head(At_protein_MAxtpm_db)

# Dataframe lncRNAs
Tp_lncRNA_MAxtpm_db <- as.data.frame(Tp_lncRNA_MAxtpm)
Tp_lncRNA_MAxtpm_db$category <- "lncRNA"
Tp_lncRNA_MAxtpm_db$geneID <-row.names(Tp_lncRNA_MAxtpm_db)
colnames(Tp_lncRNA_MAxtpm_db) <- c("MaxTPM", "category", "geneID")
Tp_lncRNA_MAxtpm_db$specie <- "T.parvula"

# seleccionar solo lincRNAs
Tp_lincRNA_MAxtpm_db <- Tp_lncRNA_MAxtpm_db %>% filter(geneID %in% Tp_lncRNAs_intergenicos_GENES_IDs$V1)
Tp_lincRNA_MAxtpm_db$category <- "lincRNA"

#head(At_lncRNA_MAxtpm_db)

# unir
allspecies_expression_protein_lncRNA_MAxtpm_db <- rbind(At_protein_MAxtpm_db,At_lincRNA_MAxtpm_db, Br_protein_MAxtpm_db, Br_lincRNA_MAxtpm_db, Bo_protein_MAxtpm_db,Bo_lincRNA_MAxtpm_db, Cr_protein_MAxtpm_db,Cr_lincRNA_MAxtpm_db, Tp_protein_MAxtpm_db,Tp_lincRNA_MAxtpm_db)
allspecies_expression_protein_lncRNA_MAxtpm_db$log2MaxTPM <- log2(allspecies_expression_protein_lncRNA_MAxtpm_db$MaxTPM)

# Grafica
library(ggplot2)
library(ggpubr)
library(ggsignif)

allspecies_expression_protein_lncRNA_MAxtpm_v1_gg <-  allspecies_expression_protein_lncRNA_MAxtpm_db %>% mutate(category = factor(category, levels = c("lincRNA", "protein-coding")))  %>% mutate(specie = factor(specie, levels = c("B.oleracea","B.rapa", "T.parvula", "C.rubella", "A.thaliana"))) %>% ggplot(aes(x=specie, y=log2MaxTPM, fill=category, colour=category)) +
  geom_boxplot(outlier.color = "black", outlier.alpha = 0.08) + 
  scale_fill_manual(values=c("#3690c0","#969696")) + #color de relleno de los boxplot
  scale_color_manual(values=c("#003f5c","#525252"))  + #color de relleno de los boxplot
  #scale_y_continuous(expand = c(0, 0), limits = c(0, 18))+ # cero en la linea (Antes habia cambiado esto)
  labs(y=expression('Log'[2]~'Max TPM'), x="") + #labels en X y Y
  theme_classic() + # sin cuadricula con fondo blanco
  #coord_cartesian(ylim =  c(0, 20)) + # delimitar
  #scale_y_continuous(breaks=c(0,0.5,1,2,3,4,5), limits = c(0,5)) + # delimitar ylim y divider el eje
  theme(text= element_text(size=23), #cambiar letra y posicion de los labels
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        axis.text.y = element_blank(), # inset italic in the axis x
        axis.text = element_text(size = 20)) +
  coord_flip()

allspecies_expression_protein_lncRNA_MAxtpm_v1_gg

write.table(allspecies_expression_protein_lncRNA_MAxtpm_db, "../Kallisto_myspecies/allspecies_data/allspecies_expression_protein_lncRNA_MAxtpm.tsv", sep="\t")

#  scale_y_continuous(breaks=c(0,500,1000,2000,3000,4000,5000), limits = c(0,5000)) # delimitar ylim y divider el eje
```


```{r Analisis estadistico}
# annotation table with adjusted pvals and y-position of the labels
allspecies_allinformation_MAxtpm_db <- allspecies_expression_protein_lncRNA_MAxtpm_db
allspecies_allinformation_MAxtpm_db <- allspecies_allinformation_MAxtpm_db %>% mutate(specie = ifelse(specie == "A.thaliana","A. thaliana", ifelse(specie == "C.rubella","C. rubella", ifelse(specie == "B.oleracea","B. oleracea", ifelse(specie == "B.rapa","B. rapa", "T. parvula")))))


allspecies_allinformation_MAxtpm_db$category <- as.factor(allspecies_allinformation_MAxtpm_db$category)
allspecies_allinformation_MAxtpm_db$specie <- as.factor(allspecies_allinformation_MAxtpm_db$specie)

anno_df <- compare_means(MaxTPM ~ category, group.by = "specie", data = allspecies_allinformation_MAxtpm_db) %>%
 mutate(y_pos = 30)

anno_df

head(allspecies_allinformation_MAxtpm_db)

#https://github.com/kassambara/ggpubr/issues/65
```


```{r allspecies_expression_protein_lncRNA_MAxtpm_v2_gg}

allspecies_expression_protein_lncRNA_MAxtpm_v2_gg <-  allspecies_allinformation_MAxtpm_db %>% mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>% ggplot(aes(x=category, y=log2MaxTPM)) + 
  geom_point(aes(color=category), position=position_jitterdodge(), alpha=0.4, show.legend =FALSE) + 
  geom_boxplot(aes(fill=category), position=position_dodge(),  show.legend =TRUE) + 
  scale_fill_manual(values=c("#3690c0","#969696")) + #color de relleno de los boxplot
  scale_color_manual(values=c("#003f5c","#525252"))  + #color de relleno de los boxplot
  facet_wrap(~specie, nrow=1) + 
  ggsignif::geom_signif(
    data=anno_df, 
    aes(xmin=group1, xmax=group2, annotations=p.signif, y_position=y_pos), 
    manual=TRUE
  ) +
  labs(y=expression('Log'[2]~'Max TPM'), x="") + #labels en X y Y
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


allspecies_expression_protein_lncRNA_MAxtpm_v2_gg
```


## Numero de lncRNAs (lincRNAs) y Protein-coding genes expresados

```{r Expresion de proteinas y lincRNAs-At}
library(dplyr)

# Seleccionar lincRNAs
AtlincRNAs_GeneID_db <- At_THESIS_lncRNAs_annotated_bed %>% filter(BioType == "lincRNA") %>% # seleccionar solo lincRNAs
  select(GeneID) %>% unique() # seleccionar GeneID


At_lncRNAs_normalized_db <- read.table("allspecies_data/At_lncRNAs_expressed_RLOGnormalized.tsv", header=T)

# solo lincRNAs expresados
#At_ExpressedGeneID_db <- At_lncRNAs_normalized_db %>% filter(AtlincRNA_geneID %in% AtlincRNAs_GeneID_db$GeneID) 
At_ExpressedGeneID_db <- At_lncRNAs_normalized_db %>% filter(GeneID %in% AtlincRNAs_GeneID_db$GeneID) 

At_ExpressedGeneID_db <- rbind(At_ExpressedGeneID_db[,-5], At_protein_normalized_db)
At_ExpressedGeneID_db$GeneID <- row.names(At_ExpressedGeneID_db)
At_ExpressedGeneID_db$GeneID <- gsub("MSTRG.", "AtRegRNA_", At_ExpressedGeneID_db$GeneID)

# Agregar biotipos
At_ExpressedGeneID_db <- At_ExpressedGeneID_db %>% mutate(Biotype =if_else(GeneID %in% AtlincRNAs_GeneID_db$GeneID, "lincRNA",  "protein"))
# Eliminar los que tengan baja expresion, osea cero
At_ExpressedGeneID_db <- At_ExpressedGeneID_db %>% filter(!(At_shoot_R1 ==0 & At_shoot_R2 ==0 & At_root_R1 ==0 & At_root_R2 ==0))

# Agregar donde se expresan diferencialmente
At_ExpressedGeneID_db <- At_ExpressedGeneID_db %>%  mutate(DEG_tissue =if_else(GeneID %in% c(At_lncRNAs_de_gene_names_UP_RegRNAIDs, At_protein_de_gene_names_UP), "shoot",  if_else(GeneID %in% c(At_lncRNAs_de_gene_names_DOWN_RegRNAIDs, At_protein_de_gene_names_DOWN), "root" , if_else(At_shoot_R1 ==0 & At_shoot_R2 ==0 & At_root_R1 ==0 & At_root_R2 ==0 , "WithoutExpressionDetected" ,"Both"))))

# cuantos presentan expresion
At_lncRNAs_Exp_total <- At_ExpressedGeneID_db %>% filter(Biotype =="lincRNA") %>% select(GeneID) %>% n_distinct()  #981 genes de lincRNAs
At_lncRNAs_Exp_shootDEG <- At_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() #43 lincRNAs shoot
At_lncRNAs_Exp_rootDEG <- At_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() #43 lincRNAs root
At_lncRNAs_Exp_BothDEG <- At_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() #895 lincRNAs Both

At_protein_Exp_total <- At_ExpressedGeneID_db %>% filter(Biotype =="protein") %>% select(GeneID) %>% n_distinct() #24461 genes de lincRNAs
At_protein_Exp_shootDEG <- At_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() #6362 lincRNAs shoot
At_protein_Exp_rootDEG <- At_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() #6437 lincRNAs root
At_protein_Exp_BothDEG <- At_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() #11662 lincRNAs Both

# make dataframe
At_ExpGeneID_lincRNAsandPG_db <- data.frame( "Biotype" = c(rep("lincRNA",3), rep("protein",3)),
            "Organ" = rep(c("shoot", "root", "Both"),2),
            "value" = c(At_lncRNAs_Exp_shootDEG, At_lncRNAs_Exp_rootDEG, At_lncRNAs_Exp_BothDEG, At_protein_Exp_shootDEG, At_protein_Exp_rootDEG, At_protein_Exp_BothDEG), 
            "total" = c(rep(At_lncRNAs_Exp_total,3), rep(At_protein_Exp_total,3)), "specie" = "A. thaliana")

At_ExpGeneID_lincRNAsandPG_db <- At_ExpGeneID_lincRNAsandPG_db %>% mutate(proportion = value/total *100)
head(At_ExpGeneID_lincRNAsandPG_db)
head(At_ExpressedGeneID_db)

write.table(At_ExpGeneID_lincRNAsandPG_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_ExpGeneID_lincRNAsandPG_db", quote=FALSE, sep="\t", row.names = F)

write.table(At_ExpressedGeneID_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_ExpressedGeneID_db.tsv", quote=FALSE, sep="\t", row.names = F)

```


```{r Expresion de proteinas y lincRNAs-Br}
library(dplyr)
Br_THESIS_lncRNAs_annotated_bed <- read.table( "../Conservacion_x_secuencia/Paper_conservation/Br_lncRNAs_annotated.tsv", header =T)
# Seleccionar lincRNAs
BrlincRNAs_GeneID_db <- Br_THESIS_lncRNAs_annotated_bed %>% filter(BioType == "lincRNA") %>% # seleccionar solo lincRNAs
  select(GeneID) %>% unique() # seleccionar GeneID
#BrlincRNAs_GeneID_db %>% unique() %>% dim() #1840 genes de lincRNAs
Br_lncRNAs_normalized_db$BrlincRNA_geneID <-row.names(Br_lncRNAs_normalized_db)
Br_lncRNAs_normalized_db$BrlincRNA_geneID <- gsub("MSTRG.", "BrRegRNA_", Br_lncRNAs_normalized_db$BrlincRNA_geneID)
#Br_lncRNAs_normalized_db %>% select(BrlincRNA_geneID) %>% n_distinct() #1739 genes en total

# solo lincRNAs expresados
Br_ExpressedGeneID_db <- Br_lncRNAs_normalized_db %>% filter(BrlincRNA_geneID %in% BrlincRNAs_GeneID_db$GeneID) 
Br_ExpressedGeneID_db$Biotype <- "lincRNA"
#Br_ExpressedGeneID_db %>% select(BrlincRNA_geneID) %>% n_distinct() #1413
Br_protein_normalized_modified <- Br_protein_normalized_db
Br_protein_normalized_modified$Biotype <- "protein"
  
Br_ExpressedGeneID_db <- rbind(Br_ExpressedGeneID_db[,-5], Br_protein_normalized_modified)
Br_ExpressedGeneID_db$GeneID <- row.names(Br_ExpressedGeneID_db)
Br_ExpressedGeneID_db$GeneID <- gsub("MSTRG.", "BrRegRNA_", Br_ExpressedGeneID_db$GeneID)

# Agregar biotipos
#Br_ExpressedGeneID_db <- Br_ExpressedGeneID_db %>% mutate(Biotype =if_else(GeneID %in% BrlincRNAs_GeneID_db$GeneID, "lincRNA",  "protein")) #1413 lincRNAs
Br_ExpressedGeneID_db %>% filter(Biotype == "lincRNA") %>% select(GeneID) %>% n_distinct() # 1413 lincRNAs
Br_ExpressedGeneID_db %>% filter(Biotype == "protein") %>% select(GeneID) %>% n_distinct() # 39607 protein-coding genes

# Eliminar los que tengan baja expresion, osea cero
Br_ExpressedGeneID_db <- Br_ExpressedGeneID_db %>% filter(!(Br_shoot_R1 ==0 & Br_shoot_R2 ==0 & Br_root_R1 ==0 & Br_root_R2 ==0))

Br_ExpressedGeneID_db %>% filter(Biotype == "lincRNA") %>% select(GeneID) %>% n_distinct()

# Agregar donde se expresan diferencialmente
Br_ExpressedGeneID_db <- Br_ExpressedGeneID_db %>% mutate(DEG_tissue = if_else(GeneID %in% c(Br_lncRNAs_de_gene_names_UP_RegRNAIDs, Br_protein_de_gene_names_UP), "shoot",  if_else(GeneID %in% c(Br_lncRNAs_de_gene_names_DOWN_RegRNAIDs,  Br_protein_de_gene_names_DOWN), "root" , if_else(Br_shoot_R1 ==0 & Br_shoot_R2 ==0 & Br_root_R1 ==0 & Br_root_R2 ==0 , "WithoutExpressionDetected" ,"Both"))))

# cuantos presentan expresion
Br_lncRNAs_Exp_total <- Br_ExpressedGeneID_db %>% filter(Biotype =="lincRNA") %>% select(GeneID) %>% n_distinct() # 333 genes de lincRNAs
Br_lncRNAs_Exp_shootDEG <- Br_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() # 45 lincRNAs shoot
Br_lncRNAs_Exp_rootDEG <- Br_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() # 94 lincRNAs root
Br_lncRNAs_Exp_BothDEG <- Br_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() # 194 lincRNAs Both

Br_protein_Exp_total <- Br_ExpressedGeneID_db %>% filter(Biotype =="protein") %>% select(GeneID) %>% n_distinct() # 32383 genes de lincRNAs
Br_protein_Exp_shootDEG <- Br_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() # 6426 lincRNAs shoot
Br_protein_Exp_rootDEG <- Br_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() # 6544 lincRNAs root
Br_protein_Exp_BothDEG <- Br_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() # 19413 lincRNAs Both
head(Br_ExpressedGeneID_db)

#Br_ExpressedGeneID_db %>% filter(GeneID %in% Br_lncRNAs_de_gene_names_UP_RegRNAIDs) %>% select(GeneID) %>% n_distinct() #217
# Br_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="shoot")
#BrlincRNAs_GeneID_db %>% filter(GeneID %in% Br_lncRNAs_de_gene_names_UP_RegRNAIDs) %>% dim() #215
#BrlincRNAs_GeneID_db %>% filter(GeneID %in% Br_lncRNAs_de_gene_names_DOWN_RegRNAIDs) %>% dim() #234

# make dataframe
Br_ExpGeneID_lincRNAsandPG_db <- data.frame( "Biotype" = c(rep("lincRNA",3), rep("protein",3)),
            "Organ" = rep(c("shoot", "root", "Both"),2),
            "value" = c(Br_lncRNAs_Exp_shootDEG, Br_lncRNAs_Exp_rootDEG, Br_lncRNAs_Exp_BothDEG, Br_protein_Exp_shootDEG, Br_protein_Exp_rootDEG, Br_protein_Exp_BothDEG), 
            "total" = c(rep(Br_lncRNAs_Exp_total,3), rep(Br_protein_Exp_total,3)), "specie" = "B. rapa")

Br_ExpGeneID_lincRNAsandPG_db <- Br_ExpGeneID_lincRNAsandPG_db %>% mutate(proportion = value/total *100)
head(Br_ExpGeneID_lincRNAsandPG_db)

write.table(Br_ExpGeneID_lincRNAsandPG_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_ExpGeneID_lincRNAsandPG_db.tsv", quote=FALSE, sep="\t", row.names = F)
write.table(Br_ExpressedGeneID_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_ExpressedGeneID_db.tsv", quote=FALSE, sep="\t", row.names = F)

```


```{r Expresion de proteinas y lincRNAs-Bo}
library(dplyr)
Bo_THESIS_lncRNAs_annotated_bed <- read.table( "../Conservacion_x_secuencia/Paper_conservation/Bo_lncRNAs_annotated.tsv", header =T)

# Seleccionar lincRNAs
BolincRNAs_GeneID_db <- Bo_THESIS_lncRNAs_annotated_bed %>% filter(BioType == "lincRNA") %>% # seleccionar solo lincRNAs
  select(GeneID) # seleccionar GeneID
#BolincRNAs_GeneID_db %>% unique() %>% dim() #926 lincRNAs
Bo_lncRNAs_normalized_db$BolincRNA_geneID <-row.names(Bo_lncRNAs_normalized_db)
Bo_lncRNAs_normalized_db$BolincRNA_geneID <- gsub("MSTRG.", "BoRegRNA_", Bo_lncRNAs_normalized_db$BolincRNA_geneID)


# solo lincRNAs expresados
Bo_ExpressedGeneID_db <- Bo_lncRNAs_normalized_db %>% filter(BolincRNA_geneID %in% BolincRNAs_GeneID_db$GeneID) 
Bo_ExpressedGeneID_db$Biotype <- "lincRNA"
#Bo_ExpressedGeneID_db %>% select(BolincRNA_geneID) %>% n_distinct() #878 lincRNAs
Bo_protein_normalized_modified <- Bo_protein_normalized_db
Bo_protein_normalized_modified$Biotype <- "protein"

Bo_ExpressedGeneID_db <- rbind(Bo_ExpressedGeneID_db[,-5], Bo_protein_normalized_modified)
Bo_ExpressedGeneID_db$GeneID <- row.names(Bo_ExpressedGeneID_db)
Bo_ExpressedGeneID_db$GeneID <- gsub("MSTRG.", "BoRegRNA_", Bo_ExpressedGeneID_db$GeneID)

# Agregar biotipos
#Bo_ExpressedGeneID_db <- Bo_ExpressedGeneID_db %>% mutate(Biotype =if_else(GeneID %in% BolincRNAs_GeneID_db$GeneID, "lincRNA",  "protein"))
Bo_ExpressedGeneID_db %>% filter(Biotype == "lincRNA") %>% select(GeneID) %>% n_distinct() # 1413 lincRNAs
Bo_ExpressedGeneID_db %>% filter(Biotype == "protein") %>% select(GeneID) %>% n_distinct() # 39607 protein-coding genes

# Eliminar los que tengan baja expresion, osea cero
Bo_ExpressedGeneID_db <- Bo_ExpressedGeneID_db %>% filter(!(Bo_shoot_R1 ==0 & Bo_shoot_R2 ==0 & Bo_root_R1 ==0 & Bo_root_R2 ==0))

# Agregar donde se expresan diferencialmente
Bo_ExpressedGeneID_db <- Bo_ExpressedGeneID_db %>%  mutate(DEG_tissue =if_else(GeneID %in% c(Bo_lncRNAs_de_gene_names_UP_RegRNAIDs, Bo_protein_de_gene_names_UP), "shoot",  if_else(GeneID %in% c(Bo_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Bo_protein_de_gene_names_DOWN), "root" , if_else(Bo_shoot_R1 ==0 & Bo_shoot_R2 ==0 & Bo_root_R1 ==0 & Bo_root_R2 ==0 , "WithoutExpressionDetected" ,"Both"))))

# cuantos presentan expresion
Bo_lncRNAs_Exp_total <- Bo_ExpressedGeneID_db %>% filter(Biotype =="lincRNA") %>% select(GeneID) %>% n_distinct() # 333 genes de lincRNAs
Bo_lncRNAs_Exp_shootDEG <- Bo_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() # 45 lincRNAs shoot
Bo_lncRNAs_Exp_rootDEG <- Bo_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() # 94 lincRNAs root
Bo_lncRNAs_Exp_BothDEG <- Bo_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() # 194 lincRNAs Both

Bo_protein_Exp_total <- Bo_ExpressedGeneID_db %>% filter(Biotype =="protein") %>% select(GeneID) %>% n_distinct() # 32383 genes de lincRNAs
Bo_protein_Exp_shootDEG <- Bo_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() # 6426 lincRNAs shoot
Bo_protein_Exp_rootDEG <- Bo_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() # 6544 lincRNAs root
Bo_protein_Exp_BothDEG <- Bo_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() # 19413 lincRNAs Both
head(Bo_ExpressedGeneID_db)

# make dataframe
Bo_ExpGeneID_lincRNAsandPG_db <- data.frame( "Biotype" = c(rep("lincRNA",3), rep("protein",3)),
            "Organ" = rep(c("shoot", "root", "Both"),2),
            "value" = c(Bo_lncRNAs_Exp_shootDEG, Bo_lncRNAs_Exp_rootDEG, Bo_lncRNAs_Exp_BothDEG, Bo_protein_Exp_shootDEG, Bo_protein_Exp_rootDEG, Bo_protein_Exp_BothDEG), 
            "total" = c(rep(Bo_lncRNAs_Exp_total,3), rep(Bo_protein_Exp_total,3)), "specie" = "B. oleracea")

Bo_ExpGeneID_lincRNAsandPG_db <- Bo_ExpGeneID_lincRNAsandPG_db %>% mutate(proportion = value/total *100)
head(Bo_ExpGeneID_lincRNAsandPG_db)

write.table(Bo_ExpGeneID_lincRNAsandPG_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_ExpGeneID_lincRNAsandPG_db.tsv", quote=FALSE, sep="\t", row.names = F)
write.table(Bo_ExpressedGeneID_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_ExpressedGeneID_db.tsv", quote=FALSE, sep="\t", row.names = F)


```


```{r Expresion de proteinas y lincRNAs-Cr}
library(dplyr)
Cr_THESIS_lncRNAs_annotated_bed <- read.table( "../Conservacion_x_secuencia/Paper_conservation/Cr_lncRNAs_annotated.tsv", header =T)

# Seleccionar lincRNAs
CrlincRNAs_GeneID_db <- Cr_THESIS_lncRNAs_annotated_bed %>% filter(BioType == "lincRNA") %>% # seleccionar solo lincRNAs
  select(GeneID) # seleccionar GeneID
Cr_lncRNAs_normalized_db$CrlincRNA_geneID <-row.names(Cr_lncRNAs_normalized_db)
Cr_lncRNAs_normalized_db$CrlincRNA_geneID <- gsub("MSTRG.", "CrRegRNA_", Cr_lncRNAs_normalized_db$CrlincRNA_geneID)

# solo lincRNAs expresados
Cr_ExpressedGeneID_db <- Cr_lncRNAs_normalized_db %>% filter(CrlincRNA_geneID %in% CrlincRNAs_GeneID_db$GeneID) 
Cr_ExpressedGeneID_db$Biotype <- "lincRNA"
#Br_ExpressedGeneID_db %>% select(BrlincRNA_geneID) %>% n_distinct() #1413
Cr_protein_normalized_modified <- Cr_protein_normalized_db
Cr_protein_normalized_modified$Biotype <- "protein"

Cr_ExpressedGeneID_db <- rbind(Cr_ExpressedGeneID_db[,-5], Cr_protein_normalized_modified)
Cr_ExpressedGeneID_db$GeneID <- row.names(Cr_ExpressedGeneID_db)
Cr_ExpressedGeneID_db$GeneID <- gsub("MSTRG.", "CrRegRNA_", Cr_ExpressedGeneID_db$GeneID)

# Agregar biotipos
#Cr_ExpressedGeneID_db <- Cr_ExpressedGeneID_db %>% mutate(Biotype =if_else(GeneID %in% CrlincRNAs_GeneID_db$GeneID, "lincRNA",  "protein"))
Cr_ExpressedGeneID_db %>% filter(Biotype == "lincRNA") %>% select(GeneID) %>% n_distinct() # 1413 lincRNAs
Cr_ExpressedGeneID_db %>% filter(Biotype == "protein") %>% select(GeneID) %>% n_distinct() # 39607 protein-coding genes

# Eliminar los que tengan baja expresion, osea cero
Cr_ExpressedGeneID_db <- Cr_ExpressedGeneID_db %>% filter(!(Cr_shoot_R1 ==0 & Cr_shoot_R2 ==0 & Cr_root_R1 ==0 & Cr_root_R2 ==0))

# Agregar donde se expresan diferencialmente
Cr_ExpressedGeneID_db <- Cr_ExpressedGeneID_db %>%  mutate(DEG_tissue =if_else(GeneID %in% c(Cr_lncRNAs_de_gene_names_UP_RegRNAIDs, Cr_protein_de_gene_names_UP), "shoot",  if_else(GeneID %in% c(Cr_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Cr_protein_de_gene_names_DOWN), "root" , if_else(Cr_shoot_R1 ==0 & Cr_shoot_R2 ==0 & Cr_root_R1 ==0 & Cr_root_R2 ==0 , "WithoutExpressionDetected" ,"Both"))))

# cuantos presentan expresion
Cr_lncRNAs_Exp_total <- Cr_ExpressedGeneID_db %>% filter(Biotype =="lincRNA") %>% select(GeneID) %>% n_distinct() # 333 genes de lincRNAs
Cr_lncRNAs_Exp_shootDEG <- Cr_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() # 45 lincRNAs shoot
Cr_lncRNAs_Exp_rootDEG <- Cr_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() # 94 lincRNAs root
Cr_lncRNAs_Exp_BothDEG <- Cr_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() # 194 lincRNAs Both

Cr_protein_Exp_total <- Cr_ExpressedGeneID_db %>% filter(Biotype =="protein") %>% select(GeneID) %>% n_distinct() # 32383 genes de lincRNAs
Cr_protein_Exp_shootDEG <- Cr_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() # 6426 lincRNAs shoot
Cr_protein_Exp_rootDEG <- Cr_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() # 6544 lincRNAs root
Cr_protein_Exp_BothDEG <- Cr_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() # 19413 lincRNAs Both
head(Cr_ExpressedGeneID_db)

# make dataframe
Cr_ExpGeneID_lincRNAsandPG_db <- data.frame( "Biotype" = c(rep("lincRNA",3), rep("protein",3)),
            "Organ" = rep(c("shoot", "root", "Both"),2),
            "value" = c(Cr_lncRNAs_Exp_shootDEG, Cr_lncRNAs_Exp_rootDEG, Cr_lncRNAs_Exp_BothDEG, Cr_protein_Exp_shootDEG, Cr_protein_Exp_rootDEG, Cr_protein_Exp_BothDEG), 
            "total" = c(rep(Cr_lncRNAs_Exp_total,3), rep(Cr_protein_Exp_total,3)), "specie" = "C. rubella")

Cr_ExpGeneID_lincRNAsandPG_db <- Cr_ExpGeneID_lincRNAsandPG_db %>% mutate(proportion = value/total *100)
head(Cr_ExpGeneID_lincRNAsandPG_db)

write.table(Cr_ExpGeneID_lincRNAsandPG_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_ExpGeneID_lincRNAsandPG_db.tsv", quote=FALSE, sep="\t", row.names = F)
write.table(Cr_ExpressedGeneID_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_ExpressedGeneID_db.tsv", quote=FALSE, sep="\t", row.names = F)

```


```{r Expresion de proteinas y lincRNAs-Tp}
library(dplyr)

Tp_THESIS_lncRNAs_annotated_bed <- read.table( "../Conservacion_x_secuencia/Paper_conservation/Tp_lncRNAs_annotated.tsv", header =T)

# Seleccionar lincRNAs
TplincRNAs_GeneID_db <- Tp_THESIS_lncRNAs_annotated_bed %>% filter(BioType == "lincRNA") %>% # seleccionar solo lincRNAs
  select(GeneID) # seleccionar GeneID
#TplincRNAs_GeneID_db %>% unique() %>% dim() #644 genes de lincRNAs
Tp_lncRNAs_normalized_db$TplincRNA_geneID <- row.names(Tp_lncRNAs_normalized_db)
Tp_lncRNAs_normalized_db$TplincRNA_geneID <- gsub("MSTRG.", "TpRegRNA_", Tp_lncRNAs_normalized_db$TplincRNA_geneID)
#Tp_lncRNAs_normalized_db %>% select(TplincRNA_geneID) %>% n_distinct() #1101 genes en total

# solo lincRNAs expresados
Tp_ExpressedGeneID_db <- Tp_lncRNAs_normalized_db %>% filter(TplincRNA_geneID %in% TplincRNAs_GeneID_db$GeneID) 
Tp_ExpressedGeneID_db$Biotype <- "lincRNA"
#Tp_ExpressedGeneID_db %>% select(TplincRNA_geneID) %>% n_distinct() #644
Tp_protein_normalized_modified <- Tp_protein_normalized_db
Tp_protein_normalized_modified$Biotype <- "protein"

Tp_ExpressedGeneID_db <- rbind(Tp_ExpressedGeneID_db[,-5], Tp_protein_normalized_modified)
Tp_ExpressedGeneID_db$GeneID <- row.names(Tp_ExpressedGeneID_db)
Tp_ExpressedGeneID_db$GeneID <- gsub("MSTRG.", "TpRegRNA_", Tp_ExpressedGeneID_db$GeneID)

# Agregar biotipos
#Tp_ExpressedGeneID_db <- Tp_ExpressedGeneID_db %>% mutate(Biotype =if_else(GeneID %in% TplincRNAs_GeneID_db$GeneID, "lincRNA",  "protein"))
# Eliminar los que tengan baja expresion, osea cero
Tp_ExpressedGeneID_db <- Tp_ExpressedGeneID_db %>% filter(!(Tp_shoot_R1 ==0 & Tp_shoot_R2 ==0 & Tp_root_R1 ==0 & Tp_root_R2 ==0))
Tp_ExpressedGeneID_db %>% filter(Biotype == "lincRNA") %>% select(GeneID) %>% n_distinct() # 595 lincRNAs
Tp_ExpressedGeneID_db %>% filter(Biotype == "protein") %>% select(GeneID) %>% n_distinct() # 20675 protein-coding genes

# Agregar donde se expresan diferencialmente
Tp_ExpressedGeneID_db <- Tp_ExpressedGeneID_db %>%  mutate(DEG_tissue =if_else(GeneID %in% c(Tp_lncRNAs_de_gene_names_UP_RegRNAIDs, Tp_protein_de_gene_names_UP), "shoot",  if_else(GeneID %in% c(Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Tp_protein_de_gene_names_DOWN), "root" , if_else(Tp_shoot_R1 ==0 & Tp_shoot_R2 ==0 & Tp_root_R1 ==0 & Tp_root_R2 ==0 , "WithoutExpressionDetected" ,"Both"))))

#Tp_ExpressedGeneID_db %>% filter(GeneID %in% Tp_lncRNAs_de_gene_names_UP_RegRNAIDs & Biotype =="lincRNA") %>% select(GeneID) %>% n_distinct() #98

# cuantos presentan expresion
Tp_lncRNAs_Exp_total <- Tp_ExpressedGeneID_db %>% filter(Biotype =="lincRNA") %>% select(GeneID) %>% n_distinct() # 595 genes de lincRNAs
#Tp_lncRNAs_Exp_shootDEG <- Tp_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() # 45 lincRNAs shoot
Tp_lncRNAs_Exp_shootDEG <-  Tp_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="shoot") %>% filter(GeneID %in% Tp_lncRNAs_de_gene_names_UP_RegRNAIDs) %>% select(GeneID) %>% n_distinct() # 98 lincRNAs shoot

#Tp_lncRNAs_Exp_rootDEG <- Tp_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() # 94 lincRNAs root
Tp_lncRNAs_Exp_rootDEG <- Tp_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="root") %>% filter(GeneID %in% Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs) %>% select(GeneID) %>% n_distinct()
#Tp_lncRNAs_Exp_BothDEG <- Tp_ExpressedGeneID_db %>% filter(Biotype =="lincRNA" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() # 194 lincRNAs Both
Tp_lncRNAs_Exp_BothDEG <- Tp_ExpressedGeneID_db %>% filter(!(GeneID %in% c(Tp_lncRNAs_de_gene_names_UP_RegRNAIDs, Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs))) %>% filter(Biotype =="lincRNA") %>% select(GeneID) %>% n_distinct() # 194 lincRNAs Both


Tp_protein_Exp_total <- Tp_ExpressedGeneID_db %>% filter(Biotype =="protein") %>% select(GeneID) %>% n_distinct() # 32383 genes de lincRNAs
Tp_protein_Exp_shootDEG <- Tp_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="shoot") %>% select(GeneID) %>% n_distinct() # 6426 lincRNAs shoot
Tp_protein_Exp_rootDEG <- Tp_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="root") %>% select(GeneID) %>% n_distinct() # 6544 lincRNAs root
Tp_protein_Exp_BothDEG <- Tp_ExpressedGeneID_db %>% filter(Biotype =="protein" & DEG_tissue =="Both") %>% select(GeneID) %>% n_distinct() # 19413 lincRNAs Both
#head(Tp_ExpressedGeneID_db)

# make dataframe
Tp_ExpGeneID_lincRNAsandPG_db <- data.frame( "Biotype" = c(rep("lincRNA",3), rep("protein",3)),
            "Organ" = rep(c("shoot", "root", "Both"),2),
            "value" = c(Tp_lncRNAs_Exp_shootDEG, Tp_lncRNAs_Exp_rootDEG, Tp_lncRNAs_Exp_BothDEG, Tp_protein_Exp_shootDEG, Tp_protein_Exp_rootDEG, Tp_protein_Exp_BothDEG), 
            "total" = c(rep(Tp_lncRNAs_Exp_total,3), rep(Tp_protein_Exp_total,3)), "specie" = "T. parvula")

Tp_ExpGeneID_lincRNAsandPG_db <- Tp_ExpGeneID_lincRNAsandPG_db %>% mutate(proportion = value/total *100)
head(Tp_ExpGeneID_lincRNAsandPG_db)

# lincRNAs expresados en shoot
#Tp_lncRNAs_de_gene_matrix_UP_db %>% filter(GeneID %in% TplincRNA_geneID_selected$GeneID) %>% select(GeneID) %>% n_distinct()
# lincRNAs expresados en root
#Tp_lncRNAs_de_gene_matrix_DOWN_db %>% filter(GeneID %in% TplincRNA_geneID_selected$GeneID) %>% select(GeneID) %>% n_distinct()

write.table(Tp_ExpGeneID_lincRNAsandPG_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_ExpGeneID_lincRNAsandPG_db.tsv", quote=FALSE, sep="\t", row.names = F)
write.table(Tp_ExpressedGeneID_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_ExpressedGeneID_db.tsv", quote=FALSE, sep="\t", row.names = F)

```


```{r COWA_ExpGeneIDs_proportion}
ExpGeneIDs_proportion_db <- rbind(At_ExpGeneID_lincRNAsandPG_db, Cr_ExpGeneID_lincRNAsandPG_db, Tp_ExpGeneID_lincRNAsandPG_db, Br_ExpGeneID_lincRNAsandPG_db, Bo_ExpGeneID_lincRNAsandPG_db)
head(ExpGeneIDs_proportion_db)

ExpGeneIDs_proportion_db$Organ <- gsub("Both", "global", ExpGeneIDs_proportion_db$Organ)

write.table(ExpGeneIDs_proportion_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/ExpGeneIDs_proportion_db.tsv", quote=FALSE, sep="\t", row.names = F)


#View(ExpGeneIDs_proportion_db)

# Cambiar both por global


# DEG
COWA_ExpGeneIDs_proportion <- ExpGeneIDs_proportion_db %>% mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>%  ggplot(aes(x=Biotype, y=proportion, fill = Organ)) + 
  geom_bar(stat="identity", width=0.7) + 
  facet_wrap(~specie, strip.position="bottom", ncol=5) +
  #geom_text(aes(label=value_v2),  vjust=-0.5, color="black",
  #          position = position_dodge(0.9), size=5) + #texto dentro de las barras
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  scale_y_continuous(expand = c(0, 0), limits = c(0, 100))+ # cero en la linea
  labs(y="Porcentaje de genes \nexpresados por organo", x="") +  # agregar salto \n
  theme_classic() + # sin cuadricula con fondo blanco
  # theme_bw() + # cuadricula con fondo blanco y marco en bordes
  theme(text= element_text(size=20), #cambiar letra y posicion de los labels
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        axis.text.y = element_text(face = "italic"), # inset italic in the axis x
        axis.text = element_text(size = 15),
        axis.text.x = element_text(size = 10),
        strip.text = element_text(face = "italic"),# nombres de especies en italicas 
        strip.placement= "outside", # colocar fuera de la grafica
        strip.background = element_blank())  # sin relleno  # nombres de especies en italicas 

COWA_ExpGeneIDs_proportion
#28B463 verde
# #B47028 cafe
```


```{r COWA_ExpGeneIDs_proportion}

# DEG
COWA_ExpGeneIDs_proportion_ENGLISH <- ExpGeneIDs_proportion_db %>% mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>%  ggplot(aes(x=Biotype, y=proportion, fill = Organ)) + 
  geom_bar(stat="identity", width=0.7) + 
  facet_wrap(~specie, strip.position="bottom", ncol=5) +
  #geom_text(aes(label=value_v2),  vjust=-0.5, color="black",
  #          position = position_dodge(0.9), size=5) + #texto dentro de las barras
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  scale_y_continuous(expand = c(0, 0), limits = c(0, 100))+ # cero en la linea
  labs(y="Percentage of genes \nexpressed for each organ", x="") +  # agregar salto \n
  theme_classic() + # sin cuadricula con fondo blanco
  # theme_bw() + # cuadricula con fondo blanco y marco en bordes
  theme(text= element_text(size=20), #cambiar letra y posicion de los labels
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        axis.text.y = element_text(face = "italic"), # inset italic in the axis x
        axis.text = element_text(size = 15),
        axis.text.x = element_text(size = 10),
        strip.text = element_text(face = "italic"),# nombres de especies en italicas 
        strip.placement= "outside", # colocar fuera de la grafica
        strip.background = element_blank())  # sin relleno  # nombres de especies en italicas 

COWA_ExpGeneIDs_proportion_ENGLISH
#28B463 verde
# #B47028 cafe


save_plot("../Conservacion_x_secuencia/Paper_conservation/Athaliana_expression_proportion.png", COWA_ExpGeneIDs_proportion_ENGLISH, base_height = 12, base_width = 8, base_asp = 1.1)

```



```{r}
ExpGeneIDs_proportion_db %>% filter(Organ == "Both" & Biotype == "lincRNA")
ExpGeneIDs_proportion_db %>% filter(Organ == "shoot" & Biotype == "lincRNA")
ExpGeneIDs_proportion_db %>% filter(Organ == "root" & Biotype == "lincRNA")

ExpGeneIDs_proportion_db %>% filter(Organ %in% c("root", "shoot") & Biotype == "lincRNA" & specie == "A. thaliana") %>% select(proportion) %>% sum() 

ExpGeneIDs_proportion_db %>% filter(Organ %in% c("root", "shoot") & Biotype == "lincRNA" & specie == "C. rubella") %>% select(proportion) %>% sum() 

ExpGeneIDs_proportion_db %>% filter(Organ %in% c("root", "shoot") & Biotype == "lincRNA" & specie == "T. parvula") %>% select(proportion) %>% sum() 

ExpGeneIDs_proportion_db %>% filter(Organ %in% c("root", "shoot") & Biotype == "lincRNA" & specie == "B. rapa") %>% select(proportion) %>% sum() 

ExpGeneIDs_proportion_db %>% filter(Organ %in% c("root", "shoot") & Biotype == "lincRNA" & specie == "B. oleracea") %>% select(proportion) %>% sum() 

```



```{r Porcentaje de los genes diferencialmente expresados}
# quedarnos solo con DEG
ExpGeneIDs_proportionDEG_db <- ExpGeneIDs_proportion_db %>% filter(!(Organ == "Both")) %>% select(-total, -proportion) 
#Obtener total
Total_DEG_byspecieandoBiotype <-ExpGeneIDs_proportionDEG_db %>% group_by(specie, Biotype) %>% summarise(Total = sum(value))
# agregarlo
ExpGeneIDs_proportionDEG_db <- left_join(ExpGeneIDs_proportionDEG_db, Total_DEG_byspecieandoBiotype, by = c("Biotype", "specie"))
# Agregar proporcion
ExpGeneIDs_proportionDEG_db <- ExpGeneIDs_proportionDEG_db %>% mutate(proportion = value/Total *100)

head(ExpGeneIDs_proportionDEG_db)

View(ExpGeneIDs_proportionDEG_db)
```


Se identificaron 2490 genes de lincRNAs (2634 transcritos)  de A. thaliana, 420 (436 transcritos)  C. rubella, 644 (661 transcritos) T. parvula, 1840 (1886 transcritos)  B. rapa y 926 (926 transcritos) B. oleracea


```{r COWB_ExpGeneIDs_proportionDEG_db}
COWB_ExpGeneIDs_proportionDEG_db <- ExpGeneIDs_proportionDEG_db %>%  mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>% ggplot(aes(x =Biotype, y =proportion)) +
  geom_point(aes(colour= factor(specie), shape= Biotype), size =5) +
  facet_wrap(~Organ, strip.position="bottom",)+ 
  scale_color_manual(values = c("#081d58","#1d91c0",'#7fcdbb', '#8c6bb1', '#810f7c')) +   
  labs(x ="", y="Genes diferencialmente \nexpresados (%)")+
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20, family="sans"), #cambiar el tamano de la letra y seleccionar letra Arial
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        axis.text.x = element_text(size =15),
        strip.placement= "outside", # colocar fuera de la grafica
        strip.text = element_text(size =15, face = "bold")) 
COWB_ExpGeneIDs_proportionDEG_db
```

```{r COWB_ExpGeneIDs_proportionDEG_db}
COWB_ExpGeneIDs_proportionDEG_ENGLISH_db <- ExpGeneIDs_proportionDEG_db %>%  mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>% ggplot(aes(x =Biotype, y =proportion)) +
  geom_point(aes(colour= factor(specie), shape= Biotype), size =5) +
  facet_wrap(~Organ, strip.position="bottom",)+ 
  scale_color_manual(values = c("#081d58","#1d91c0",'#7fcdbb', '#8c6bb1', '#810f7c')) +   
  labs(x ="", y="DEG (%)")+
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20, family="sans"), #cambiar el tamano de la letra y seleccionar letra Arial
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        axis.text.x = element_text(size =15),
        strip.placement= "outside", # colocar fuera de la grafica
        strip.text = element_text(size =15, face = "bold")) 
COWB_ExpGeneIDs_proportionDEG_ENGLISH_db


save_plot("../Conservacion_x_secuencia/Paper_conservation/Figura2AB_PArt2_DEG.png", COWB_ExpGeneIDs_proportionDEG_ENGLISH_db, base_height = 8, base_width = 12, base_asp = 1.1)
```


```{r}
ExpGeneIDs_proportionDEG_db %>%  mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>% ggplot(aes(x =Organ, y =proportion)) +
  geom_point(aes(colour= factor(specie), shape= Biotype), size =5) +
  facet_wrap(~Biotype)+ 
  scale_color_manual(values = c("#081d58","#1d91c0",'#7fcdbb', '#8c6bb1', '#810f7c')) +   
  labs(x ="", y="Genes diferencialmente expresados (%)")+
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=15, family="sans"), #cambiar el tamano de la letra y seleccionar letra Arial
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        axis.text.x = element_text(size =12),
        strip.text = element_text(size =12)) 
```


## Patrones de expresion por tejido (MaxTPM)

```{r Analisis estadistico}
#
allspecies_allinformation_MAxtpm_Organs_db <- allspecies_allinformation_MAxtpm_db %>% mutate(DEG_tissue =if_else(geneID %in% c(At_lncRNAs_de_gene_names_UP_RegRNAIDs, At_protein_de_gene_names_UP, Br_lncRNAs_de_gene_names_UP_RegRNAIDs, Br_protein_de_gene_names_UP, Bo_lncRNAs_de_gene_names_UP_RegRNAIDs, Bo_protein_de_gene_names_UP, Tp_lncRNAs_de_gene_names_UP_RegRNAIDs, Tp_protein_de_gene_names_UP, Cr_lncRNAs_de_gene_names_UP_RegRNAIDs, Cr_protein_de_gene_names_UP), "shoot",  
                if_else(geneID %in% c(At_lncRNAs_de_gene_names_DOWN_RegRNAIDs, At_protein_de_gene_names_DOWN, Br_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Br_protein_de_gene_names_DOWN, Bo_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Bo_protein_de_gene_names_DOWN, Cr_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Cr_protein_de_gene_names_DOWN, Tp_lncRNAs_de_gene_names_DOWN_RegRNAIDs, Tp_protein_de_gene_names_DOWN), "root", "Both")))


allspecies_allinformation_MAxtpm_db$category <- as.factor(allspecies_allinformation_MAxtpm_db$category)
allspecies_allinformation_MAxtpm_db$specie <- as.factor(allspecies_allinformation_MAxtpm_db$specie)

anno_df <- compare_means(MaxTPM ~ category, group.by = "specie", data = allspecies_allinformation_MAxtpm_db) %>%
 mutate(y_pos = 30)

anno_df

head(allspecies_allinformation_MAxtpm_db)

#https://github.com/kassambara/ggpubr/issues/65
```


```{r allspecies_expression_protein_lncRNA_MAxtpm_v2_gg}

allspecies_allinformation_MAxtpm_Organs_db %>% mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>% ggplot(aes(x=category, y=log2MaxTPM)) + 
  #geom_point(aes(color=category), position=position_jitterdodge(), alpha=0.4, show.legend =FALSE) + 
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  #scale_fill_manual(values=c("#3690c0","#969696")) + #color de relleno de los boxplot
  #scale_color_manual(values=c("#003f5c","#525252"))  + #color de relleno de los boxplot
  facet_wrap(~specie, nrow=1) + 
  #ggsignif::geom_signif(
 #   data=anno_df, 
  #  aes(xmin=group1, xmax=group2, annotations=p.signif, y_position=y_pos), 
  #  manual=TRUE
  #) +
  labs(y=expression('Log'[2]~'Max TPM'), x="") + #labels en X y Y
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 



```

```{r MAx Rlog- lincRNAs y proteinas}
At_ExpressedGeneID_MAxRlog_db <- At_ExpressedGeneID_db %>% mutate(MaxRlog = apply(At_ExpressedGeneID_db[,1:4], 1, max))
At_ExpressedGeneID_MAxRlog_db <- At_ExpressedGeneID_MAxRlog_db[,c(1:4,6,5,7,8)]
At_ExpressedGeneID_MAxRlog_db$specie <- "A. thaliana"
colnames(At_ExpressedGeneID_MAxRlog_db) <- c("Shoot_R1_rlog", "Shoot_R2_rlog", "Root_R1_rlog", "Root_R2_rlog", "Biotype", "GeneID",      "DEG_tissue", "MaxRlog", "specie")

Br_ExpressedGeneID_MAxRlog_db <- Br_ExpressedGeneID_db %>% mutate(MaxRlog = apply(Br_ExpressedGeneID_db[,1:4], 1, max))
Br_ExpressedGeneID_MAxRlog_db$specie <- "B. rapa"
colnames(Br_ExpressedGeneID_MAxRlog_db) <- c("Shoot_R1_rlog", "Shoot_R2_rlog", "Root_R1_rlog", "Root_R2_rlog", "Biotype", "GeneID",      "DEG_tissue", "MaxRlog", "specie")

Bo_ExpressedGeneID_MAxRlog_db <- Bo_ExpressedGeneID_db %>% mutate(MaxRlog = apply(Bo_ExpressedGeneID_db[,1:4], 1, max))
Bo_ExpressedGeneID_MAxRlog_db$specie <- "B. oleracea"
colnames(Bo_ExpressedGeneID_MAxRlog_db) <- c("Shoot_R1_rlog", "Shoot_R2_rlog", "Root_R1_rlog", "Root_R2_rlog", "Biotype", "GeneID",      "DEG_tissue", "MaxRlog", "specie")

Tp_ExpressedGeneID_MAxRlog_db <- Tp_ExpressedGeneID_db %>% mutate(MaxRlog = apply(Tp_ExpressedGeneID_db[,1:4], 1, max))
Tp_ExpressedGeneID_MAxRlog_db$specie <- "T. parvula"
colnames(Tp_ExpressedGeneID_MAxRlog_db) <- c("Shoot_R1_rlog", "Shoot_R2_rlog", "Root_R1_rlog", "Root_R2_rlog", "Biotype", "GeneID",      "DEG_tissue", "MaxRlog", "specie")

Cr_ExpressedGeneID_MAxRlog_db <- Cr_ExpressedGeneID_db %>% mutate(MaxRlog = apply(Cr_ExpressedGeneID_db[,1:4], 1, max))
Cr_ExpressedGeneID_MAxRlog_db$specie <- "C. rubella"
colnames(Cr_ExpressedGeneID_MAxRlog_db) <- c("Shoot_R1_rlog", "Shoot_R2_rlog", "Root_R1_rlog", "Root_R2_rlog", "Biotype", "GeneID",      "DEG_tissue", "MaxRlog", "specie")

#unir
Allspecies_ExpressedGeneID_MAxRlog_db <- rbind(At_ExpressedGeneID_MAxRlog_db, Br_ExpressedGeneID_MAxRlog_db, Bo_ExpressedGeneID_MAxRlog_db, Cr_ExpressedGeneID_MAxRlog_db, Tp_ExpressedGeneID_MAxRlog_db)

head(Allspecies_ExpressedGeneID_MAxRlog_db)
```

```{r allspecies_expression_protein_lncRNA_MAxtpm_v2_gg}
library(dplyr)
library(ggplot2)

tiff(file="allspecies_expression_MaxRlog.tiff", width = 18, height = 10, units = 'in', res = 150)

Allspecies_ExpressedGeneID_MAxRlog_db %>% ggplot(aes(x=Biotype, y=MaxRlog)) + 
  #geom_point(aes(color=category), position=position_jitterdodge(), alpha=0.4, show.legend =FALSE) + 
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  #scale_fill_manual(values=c("#3690c0","#969696")) + #color de relleno de los boxplot
  #scale_color_manual(values=c("#003f5c","#525252"))  + #color de relleno de los boxplot
  facet_wrap(~specie, nrow=1) + 
  #ggsignif::geom_signif(
 #   data=anno_df, 
  #  aes(xmin=group1, xmax=group2, annotations=p.signif, y_position=y_pos), 
  #  manual=TRUE
  #) +
  labs(y=expression('Max Rlog'), x="") + #labels en X y Y
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 

dev.off()

```

## Boxplot - Valores de Rlog de todos los datos

```{r all normalized counts Rlog- lincRNAs y proteinas}
# Modificar tabla para Contener toda la expresion por genes
library(tidyr)
library(tidyverse)

#> A. thaliana
At_expression_plot <-pivot_longer(At_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

At_expression_plot$specie <- "A. thaliana"

#> B. rapa
Br_expression_plot <-pivot_longer(Br_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

Br_expression_plot$specie <- "B. rapa"

#> B. oleracea
Bo_expression_plot <-pivot_longer(Bo_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

Bo_expression_plot$specie <- "B. oleracea"

#> C. rubella
Cr_expression_plot <-pivot_longer(Cr_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

Cr_expression_plot$specie <- "C. rubella"

#> T. parvula
Tp_expression_plot <-pivot_longer(Tp_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

Tp_expression_plot$specie <- "T. parvula"


#unir
Allspecies_ExpressedGeneID_allRlog_db <- rbind(At_expression_plot, Br_expression_plot, Bo_expression_plot, Cr_expression_plot, Tp_expression_plot)

head(Allspecies_ExpressedGeneID_allRlog_db)

write.table(At_expression_plot,file ="S:/Repositorio_scripts/Expressed_lncRNAs/At_expression_plot.tsv", quote=FALSE, sep="\t", row.names = F)

write.table(Br_expression_plot,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Br_expression_plot.tsv", quote=FALSE, sep="\t", row.names = F)

write.table(Bo_expression_plot,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Bo_expression_plot.tsv", quote=FALSE, sep="\t", row.names = F)

write.table(Cr_expression_plot,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Cr_expression_plot.tsv", quote=FALSE, sep="\t", row.names = F)

write.table(Tp_expression_plot,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Tp_expression_plot.tsv", quote=FALSE, sep="\t", row.names = F)

write.table(Allspecies_ExpressedGeneID_allRlog_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Allspecies_ExpressedGeneID_allRlog_db.tsv", quote=FALSE, sep="\t", row.names = F)


# ejemplo: https://github.com/hbctraining/publication_perfect/blob/main/lessons/04_boxplot_application_of_basic_plotting.md
```

```{r Analisis estadistico-Wilcoxtest}
library(ggpubr)
Allspecies_ExpressedGeneID_allRlog_test <- Allspecies_ExpressedGeneID_allRlog_db
Allspecies_ExpressedGeneID_allRlog_test$Biotype <- as.factor(Allspecies_ExpressedGeneID_allRlog_test$Biotype)
Allspecies_ExpressedGeneID_allRlog_test$specie <- as.factor(Allspecies_ExpressedGeneID_allRlog_test$specie)

anno_df_rlog <- compare_means(normalized_counts ~ Biotype, group.by = "specie", data = Allspecies_ExpressedGeneID_allRlog_test, method="wilcox.test") %>%
 mutate(y_pos = 30)

anno_df_rlog

head(anno_df_rlog)

# Columna
Allspecies_ExpressedGeneID_allRlog_test$BiotypeTissue <- paste(Allspecies_ExpressedGeneID_allRlog_test$Biotype, Allspecies_ExpressedGeneID_allRlog_test$DEG_tissue, sep="_")

# Two-way Anova
library(car)
library(dplyr)
anno_aov3 <- aov(normalized_counts ~ BiotypeTissue * specie, data = Allspecies_ExpressedGeneID_allRlog_test)
summary(anno_aov3)
model.tables(anno_aov3, type="means", se = TRUE)
# Compute two-way ANOVA test in R for unbalanced designs
Anova(anno_aov3, type = "III")
#https://github.com/kassambara/ggpubr/issues/65
#http://www.sthda.com/english/wiki/two-way-anova-test-in-r


# Comparaciones multiples
anno_df_rlog <-  compare_means(normalized_counts ~ BiotypeTissue, group.by = "specie", data = Allspecies_ExpressedGeneID_allRlog_test, method="wilcox.test", p.adjust.method="BH") %>%
 mutate(y_pos = 30)
anno_df_rlog

Allspecies_ExpressedGeneID_allRlog_test %>% group_by(specie) %>%
   compare_means(normalized_counts ~ BiotypeTissue, paired=TRUE, p.adjust.method="BH")

# https://www.datanovia.com/en/blog/how-to-perform-t-test-for-multiple-variables-in-r-pairwise-group-comparisons/
```



```{r allspecies_expression_protein_lncRNA_MAxtpm_v2_gg}
library(dplyr)
library(ggplot2)

#tiff(file="allspecies_expression_MaxRlog.tiff", width = 18, height = 10, units = 'in', res = 150)

# Cambiar Both por global
Allspecies_ExpressedGeneID_allRlog_db$DEG_tissue <- gsub("Both", "global", Allspecies_ExpressedGeneID_allRlog_db$DEG_tissue)

write.table(Allspecies_ExpressedGeneID_allRlog_db,file ="S:/Repositorio_scripts/Expressed_lncRNAs/Allspecies_ExpressedGeneID_allRlog_db.tsv", quote=FALSE, sep="\t", row.names = F)

Allspecies_ExpressedGeneID_allRlog_db %>% ggplot(aes(x=Biotype, y=normalized_counts)) + 
  #geom_point(aes(color=category), position=position_jitterdodge(), alpha=0.4, show.legend =FALSE) + 
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  facet_wrap(~specie, nrow=1) + 
  #scale_y_continuous(expand = c(0, 0), limits = c(-2, 20))+ # cero en la linea
 # ggsignif::geom_signif(
#    data=anno_df_rlog, 
 #   aes(xmin=group1, xmax=group2, annotations=p.signif, y_position=y_pos), 
#    manual=TRUE
#  ) +
  labs(y=expression('Normalized Counts (Rlog)'), x="") + #labels en X y Y
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


#dev.off()

```

```{r COWB_ExpGeneIDs_proportionDEG_db}
COWC_PatronesExp_db <- Allspecies_ExpressedGeneID_allRlog_db %>%  mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>% ggplot(aes(x=Biotype, y=normalized_counts)) +
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  facet_wrap(~specie, nrow=1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 

COWC_PatronesExp_db
```



## Expresion rlog - solo conservados

```{r all normalized counts Rlog- lincRNAs y proteinas}
# Modificar tabla para Contener toda la expresion por genes
library(tidyr)
library(tidyverse)
library(dplyr)

# Lista de ortologos

#> A. thaliana, lista de genes conservados mas las proteinas ortologas
At_expression_CSS_plot <- At_expression_plot %>% filter(GeneID %in% c(AtlincRNA_At_CSS_Expressed_GeneIDs$AtlincRNA_geneID, At_Orthologos_IDs_allspecies$Arabidopsis_thaliana))
#At_expression_CSS_plot %>% View()

# lncRNAs especie especifica
AtlincRNA_species_specific_IDs <- read.table("S:/Repositorio_scripts/Sequence_analysis/AtlincRNA_species_specific.tsv", header= F)
AtlincRNA_species_specific_IDs$V1 <- gsub("MSTRG.","AtRegRNA_", AtlincRNA_species_specific_IDs$V1)
At_expression_lncRNA_Specific_Specie_plot <- At_expression_plot %>% filter(GeneID %in% AtlincRNA_species_specific_IDs$V1)
                                
#> B. rapa
Br_expression_plot <-pivot_longer(Br_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

Br_expression_plot$specie <- "B. rapa"

#> B. oleracea
Bo_expression_plot <-pivot_longer(Bo_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

Bo_expression_plot$specie <- "B. oleracea"

#> C. rubella
Cr_expression_plot <-pivot_longer(Cr_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

Cr_expression_plot$specie <- "C. rubella"

#> T. parvula
Tp_expression_plot <-pivot_longer(Tp_ExpressedGeneID_db,
             cols = 1:4,
             names_to ="samples", # Ejecutar en todas las columnas
             values_to = "normalized_counts") # crear una nueva columna con la expresion, la cual esta normalizada en rlog

Tp_expression_plot$specie <- "T. parvula"


#unir
Allspecies_ExpressedGeneID_allRlog_db <- rbind(At_expression_plot, Br_expression_plot, Bo_expression_plot, Cr_expression_plot, Tp_expression_plot)

head(Allspecies_ExpressedGeneID_allRlog_db)

View(Allspecies_ExpressedGeneID_allRlog_db)
# ejemplo: https://github.com/hbctraining/publication_perfect/blob/main/lessons/04_boxplot_application_of_basic_plotting.md
```





```{r allspecies_expression_protein_lncRNA_MAxtpm_v2_gg}
library(dplyr)
library(ggplot2)

tiff(file="../Conservacion_x_secuencia/Paper_conservation/Athaliana_expression_MaxRlog.tiff", width = 9, height = 6, units = 'in', res = 150)


At_expression_CSS_plot %>% ggplot(aes(x=Biotype, y=normalized_counts)) + 
  #geom_point(aes(color=category), position=position_jitterdodge(), alpha=0.4, show.legend =FALSE) + 
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  #facet_wrap(~specie, nrow=1) + 
  #scale_y_continuous(expand = c(0, 0), limits = c(-2, 20))+ # cero en la linea
 # ggsignif::geom_signif(
#    data=anno_df_rlog, 
 #   aes(xmin=group1, xmax=group2, annotations=p.signif, y_position=y_pos), 
#    manual=TRUE
#  ) +
  labs(y=expression('Normalized Counts (Rlog)'), x="") + #labels en X y Y
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 

# lincRNA especie especifica
At_expression_lncRNA_Specific_Specie_plot %>% ggplot(aes(x=Biotype, y=normalized_counts)) + 
  #geom_point(aes(color=category), position=position_jitterdodge(), alpha=0.4, show.legend =FALSE) + 
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  #facet_wrap(~specie, nrow=1) + 
  #scale_y_continuous(expand = c(0, 0), limits = c(-2, 20))+ # cero en la linea
 # ggsignif::geom_signif(
#    data=anno_df_rlog, 
 #   aes(xmin=group1, xmax=group2, annotations=p.signif, y_position=y_pos), 
#    manual=TRUE
#  ) +
  labs(y=expression('Normalized Counts (Rlog)'), x="") + #labels en X y Y
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 

dev.off()



pdf(file="../Conservacion_x_secuencia/Paper_conservation/Athaliana_expression_MaxRlog.pdf", width = 9, height = 6)


At_expression_CSS_plot %>% ggplot(aes(x=Biotype, y=normalized_counts)) + 
  #geom_point(aes(color=category), position=position_jitterdodge(), alpha=0.4, show.legend =FALSE) + 
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  #facet_wrap(~specie, nrow=1) + 
  #scale_y_continuous(expand = c(0, 0), limits = c(-2, 20))+ # cero en la linea
 # ggsignif::geom_signif(
#    data=anno_df_rlog, 
 #   aes(xmin=group1, xmax=group2, annotations=p.signif, y_position=y_pos), 
#    manual=TRUE
#  ) +
  labs(y=expression('Normalized Counts (Rlog)'), x="") + #labels en X y Y
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "top", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


dev.off()

```


```{r}
At_expression_CSS_plot %>% View()

# Numero de genes conservados y expresados de manera global
At_expression_CSS_plot %>% filter(Biotype == "lincRNA" & DEG_tissue == "Both" & specie == "A. thaliana") %>% dplyr::select(GeneID) %>% n_distinct()

# Numero de genes conservados y expresados de manera raices
At_expression_CSS_plot %>% filter(Biotype == "lincRNA" & DEG_tissue == "root" & specie == "A. thaliana") %>% dplyr::select(GeneID) %>% n_distinct()

# Numero de genes conservados y expresados de manera parte aerea
At_expression_CSS_plot %>% filter(Biotype == "lincRNA" & DEG_tissue == "shoot" & specie == "A. thaliana") %>% dplyr::select(GeneID) %>% n_distinct()


# Numero de genes conservados y expresados de manera global
At_expression_CSS_plot %>% filter(Biotype == "protein" & DEG_tissue == "Both" & specie == "A. thaliana") %>% dplyr::select(GeneID) %>% n_distinct()

# Numero de genes conservados y expresados de manera raices
At_expression_CSS_plot %>% filter(Biotype == "protein" & DEG_tissue == "root" & specie == "A. thaliana") %>% dplyr::select(GeneID) %>% n_distinct()

# Numero de genes conservados y expresados de manera parte aerea
At_expression_CSS_plot %>% filter(Biotype == "protein" & DEG_tissue == "shoot" & specie == "A. thaliana") %>% dplyr::select(GeneID) %>% n_distinct()
```



```{r COWB_ExpGeneIDs_proportionDEG_db}
COWC_PatronesExp_db <- Allspecies_ExpressedGeneID_allRlog_db %>%  mutate(specie = factor(specie, levels = c("A. thaliana", "C. rubella", "T. parvula", "B. rapa", "B. oleracea"))) %>% ggplot(aes(x=Biotype, y=normalized_counts)) +
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  facet_wrap(~specie, nrow=1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 

COWC_PatronesExp_db
```


## Grafica de barras para expresion por genes

```{r Niveles de expresion 1 - IPS1}
Allspecies_ExpressedGeneID_allRlog_db %>% filter(GeneID == "AT3G09922" & specie == "A. thaliana") %>% ggplot(aes(x=samples, y=normalized_counts)) +
  geom_bar(stat="identity") + 
  #scale_fill_manual(values=c("#B47028", "#28B463" )) + #colores de relleno
  #facet_wrap(~specie, nrow=1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


```

```{r Niveles de expresion 1 - IPS1}
Allspecies_ExpressedGeneID_allRlog_db %>% filter(GeneID == "AT1G10682" & specie == "A. thaliana") %>% ggplot(aes(x=samples, y=normalized_counts)) +
  geom_bar(stat="identity") + 
  #scale_fill_manual(values=c("#B47028", "#28B463" )) + #colores de relleno
  #facet_wrap(~specie, nrow=1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


```


```{r Niveles de expresion 1 - IPS1}
Allspecies_ExpressedGeneID_allRlog_db %>% filter(GeneID == "AT1G10682" & specie == "A. thaliana") %>% ggplot(aes(x=samples, y=normalized_counts)) +
  geom_bar(stat="identity") + 
  #scale_fill_manual(values=c("#B47028", "#28B463" )) + #colores de relleno
  #facet_wrap(~specie, nrow=1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


```

```{r Niveles de expresion 1 - IPS1}
Allspecies_ExpressedGeneID_allRlog_db %>% filter(GeneID == "AT3G25795" & specie == "A. thaliana") %>% ggplot(aes(x=samples, y=normalized_counts)) +
  geom_bar(stat="identity") + 
  #scale_fill_manual(values=c("#B47028", "#28B463" )) + #colores de relleno
  #facet_wrap(~specie, nrow=1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


```


```{r Niveles de expresion 1 - IPS1}
Allspecies_ExpressedGeneID_allRlog_db %>% filter(GeneID == "AT3G25795" & specie == "A. thaliana") %>% ggplot(aes(x=samples, y=normalized_counts)) +
  geom_bar(stat="identity") + 
  #scale_fill_manual(values=c("#B47028", "#28B463" )) + #colores de relleno
  #facet_wrap(~specie, nrow=1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


```


```{r Niveles de expresion 1 - IPS1}
Allspecies_ExpressedGeneID_allRlog_db %>% arrange(desc(normalized_counts)) %>% filter(specie == "A. thaliana" & Biotype == "lincRNA") %>% head(20) %>% ggplot(aes(x=reorder(samples, normalized_counts), y=normalized_counts)) +
  geom_bar(stat="identity") + 
  #scale_fill_manual(values=c("#B47028", "#28B463" )) + #colores de relleno
  facet_wrap(~GeneID,  ncol =1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


```

```{r COWB_ExpGeneIDs_proportionDEG_db}
Allspecies_ExpressedGeneID_allRlog_db %>% arrange(desc(normalized_counts)) %>% 
  filter(specie == "A. thaliana" & Biotype == "lincRNA") %>% head(15) %>% 
  ggplot(aes(x=samples, y=normalized_counts)) +
  geom_boxplot(aes(fill=DEG_tissue), position=position_dodge(),  show.legend =TRUE) + 
  scale_fill_manual(values=c('#003f5c', "#B47028", "#28B463" )) + #colores de relleno
  facet_wrap(~GeneID, nrow=1) + 
  labs(y=expression('Cuentas normalizadas (Rlog)'), x="") + #labels en X y Y, Normalized Counts
  theme_classic() + # sin cuadricula con fondo blanco
  theme(text= element_text(size=20), 
        legend.position = "none", # posicion de la leyenda
        legend.title = element_blank(), # legend without title
        strip.background = element_blank(),  # sin relleno
        #axis.text.x = element_blank(),
        strip.text = element_text(face = "italic")) 


```




```{r paquetes}
#BiocManager::install('ComplexHeatmap')
#BiocManager::install('circlize')

require(RColorBrewer)
require(ComplexHeatmap)
require(circlize)
require(digest)
require(cluster)

# Matriz de expresion
AtlincRNA_allspecies_rlog_mat <- At_ExpressedGeneID_db %>% filter(Biotype == "lincRNA")

# metadata
#AtlincRNA_allspecies_metadata_TPM <- data.frame("Tissue" = rep(c(rep("shoot",2), rep("root",2)),5), "species" = c(rep("Arabidopsis_thaliana", 4), rep("Brassica_rapa", 4), rep("Brassica_oleracea", 4), rep("Capsella_rubella", 4), rep("Thellungiella_parvula", 4)))
#row.names(AtlincRNA_allspecies_metadata_TPM) <- names(AtlincRNA_allspecies_EXpTPM_CSS_mat)


# AtlincRNA_allspecies_GeneID_TPM_table_OnlyExpressed_completed[,c(26,27)]
  
#  
head(At_ExpressedGeneID_db)
tail(At_ExpressedGeneID_db)
head(AtlincRNA_allspecies_rlog_mat)
```


```{r}
# This is quite standard when performing clustering and generating a heatmap.
AtlincRNA_allspecies_rlog_mat <- AtlincRNA_allspecies_rlog_mat[,1:4] #Ordenaar tabla
  
AtlincRNA_allspecies_rlog_mat_heat <- t(scale(t(AtlincRNA_allspecies_rlog_mat)))
head(AtlincRNA_allspecies_rlog_mat_heat)

# colores (azul, negro hasta amarillo)
#myCol <- colorRampPalette(c('dodgerblue', 'black', 'yellow'))(100)
AtlincRNA_myBreaks <- seq(min(AtlincRNA_allspecies_rlog_mat_heat), max(AtlincRNA_allspecies_rlog_mat_heat), length.out = 100)


Heatmap(AtlincRNA_allspecies_rlog_mat_heat, name = "mat", rect_gp = gpar(col = "white", lwd = 2),
    column_title = "set cell borders")



#annotation that we want to use
  # In this example, the 'ann' object turns out to be the exact same as 'metadata'

# "Arabidopsis_thaliana", 4), rep("Brassica_rapa", 4), rep("Brassica_oleracea", 4), rep("Capsella_rubella", 4), rep("Thellungiella_parvula", 4
  AtlincRNA_allspecies_ann <- data.frame(
    Tissue = AtlincRNA_allspecies_metadata_TPM$Tissue,
    Specie = AtlincRNA_allspecies_metadata_TPM$species,
    stringsAsFactors = FALSE)

  # create the colour mapping
  # color in columns by tissues
  AtlincRNA_allspecies_ha = HeatmapAnnotation(tissue = c("shoot", "shoot", "root", "root"),
    col = list(
    Tissue = c('shoot' = '#4daf4a', 'root' = '#ff7f00'),
    Specie = c('Arabidopsis_thaliana' = "#081d58", 'Brassica_rapa' = '#1d91c0', 'Brassica_oleracea' = '#7fcdbb', 'Capsella_rubella'= '#8c6bb1', 'Thellungiella_parvula'='#810f7c'))
    )


  AtlincRNA_allspecies_HM_colours <- list(
    Tissue = c('shoot' = '#4daf4a', 'root' = '#ff7f00'),
    Specie = c('Arabidopsis_thaliana' = "#081d58", 'Brassica_rapa' = '#1d91c0', 'Brassica_oleracea' = '#7fcdbb', 'Capsella_rubella'= '#8c6bb1', 'Thellungiella_parvula'='#810f7c'))

  # now create the ComplexHeatmap annotation object
  # as most of these parameters are self-explanatory, comments will only appear where needed
  AtlincRNA_allspecies_CSSandSpeciesSpecific_colAnn <- HeatmapAnnotation(
    df = AtlincRNA_allspecies_ann,
    which = 'col', # 'col' (samples) or 'row' (gene) annotation?
    na_col = 'white', # default colour for any NA values in the annotation data-frame, 'ann'
    col = AtlincRNA_allspecies_HM_colours,
    annotation_height = 0.6,
    annotation_width = unit(1, 'cm'),
    gap = unit(1, 'mm'),
    annotation_legend_param = list(
      Tissue = list(
        nrow = 2,
        title = 'Tissues',
        title_position = 'topcenter',
        legend_direction = 'vertical',
        title_gp = gpar(fontsize = 12, fontface = 'bold'),
        labels_gp = gpar(fontsize = 12, fontface = 'bold')),
      Specie = list(
        nrow = 5, # number of rows across which the legend will be arranged
        title = 'Species',
        title_position = 'topcenter',
        legend_direction = 'vertical',
        title_gp = gpar(fontsize = 12, fontface = 'bold'),
        labels_gp = gpar(fontsize = 12, fontface = 'bold'))

      ))

  # Nombres en row
AtlincRNAs_shoot_genelabels <- rowAnnotation(
      Genes = anno_mark(
      at = AtlincRNAs_shoot_positions, #seq(1, nrow(AtlincRNAs_heat), 40), 
      labels = rownames(AtlincRNAs_ONlyshoot_heat)[AtlincRNAs_shoot_positions],
      labels_gp = gpar(fontsize = 5, fontface = 'bold'),
      padding = 0.75),
      width = unit(2.0, 'cm') +
      max_text_width(rownames(AtlincRNAs_ONlyshoot_heat)[AtlincRNAs_shoot_positions],
      gp = gpar(fontsize = 5,  fontface = 'bold')))


Heatmap(AtlincRNA_allspecies_rlog_mat_heat,
        name = 'rlog', #nombre
        # Ordena grafica
        column_split = 2, #Split by dendrogram
        clustering_distance_rows = "pearson", # Dendogramas de correlacion de pearson
        row_dend_reorder = TRUE, #orden
        cluster_columns = agnes(t(AtlincRNA_allspecies_rlog_mat_heat)), # Clustering de muestras
        show_column_names = TRUE, column_title = NULL, row_title=NULL) # sin nombre de la columna
```



```{r}
# colores
myCol <- colorRampPalette(c('#2166ac', '#67a9cf', '#d1e5f0','#f7f7f7','#fddbc7', '#ef8a62', '#b2182b'))(100)
#myCol <- colorRampPalette(c('#2166ac', '#f7f7f7','#b2182b'))(100)

#https://loading.io/color/feature/RdBu-7/
#myCol <- grDevices::colorRampPalette(RColorBrewer::brewer.pal(7, "RdBu"))
#https://github.com/talgalili/heatmaply/blob/master/R/RColorBrewer.R

AtlincRNA_allspecies_CSSandSpeciesSpecific_hmap <- Heatmap(AtlincRNA_allspecies_EXpTPM_CSS_heat,
        # BArra de expresion
        name = 'Gene\nZ-score\nlog2 TPM', #nombre
        col = colorRamp2(AtlincRNA_myBreaks, myCol), # colores de azul, negro hasta amarillo
        
        # parameters for the colour-bar that represents gradient of expression
        heatmap_legend_param = list(
          color_bar = "continuous",
          at = c(-3.75,-2,-1,0,1,2,3.75),  #Especificar escalas en la barra de expresion
          legend_direction = 'vertical',
          legend_width = unit(8, 'cm'),
          legend_height = unit(5.0, 'cm'),
          title_position = 'topcenter',
          title_gp=gpar(fontsize = 12, fontface = 'bold'),
          labels_gp=gpar(fontsize = 12, fontface = 'bold'),
          direction = "vertical"),
        
        # Ordena grafica
        row_dend_reorder = TRUE, #orden

        # Dividir por tejidos
        #column_split = rep(c("Shoot", "Root"), 10),

        #row_km = 0, column_km = 2,  row_gap = unit(3, "mm"), #separaciones por clusters
        #top_annotation = ha,      # nombres de los genes conservados por sintenia y microsecuencia
        show_column_names = TRUE, # sin nombre de la columna
        column_title = NULL,
        row_title=NULL)
        

        

draw(AtlincRNA_allspecies_CSSandSpeciesSpecific_hmap,
    merge_legends =TRUE, #unir las leyendas
    heatmap_legend_side = 'left',
    annotation_legend_side = 'left',
    row_sub_title_side = 'left')
```



## Unir grafica

```{r}
library(patchwork)

tiff(file="FigS2CD_DEGlincRNAs_PC_gene.tiff", width = 17, height = 10, units = 'in', res = 300)
(COWB_ExpGeneIDs_proportionDEG_db | COWA_ExpGeneIDs_proportion | COWC_PatronesExp_db) + plot_layout(ncol = 3, byrow = TRUE) + plot_annotation(tag_levels = 'A', title = 'Fig S2')
dev.off()
```





## Anotacion funcional

```{r Paquetes, eval=FALSE}
#instalacion de BiocManager
#if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

# Obtencion de bases de datos
#BiocManager::install("biomaRt")

#Analisis diferencial
#BiocManager::install("edgeR") # expresion diferencial
#BiocManager::install("statmod") #paquete para graficar diagramas de dispersion

# Graficas
#install.packages("gplots") # graficar heatmap
#install.packages("ggplot2") # graficar las demas graficas
#install.packages("ggpmisc") # extensionn de ggplot2, para la regresion lineal entre replicas

# Asignacion de terminos GO y KEGG
#BiocManager::install("topGO")
#BiocManager::install("Rgraphviz")
```


Obtencion de bases de datos

Se descargaron por medio del paquete *biomaRt* la base de datos de *plant emsembl*, seleccionando la categoria de *plants_mart*, la cual contiene los genes de 47 especies. Para mas informacion sobre el empleo de biomart: [manual_biomaRT](https://bioconductor.org/packages/3.12/bioc/manuals/biomaRt/man/biomaRt.pdf) y [guia](https://m.ensembl.org/info/data/biomart/biomart_r_package.html).

```{r}
#Bases de datos contenidas en ensembl plant
library(biomaRt)
listMarts(host="plants.ensembl.org")

# hongos-David
listMarts(host="ensembl.org")

#Lista de especies contenidas
plant_ensembl <- useEnsembl(biomart="plants_mart", host ="plants.ensembl.org" )
head(listDatasets(plant_ensembl))

# todos lo que contiene enssembl
all_ensembl <- useEnsembl(biomart="ensembl")
head(listDatasets(all_ensembl))


View(listDatasets(all_ensembl))
```

Se seleccionaron los de *A.thaliana* senalados como **athaliana_eg_gene** y los de *B.rapa* como **brapa_eg_gene**.

<div class="alert alert-block alert-warning">
<b>Nota:</b>*B. oleracea* es **boleracea_eg_gene**, pero *C. rubella* ni *T. parvula* aparecen en esta base de datos.
</div>


```{r EnsemblPlants-At, echo=T}
#NO EJECUTAR- variable ya cargada
athaliana_database <- useMart(biomart = "plants_mart", host = "plants.ensembl.org", dataset = "athaliana_eg_gene")
head(listFilters(athaliana_database))
head(listAttributes(athaliana_database))
```


```{r EnsemblPlants-Br, echo=T}
#NO EJECUTAR- variable ya cargada
brapa_database <- useMart(biomart = "plants_mart", host = "plants.ensembl.org", dataset = "brapa_eg_gene")
head(listFilters(brapa_database))
head(listAttributes(brapa_database))
#View(listAttributes(brapa_database))
```

Extraer biotipos de genes
En esta seccion se extraen los id de los genes pertenecientes a las categorias: **ncRNAs, snoRNAs, snRNAs, tRNA, rRNA, lncRNA, protein-coding, pre-miRNAs**, etc.

```{r Extraer biotipos de genes - At, echo=T}
athaliana_biotype_genes <- getBM(c("ensembl_gene_id","gene_biotype"), mart=athaliana_database)
unique(athaliana_biotype_genes$gene_biotype)

```

```{r Extraer biotipos de genes - Br, echo=T}
brapa_biotype_genes <- getBM(c("ensembl_gene_id","gene_biotype"), mart=brapa_database)
unique(brapa_biotype_genes$gene_biotype)
```


## Analisis de terminos GO (ProteinCodingGenes)

Se empleo la funcion **getBM** para filtrar los datos previamente recolectados de Ensembl Plant con la paqueteria **biomaRT**. Ademas, lso datos se guardaron en forma de lista para ser procesados posteriormente por **TopGO (ENSEMBL)**. *Annotationhub (NCBI)*

```{r Extraer los terminos GO de Ensembl Plants -At}
library("topGO")
#BiocManager::install("topGO")
# Extraer los terminos GO
At_GOgenes <- getBM(attributes=c('ensembl_gene_id', 'go_id'), mart=athaliana_database)

head(At_GOgenes)
# Guardamos la lista de genes un archivo tsv.
write.table(At_GOgenes, file = "./At_geneid2go.map",sep = "\t", row.names = FALSE, col.names = FALSE, quote = FALSE)
```

```{r Extraer los terminos GO de Ensembl Plants -Br, eval=F}
# Extraer los terminos GO
Br_GOgenes <- getBM(attributes=c('ensembl_gene_id', 'go_id'), mart=brapa_database)

# Guardamos la lista de genes un archivo tsv.
write.table(Br_GOgenes, file = "./Br_geneid2go.map",sep = "\t", row.names = FALSE, col.names = FALSE, quote = FALSE)
```

### A) Conversion de Ensembl IDs a Entrez IDs

```{r Universo y expression-ProteinCodingGenes-At}
#---------------------- A) Conversion de Ensembl IDs a Entrez IDs----------------------
library(topGO)
# > Universo
# Despues de esto la podemos leer y transformarla al formato necesario para poder usarla como base de datos por topGO
At_geneID2GO <- readMappings("./At_geneid2go.map")
str(head(At_geneID2GO))

#Obtener nombre de los genes de GO a gene ID
GO2geneID <- inverseList(At_geneID2GO)
str(head(GO2geneID))

# Se tiene que modificar la lista de genes para que pueda ser utilizada por topGO
At_geneNames <- names(At_geneID2GO) # universo
head(At_geneNames)
```

```{r Universo y expression-ProteinCodingGenes-Br}
#---------------------- A) Conversion de Ensembl IDs a Entrez IDs----------------------
library(topGO)
# > Universo
# Despues de esto la podemos leer y transformarla al formato necesario para poder usarla como base de datos por topGO
Br_geneID2GO <- readMappings("./Br_geneid2go.map")
str(head(Br_geneID2GO))

#Obtener nombre de los genes de GO a gene ID
Br_GO2geneID <- inverseList(Br_geneID2GO)
str(head(Br_GO2geneID))

# Se tiene que modificar la lista de genes para que pueda ser utilizada por topGO
Br_geneNames <- names(Br_geneID2GO) # universo
head(Br_geneNames)
```


```{r Protein-coding genes-ProteinCodingGenes-At}
#seleccionar solo conservados (genes codificantes ortologos)
At_protein_de_gene_matrix_UP_db <- as.data.frame(At_protein_de_gene_matrix_UP)
At_protein_de_gene_matrix_UP_db$GeneID <- rownames(At_protein_de_gene_matrix_UP_db)

At_protein_de_gene_names_Ortologos_UP <-  rownames(At_protein_de_gene_matrix_UP_db %>% 
  filter(GeneID %in% At_Orthologos_IDs_allspecies$Arabidopsis_thaliana))

At_protein_de_gene_matrix_DOWN_db <- as.data.frame(At_protein_de_gene_matrix_DOWN)
At_protein_de_gene_matrix_DOWN_db$GeneID <- rownames(At_protein_de_gene_matrix_DOWN_db)

At_protein_de_gene_names_Ortologos_DOWN <-  rownames(At_protein_de_gene_matrix_DOWN_db %>% 
  filter(GeneID %in% At_Orthologos_IDs_allspecies$Arabidopsis_thaliana))

# > Parte aerea
#genes diferencialmente expresados en la parte aerea
#At_gene.names.up.go <- unlist(strsplit(At_protein_de_gene_names_UP, split = "|", fixed = TRUE))
#At_geneList.up <- factor(as.integer(At_geneNames %in% At_protein_de_gene_names_UP))
At_gene.names.up.go <- unlist(strsplit(At_protein_de_gene_names_Ortologos_UP, split = "|", fixed = TRUE))
At_geneList.up <- factor(as.integer(At_geneNames %in% At_protein_de_gene_names_Ortologos_UP))
names(At_geneList.up) <- At_geneNames
head(At_geneList.up)
str(At_geneList.up)

# > Raices
# Se tiene que modificar la lista de genes para que pueda ser utilizada por topGO
#At_gene.names.down.go <- unlist(strsplit(At_protein_de_gene_names_DOWN, split = "|", fixed = TRUE))
#At_geneList.down <- factor(as.integer(At_geneNames %in% At_protein_de_gene_names_DOWN))
At_gene.names.down.go <- unlist(strsplit(At_protein_de_gene_names_Ortologos_DOWN, split = "|", fixed = TRUE))
At_geneList.down <- factor(as.integer(At_geneNames %in% At_protein_de_gene_names_Ortologos_DOWN))
names(At_geneList.down) <- At_geneNames
head(At_geneList.down)

#---------------### B) Obtencion de terminos GO---------------------------------
# > Parte aerea
# Despues de esto y con la lista de genes previamente identificados podemos generar un objeto GO.
At_GOdata.up <- new("topGOdata", ontology = "BP", allGenes = At_geneList.up, 
              annot = annFUN.gene2GO, gene2GO= At_geneID2GO)
At_GOdata.up

#Numero de genes
numGenes(At_GOdata.up)

# > Raices
At_GOdata.down <- new("topGOdata", ontology = "BP", allGenes = At_geneList.down, 
              annot = annFUN.gene2GO, gene2GO= At_geneID2GO)
At_GOdata.down

#Numero de genes
numGenes(At_GOdata.down)

#------------### C) Prueba de Fisher-------------------------------
# > Parte aerea
At_resultsFisher.up <- runTest(At_GOdata.up, algorithm = "classic", statistic = "fisher")
At_resultsFisher.up

#> Raices
At_resultsFisher.down <- runTest(At_GOdata.down, algorithm = "classic", statistic = "fisher")
At_resultsFisher.down

#Nombre de los genes
At_shoot_a <- genes(At_GOdata.up)
#At_shoot_a
At_root_a <- genes(At_GOdata.down)

# una muestra de genes mas interesantes
At_shoot_selGenes <- sample(At_shoot_a, 10) 
At_shoot_gs <- geneScore(At_GOdata.up, whichGenes = At_shoot_selGenes)

At_root_selGenes <- sample(At_root_a, 10) 
At_root_gs <- geneScore(At_GOdata.down, whichGenes = At_root_selGenes)

# > Muestra de genes anotados
sg <- sigGenes(At_GOdata.up)
str(sg)

# Numero de genes anotados
numSigGenes(At_GOdata.up) # 22850 genes con terminos GO en shoot
numSigGenes(At_GOdata.down) #22512 genes con terminos GO en root

# ----------------Top 10 numero de genes encontrados en los GO----------
sel.terms <- sample(usedGO(At_GOdata.up), 10)
num.ann.genes <- countGenesInTerm(At_GOdata.up, sel.terms) ## the number of annotated genes
num.ann.genes

ann.genes <- genesInTerm(At_GOdata.up, sel.terms) ## get the annotations
head(ann.genes)

# estadistico
termStat(At_GOdata.up, sel.terms)
```


```{r Protein-coding genes-ProteinCodingGenes-Br}
# > Parte aerea
#genes diferencialmente expresados en la parte aerea
Br_gene.names.up.go <- unlist(strsplit(Br_protein_de_gene_names_UP, split = "|", fixed = TRUE))
Br_geneList.up <- factor(as.integer(Br_geneNames %in% Br_protein_de_gene_names_UP))
names(Br_geneList.up) <- Br_geneNames
head(Br_geneList.up)
str(Br_geneList.up)

# > Raices
# Se tiene que modificar la lista de genes para que pueda ser utilizada por topGO
Br_gene.names.down.go <- unlist(strsplit(Br_protein_de_gene_names_DOWN, split = "|", fixed = TRUE))
Br_geneList.down <- factor(as.integer(Br_geneNames %in% Br_protein_de_gene_names_DOWN))
names(Br_geneList.down) <- Br_geneNames
head(Br_geneList.down)

#---------------### B) Obtencion de terminos GO---------------------------------
# > Parte aerea
# Despues de esto y con la lista de genes previamente identificados podemos generar un objeto GO.
Br_GOdata.up <- new("topGOdata", ontology = "BP", allGenes = Br_geneList.up, 
              annot = annFUN.gene2GO, gene2GO= Br_geneID2GO)
Br_GOdata.up

#Numero de genes
numGenes(Br_GOdata.up)

# > Raices
Br_GOdata.down <- new("topGOdata", ontology = "BP", allGenes = Br_geneList.down, 
              annot = annFUN.gene2GO, gene2GO= Br_geneID2GO)
Br_GOdata.down

#Numero de genes
numGenes(Br_GOdata.down)

#------------### C) Prueba de Fisher-------------------------------
# > Parte aerea
Br_resultsFisher.up <- runTest(Br_GOdata.up, algorithm = "classic", statistic = "fisher")
Br_resultsFisher.up

#> Raices
Br_resultsFisher.down <- runTest(Br_GOdata.down, algorithm = "classic", statistic = "fisher")
Br_resultsFisher.down

#Nombre de los genes
Br_shoot_a <- genes(Br_GOdata.up)
#At_shoot_a
Br_root_a <- genes(Br_GOdata.down)

# una muestra de genes mas interesantes
Br_shoot_selGenes <- sample(Br_shoot_a, 10) 
Br_shoot_gs <- geneScore(Br_GOdata.up, whichGenes = Br_shoot_selGenes)

Br_root_selGenes <- sample(Br_root_a, 10) 
Br_root_gs <- geneScore(Br_GOdata.down, whichGenes = Br_root_selGenes)

# > Muestra de genes anotados
sg <- sigGenes(Br_GOdata.up)
str(sg)

# Numero de genes anotados
numSigGenes(Br_GOdata.up) # 22850 genes con terminos GO en shoot
numSigGenes(Br_GOdata.down) #22512 genes con terminos GO en root

# ----------------Top 10 numero de genes encontrados en los GO----------
sel.terms <- sample(usedGO(Br_GOdata.up), 10)
num.ann.genes <- countGenesInTerm(Br_GOdata.up, sel.terms) ## the number of annotated genes
num.ann.genes

ann.genes <- genesInTerm(Br_GOdata.up, sel.terms) ## get the annotations
head(ann.genes)

# estadistico
termStat(Br_GOdata.up, sel.terms)
```


### D) Lista de terminos GOs

```{r Terminos GO -ProteinCodingGenes -At}
library(topGO)
# > Terminos GO encontrados en la parte aerea
# Summary de todos los resultados. 
At_shoot.table <- GenTable(At_GOdata.up, classicFisher = At_resultsFisher.up,
         orderBy = "classicFisher", ranksOf = "classicFisher", topNodes = 10)

# corregir de chr a num en classicFisher
At_shoot.table$classicFisher <- as.numeric(At_shoot.table$classicFisher)

# Filtrado
At_shootGO <- as.data.frame(At_shoot.table[At_shoot.table$classicFisher <= 0.05,])
At_shootGO
str(At_shootGO)
dim(At_shootGO)

# >Terminos GO encontrados en las raices
# Summary de todos los resultados. 
At_root.table <- GenTable(At_GOdata.down, classicFisher = At_resultsFisher.down,
         orderBy = "classicFisher", ranksOf = "classicFisher", topNodes = 10)


showGroupDensity(At_GOdata.down, "GO:0010015", ranks = FALSE, rm.one = FALSE)

# corregir de chr a num en classicFisher
At_root.table$classicFisher <- as.numeric(At_root.table$classicFisher)

At_rootGO <- as.data.frame(At_root.table[At_root.table$classicFisher <= 0.05,])
At_rootGO
str(At_rootGO)
dim(At_rootGO)
```

```{r Terminos GO-ProteinCodingGenes -Br}
library(topGO)
# > Terminos GO encontrados en la parte aerea
# Summary de todos los resultados. 
Br_shoot.table <- GenTable(Br_GOdata.up, classicFisher = Br_resultsFisher.up,
         orderBy = "classicFisher", ranksOf = "classicFisher", topNodes = 10)

# corregir de chr a num en classicFisher
Br_shoot.table$classicFisher <- as.numeric(Br_shoot.table$classicFisher)

# Filtrado
Br_shootGO <- as.data.frame(Br_shoot.table[Br_shoot.table$classicFisher <= 0.05,])
Br_shootGO
str(Br_shootGO)
dim(Br_shootGO)

# >Terminos GO encontrados en las raices
# Summary de todos los resultados. 
Br_root.table <- GenTable(Br_GOdata.down, classicFisher = Br_resultsFisher.down,
         orderBy = "classicFisher", ranksOf = "classicFisher", topNodes = 10)


showGroupDensity(Br_GOdata.down, "GO:0010015", ranks = FALSE, rm.one = FALSE)

# corregir de chr a num en classicFisher
Br_root.table$classicFisher <- as.numeric(Br_root.table$classicFisher)

Br_rootGO <- as.data.frame(Br_root.table[Br_root.table$classicFisher <= 0.05,])
Br_rootGO
str(Br_rootGO)
dim(Br_rootGO)
```


### E) Grafica de los terminos GO en A. thaliana


```{r dataframe para shoot y root -ProteinCodingGenes -At}
#Parte Aerea
At_shoot_visualplot <- At_shootGO[,c(1,2,3,6)] #col ID, term, annotated and classicFisher
head(At_shoot_visualplot)
dim(At_shoot_visualplot)

#raices
At_root_visualplot <- At_rootGO[,c(1,2,3,6)]
head(At_root_visualplot)
dim(At_root_visualplot)
```


```{r dataframe para shoot y root -ProteinCodingGenes -Br}
#Parte Aerea
Br_shoot_visualplot <- Br_shootGO[,c(1,2,3,6)] #col ID, term, annotated and classicFisher
head(Br_shoot_visualplot)
dim(Br_shoot_visualplot)

#raices
Br_root_visualplot <- Br_rootGO[,c(1,2,3,6)]
head(Br_root_visualplot)
dim(Br_root_visualplot)
```

```{r Grafica Dotplot dde Goterms-ProteinCodingGenes-At}
library(ggplot2)

#conversion de chr a num
At_shoot_visualplot$classicFisher <- as.numeric(At_shoot_visualplot$classicFisher)
str(At_shoot_visualplot)
At_shoot_visualplot$Organ <- "shoot"

At_root_visualplot$classicFisher <- as.numeric(At_root_visualplot$classicFisher)
str(At_root_visualplot)
At_root_visualplot$Organ <- "root"

# unir ambos datos
At_organ_visualplot <- rbind(At_shoot_visualplot, At_root_visualplot)
colnames(At_organ_visualplot) <- c("GO.ID", "Term", "Annotated", "padjust", "Organ")
# Orden
At_organ_visualplot <-At_organ_visualplot[order(At_organ_visualplot$Annotated, decreasing = T), ]

library(ggplot2)

At_organ_visualplot %>% mutate(Term = factor(Term, levels = c("protein import into chloroplast stroma", "chloroplast RNA modification", "protein localization to chloroplast", "establishment of protein localization to...", "protein targeting to chloroplast", "photosynthesis, light reaction", "trichoblast maturation", "cell maturation", "trichoblast differentiation", "photosynthesis", "root epidermal cell differentiation", "phenylpropanoid metabolic process", "response to red or far red light", "response to chitin", "root morphogenesis", "response to light stimulus", "response to radiation", "response to oxygen-containing compound", "response to chemical", "response to stimulus"))) %>% ggplot(aes(x=Organ, y = Term, size= Annotated, color =padjust)) + geom_point(alpha=1.0) +
  scale_color_gradient(low="blue", high="red") +
  labs(title = "Enriquecimiento de terminos GO (protein-coding genes)", #GO Biological processes
       x = "A.thaliana", y = "",
      caption = 'Cut-off of padj=0.05') + # agregar x y y label
  theme_light() +
  theme(text = element_text(size=12, family="sans"), 
        axis.title.x = element_text(face = "italic",  family="sans")) # controlar label x

# classicFisher = p.adjust
# Annotated = Count
```

```{r Barplot Goterms- ProteinCodingGenes-At}
library(ggplot2)
# At_organ_visualplot %>% arrange(desc(Annotated))
At_GOterm_organ_visualplot <- At_organ_visualplot %>% ggplot(aes(x=Annotated, y = reorder(Term, Annotated), fill=Organ)) + 
  geom_bar(stat = "identity") +
  geom_text(
        aes(label = Annotated), # Agregar informacion en numeros
        color = "black",
        size = 4,
        hjust=0.5,
        position = position_dodge(0.9)) +
  scale_fill_manual(values=c("#B47028", "#28B463")) +#colores de relleno, cafe y verde (root and shoot)
  facet_wrap(~Organ, scales="free") +
  labs(title = "Enriquecimiento de terminos GO (protein-coding genes)", #GO Biological processes
       x = "A.thaliana", y = "",
      caption = 'Cut-off of padj=0.05') + # agregar x y y label
  theme_classic()+
  theme(text = element_text(size=12, family="sans"), 
        axis.title.x = element_text(face = "italic",  family="sans"),# controlar label x
        legend.position = "none", 
        legend.title = element_blank(), 
        strip.background = element_blank()) 

At_GOterm_organ_visualplot

library(cowplot)
save_plot("../Conservacion_x_secuencia/Tesis_figuras/Figura_At_GOterms.png", At_GOterm_organ_visualplot, base_height = 5, base_width = 8, base_asp = 1.1)

save_plot("../Conservacion_x_secuencia/Tesis_figuras/Figura_At_GOterms.pdf", At_GOterm_organ_visualplot, base_height = 5, base_width = 8, base_asp = 1.1)

```


```{r Grafica Dotplot dde Goterms-ProteinCodingGenes-Br}
library(ggplot2)

#conversion de chr a num
Br_shoot_visualplot$classicFisher <- as.numeric(Br_shoot_visualplot$classicFisher)
str(Br_shoot_visualplot)
Br_shoot_visualplot$Organ <- "shoot"

Br_root_visualplot$classicFisher <- as.numeric(Br_root_visualplot$classicFisher)
str(Br_root_visualplot)
Br_root_visualplot$Organ <- "root"

# unir ambos datos
Br_organ_visualplot <- rbind(Br_shoot_visualplot, Br_root_visualplot)
colnames(Br_organ_visualplot) <- c("GO.ID", "Term", "Annotated", "padjust", "Organ")
# Orden
Br_organ_visualplot <-Br_organ_visualplot[order(Br_organ_visualplot$Annotated, decreasing = T), ]

library(ggplot2)

Br_organ_visualplot %>% ggplot(aes(x=Organ, y = Term, size= Annotated, color =padjust)) + geom_point(alpha=1.0) +
  scale_color_gradient(low="blue", high="red") +
  labs(title = "Enriquecimiento de terminos GO (protein-coding genes)", #GO Biological processes
       x = "B. rapa", y = "",
      caption = 'Cut-off of padj=0.05') + # agregar x y y label
  theme_light() +
  theme(text = element_text(size=12, family="sans"), 
        axis.title.x = element_text(face = "italic",  family="sans")) # controlar label x

# classicFisher = p.adjust
# Annotated = Count
```

```{r Barplot Goterms- ProteinCodingGenes-Br}
library(ggplot2)
Br_organ_visualplot %>% ggplot(aes(x=Annotated, y = Term, fill=Organ)) + 
  geom_bar(stat = "identity") +
  geom_text(
        aes(label = Annotated), # Agregar informacion en numeros
        color = "black",
        size = 4,
        hjust=0.5,
        position = position_dodge(0.9)) +
  scale_fill_manual(values=c("#B47028", "#28B463")) +#colores de relleno, cafe y verde (root and shoot)
  facet_wrap(~Organ, scales="free") +
  labs(title = "Enriquecimiento de terminos GO (protein-coding genes)", #GO Biological processes
       x = "B. rapa", y = "",
      caption = 'Cut-off of padj=0.05') + # agregar x y y label
  theme_classic()+
  theme(text = element_text(size=12, family="sans"), 
        axis.title.x = element_text(face = "italic",  family="sans"),# controlar label x
        legend.position = "none", 
        legend.title = element_blank(), 
        strip.background = element_blank()) 


```







## Reducir terminos con REVIGO

```{r TreeMAp con revigo y condensacion de Goterm-At}
#BiocManager::install("rrvgo")
library(rrvgo)
#BiocManager::install("org.At.tair.db")

#go_analysis <- read.delim(system.file("extdata/example.txt", package="rrvgo"))
simMatrix <- calculateSimMatrix(At_shoot_visualplot$GO.ID,
                                orgdb="org.At.tair.db",
                                ont="BP",
                                method="Rel")

scores <- setNames(-log10(At_shoot_visualplot$classicFisher), At_shoot_visualplot$GO.ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.At.tair.db")
# Otras especies https://www.bioconductor.org/packages/devel/bioc/vignettes/rrvgo/inst/doc/rrvgo.html
# https://www.bioconductor.org/packages/devel/bioc/vignettes/rrvgo/inst/doc/rrvgo.html

# Grafica en burbuja
#scatterPlot(simMatrix, reducedTerms)
# Grafica de letras
#wordcloudPlot(reducedTerms, min.freq=1, colors="black")

# Reducir el numero de Goterms en clusters
treemapPlot(reducedTerms)
```


```{r Checar terminos}
head(reducedTerms)
```

```{r}
library(dplyr)
reducedTerms %>% filter(parentTerm == "photosynthesis") #%>% select(size)%>% colSums()

#
At_shoot_visualplot %>% filter(GO.ID %in% reducedTerms$go) #%>% select(Annotated) %>% colSums()


reducedTerms %>% filter(term == "photosynthesis") #%>% select(size)%>% colSums()
At_shoot_visualplot %>% filter(Term == "photosynthesis") #%>% select(Annotated) %>% colSums()

```
































